{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2997e421",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os \n",
    "import pandas as pd \n",
    "pd.get_option(\"display.max_columns\", None)\n",
    "pd.options.display.max_columns = None\n",
    "import numpy as np\n",
    "import glob\n",
    "import json\n",
    "from IPython.display import HTML\n",
    "os.chdir(\"/mnt/BioHome/jreyna/jreyna/projects/dchallenge/\")\n",
    "outdir = 'results/main/sgls/combined/'\n",
    "os.makedirs(outdir, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "54972544",
   "metadata": {},
   "outputs": [],
   "source": [
    "major_cols = [\n",
    " 'gwas_source',\n",
    " 'ge_source',\n",
    " 'loop_source',\n",
    " 'sid',\n",
    " 'rsid',\n",
    " 'geneid',\n",
    " 'gene_name',\n",
    " 'chrom',\n",
    " 'snp_pos',\n",
    " 'tss_start',\n",
    " 'tss_end',\n",
    " 'is_eqtl_pair',\n",
    " 'is_coloc_pair',\n",
    " 'is_closest_gene',\n",
    " 'has_fithichip_loop',\n",
    " 'eqtl_pval',\n",
    " #'eqtl_fdr',\n",
    " 'dist',\n",
    " 'ppH4',\n",
    " 'gene_start',\n",
    " 'gene_end',\n",
    " 'ref',\n",
    " 'alt',\n",
    " 'AC',\n",
    " 'AF',\n",
    " 'AN',\n",
    " 'gwas_pval',\n",
    " 'gene_strand',\n",
    " 'eqtl_source']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e09c3e78",
   "metadata": {},
   "source": [
    "## Loading GENCODE information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "a7eddbea",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_info = pd.read_table('results/refs/gencode/v30/gencode.v30.annotation.bed', header=None)\n",
    "gene_dict = pd.concat([gene_info.iloc[:, [5,6]], gene_info.iloc[:, [6,5]]])\n",
    "gene_dict.drop_duplicates(inplace=True)\n",
    "gene_dict.set_index(5, inplace=True)\n",
    "gene_dict = gene_dict.squeeze()\n",
    "gene_dict = gene_dict.to_dict()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "434c4499",
   "metadata": {},
   "source": [
    "## Checking the samplesheet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "78fbfc17",
   "metadata": {},
   "outputs": [],
   "source": [
    "template = 'results/main/GRCh37/sgls/ldpairs/eQTL_Catalogue/{gwas_source}/{eqtl_source}/{ge_source}/{loop_source}/script_version/master.tsv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "c4f3e0ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "samplesheet = pd.read_table('config/sgl_samplesheets/sgl.samplesheet.tsv', comment='#')\n",
    "new_cols = []\n",
    "for i, sr in samplesheet.iterrows():\n",
    "    \n",
    "    d = {'gwas_source': sr.gwas,\n",
    "         'eqtl_source': sr.eqtl_db,\n",
    "         'ge_source': sr.eqtl_origin,\n",
    "         'loop_source': sr.loop_origin}\n",
    "    master_fn = template.format(**d)\n",
    "    test = os.path.exists(master_fn)\n",
    "    new_cols.append([test, master_fn])\n",
    "new_cols = np.array(new_cols)\n",
    "\n",
    "samplesheet['analyzed'] = new_cols[:, 0]\n",
    "samplesheet['filename'] = new_cols[:, 1]\n",
    "samplesheet.sort_values(['analyzed', 'eqtl_db', 'eqtl_origin', 'loop_origin'], \\\n",
    "                       ascending=[False, True, True, True], inplace=True)\n",
    "samplesheet.reset_index(drop=True, inplace=True)\n",
    "samplesheet.columns = ['GWAS Source', 'eQTL Source', 'GE Source', 'Loop Source', 'HiChIP Map', 'Analyzed?', 'File Name']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "49e05840",
   "metadata": {},
   "source": [
    "## Combining the other datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "afdc5a56",
   "metadata": {},
   "outputs": [],
   "source": [
    "# loading and concat all the data\n",
    "data = []\n",
    "good_samples = samplesheet.loc[(samplesheet['Analyzed?'] == 'True')]\n",
    "for i, sr in good_samples.iterrows(): \n",
    "    \n",
    "    df = pd.read_table(sr['File Name'], header=0)\n",
    "    df['gwas_source'] = sr['GWAS Source']\n",
    "    df['eqtl_source'] = sr['eQTL Source']\n",
    "    df['ge_source'] = sr['GE Source']\n",
    "    df['loop_source'] = sr['Loop Source']\n",
    "    \n",
    "    # combine the columns gene_id and geneid\n",
    "    #df.loc[~df['gene_id'].isna(), 'geneid'] = df.loc[~df['gene_id'].isna(), 'gene_id']\n",
    "    #df.loc[:, 'gene_name'] = df.loc[:, 'geneid'].apply(lambda x: gene_dict[x])\n",
    "    df = df.loc[(df.is_coloc_pair == 1) & (df.has_fithichip_loop == 1)]\n",
    "    \n",
    "    if df.shape[0] > 0: \n",
    "        data.append(df)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "5d506079",
   "metadata": {},
   "outputs": [],
   "source": [
    "sgl_data = pd.concat(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "9da76cd4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sid</th>\n",
       "      <th>rsid</th>\n",
       "      <th>geneid</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>tss_start</th>\n",
       "      <th>tss_end</th>\n",
       "      <th>is_eqtl_pair</th>\n",
       "      <th>is_coloc_pair</th>\n",
       "      <th>is_closest_gene</th>\n",
       "      <th>has_fithichip_loop</th>\n",
       "      <th>has_colocSNP_anchor</th>\n",
       "      <th>is_ld_snp</th>\n",
       "      <th>eqtl_pval</th>\n",
       "      <th>eqtl_beta</th>\n",
       "      <th>dist</th>\n",
       "      <th>ppH0</th>\n",
       "      <th>ppH1</th>\n",
       "      <th>ppH2</th>\n",
       "      <th>ppH3</th>\n",
       "      <th>ppH4</th>\n",
       "      <th>gene_start</th>\n",
       "      <th>gene_end</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gwas_slope</th>\n",
       "      <th>gwas_slope_se</th>\n",
       "      <th>gwas_pval</th>\n",
       "      <th>gene_start.1</th>\n",
       "      <th>gene_end.1</th>\n",
       "      <th>gene_strand</th>\n",
       "      <th>gwas_source</th>\n",
       "      <th>eqtl_source</th>\n",
       "      <th>ge_source</th>\n",
       "      <th>loop_source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2259</th>\n",
       "      <td>21:43825722</td>\n",
       "      <td>rs9981624</td>\n",
       "      <td>ENSG00000160183</td>\n",
       "      <td>TMPRSS3</td>\n",
       "      <td>chr21</td>\n",
       "      <td>44830000</td>\n",
       "      <td>43816954</td>\n",
       "      <td>43816955</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.060011</td>\n",
       "      <td>0.127783</td>\n",
       "      <td>0.812207</td>\n",
       "      <td>43791946</td>\n",
       "      <td>43816955</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1175.0</td>\n",
       "      <td>0.234625</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>0.152034</td>\n",
       "      <td>0.019004</td>\n",
       "      <td>1.244192e-15</td>\n",
       "      <td>43791946</td>\n",
       "      <td>43816955</td>\n",
       "      <td>-</td>\n",
       "      <td>T1D_25751624</td>\n",
       "      <td>GENCORD</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3538</th>\n",
       "      <td>21:43825722</td>\n",
       "      <td>rs9981624</td>\n",
       "      <td>ENSG00000160183</td>\n",
       "      <td>TMPRSS3</td>\n",
       "      <td>chr21</td>\n",
       "      <td>44830000</td>\n",
       "      <td>43816954</td>\n",
       "      <td>43816955</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.061067</td>\n",
       "      <td>0.112535</td>\n",
       "      <td>0.826398</td>\n",
       "      <td>43791946</td>\n",
       "      <td>43816955</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1175.0</td>\n",
       "      <td>0.234625</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>0.152034</td>\n",
       "      <td>0.019004</td>\n",
       "      <td>1.244192e-15</td>\n",
       "      <td>43791946</td>\n",
       "      <td>43816955</td>\n",
       "      <td>-</td>\n",
       "      <td>T1D_25751624</td>\n",
       "      <td>Schmiedel_2018</td>\n",
       "      <td>Tfh_memory</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              sid       rsid           geneid gene_name  chrom   snp_pos  \\\n",
       "2259  21:43825722  rs9981624  ENSG00000160183   TMPRSS3  chr21  44830000   \n",
       "3538  21:43825722  rs9981624  ENSG00000160183   TMPRSS3  chr21  44830000   \n",
       "\n",
       "      tss_start   tss_end  is_eqtl_pair  is_coloc_pair  is_closest_gene  \\\n",
       "2259   43816954  43816955             0              1                0   \n",
       "3538   43816954  43816955             0              1                0   \n",
       "\n",
       "      has_fithichip_loop  has_colocSNP_anchor  is_ld_snp  eqtl_pval  \\\n",
       "2259                   1                    1          0        NaN   \n",
       "3538                   1                    1          0        NaN   \n",
       "\n",
       "      eqtl_beta  dist  ppH0  ppH1      ppH2      ppH3      ppH4  gene_start  \\\n",
       "2259        NaN   NaN   0.0   0.0  0.060011  0.127783  0.812207    43791946   \n",
       "3538        NaN   NaN   0.0   0.0  0.061067  0.112535  0.826398    43791946   \n",
       "\n",
       "      gene_end  ref  alt      AC        AF      AN  gwas_slope  gwas_slope_se  \\\n",
       "2259  43816955  NaN  NaN  1175.0  0.234625  5008.0    0.152034       0.019004   \n",
       "3538  43816955  NaN  NaN  1175.0  0.234625  5008.0    0.152034       0.019004   \n",
       "\n",
       "         gwas_pval  gene_start.1  gene_end.1 gene_strand   gwas_source  \\\n",
       "2259  1.244192e-15      43791946    43816955           -  T1D_25751624   \n",
       "3538  1.244192e-15      43791946    43816955           -  T1D_25751624   \n",
       "\n",
       "         eqtl_source   ge_source       loop_source  \n",
       "2259         GENCORD      T-cell  CD4_T-cell_naive  \n",
       "3538  Schmiedel_2018  Tfh_memory  CD4_T-cell_naive  "
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sgl_data.loc[(sgl_data.gene_name == 'TMPRSS3') & (sgl_data.is_ld_snp == 0)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "8e0e245a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sid</th>\n",
       "      <th>rsid</th>\n",
       "      <th>geneid</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>tss_start</th>\n",
       "      <th>tss_end</th>\n",
       "      <th>is_eqtl_pair</th>\n",
       "      <th>is_coloc_pair</th>\n",
       "      <th>is_closest_gene</th>\n",
       "      <th>has_fithichip_loop</th>\n",
       "      <th>has_colocSNP_anchor</th>\n",
       "      <th>is_ld_snp</th>\n",
       "      <th>eqtl_pval</th>\n",
       "      <th>eqtl_beta</th>\n",
       "      <th>dist</th>\n",
       "      <th>ppH0</th>\n",
       "      <th>ppH1</th>\n",
       "      <th>ppH2</th>\n",
       "      <th>ppH3</th>\n",
       "      <th>ppH4</th>\n",
       "      <th>gene_start</th>\n",
       "      <th>gene_end</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gwas_slope</th>\n",
       "      <th>gwas_slope_se</th>\n",
       "      <th>gwas_pval</th>\n",
       "      <th>gene_start.1</th>\n",
       "      <th>gene_end.1</th>\n",
       "      <th>gene_strand</th>\n",
       "      <th>gwas_source</th>\n",
       "      <th>eqtl_source</th>\n",
       "      <th>ge_source</th>\n",
       "      <th>loop_source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [sid, rsid, geneid, gene_name, chrom, snp_pos, tss_start, tss_end, is_eqtl_pair, is_coloc_pair, is_closest_gene, has_fithichip_loop, has_colocSNP_anchor, is_ld_snp, eqtl_pval, eqtl_beta, dist, ppH0, ppH1, ppH2, ppH3, ppH4, gene_start, gene_end, ref, alt, AC, AF, AN, gwas_slope, gwas_slope_se, gwas_pval, gene_start.1, gene_end.1, gene_strand, gwas_source, eqtl_source, ge_source, loop_source]\n",
       "Index: []"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sgl_data.loc[(sgl_data.is_eqtl_pair == 1) & (sgl_data.is_ld_snp == 0)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "311818ba",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c886f64e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "17ca3598",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "1bcd628a",
   "metadata": {},
   "source": [
    "### Checking the uniq set of genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "979404fa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>geneid</th>\n",
       "      <th>genename</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000230918</th>\n",
       "      <td>3</td>\n",
       "      <td>AC008063.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000271889</th>\n",
       "      <td>1</td>\n",
       "      <td>AC016747.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000259474</th>\n",
       "      <td>1</td>\n",
       "      <td>AC027228.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000224745</th>\n",
       "      <td>1</td>\n",
       "      <td>AC063965.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000107796</th>\n",
       "      <td>1</td>\n",
       "      <td>ACTA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000228124</th>\n",
       "      <td>3</td>\n",
       "      <td>AL096678.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000226455</th>\n",
       "      <td>2</td>\n",
       "      <td>AL121787.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000260271</th>\n",
       "      <td>2</td>\n",
       "      <td>AL132996.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000273797</th>\n",
       "      <td>1</td>\n",
       "      <td>AL133445.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000229664</th>\n",
       "      <td>2</td>\n",
       "      <td>AL137186.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000231128</th>\n",
       "      <td>1</td>\n",
       "      <td>AL137856.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000213994</th>\n",
       "      <td>2</td>\n",
       "      <td>AL157395.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000259097</th>\n",
       "      <td>1</td>\n",
       "      <td>AL163932.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000273483</th>\n",
       "      <td>1</td>\n",
       "      <td>AL354760.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000272703</th>\n",
       "      <td>1</td>\n",
       "      <td>AP005137.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000150637</th>\n",
       "      <td>1</td>\n",
       "      <td>CD226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000203760</th>\n",
       "      <td>2</td>\n",
       "      <td>CENPW</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000136982</th>\n",
       "      <td>1</td>\n",
       "      <td>DSCC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000188996</th>\n",
       "      <td>1</td>\n",
       "      <td>HUS1B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000229414</th>\n",
       "      <td>1</td>\n",
       "      <td>KCNQ1-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000224034</th>\n",
       "      <td>1</td>\n",
       "      <td>LINC02561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000173239</th>\n",
       "      <td>1</td>\n",
       "      <td>LIPM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000196436</th>\n",
       "      <td>1</td>\n",
       "      <td>NPIPB15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000090104</th>\n",
       "      <td>1</td>\n",
       "      <td>RGS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000197728</th>\n",
       "      <td>3</td>\n",
       "      <td>RPS26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000254827</th>\n",
       "      <td>1</td>\n",
       "      <td>SLC22A18AS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000160183</th>\n",
       "      <td>2</td>\n",
       "      <td>TMPRSS3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000183439</th>\n",
       "      <td>1</td>\n",
       "      <td>TRIM61</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 geneid    genename\n",
       "ENSG00000230918       3  AC008063.1\n",
       "ENSG00000271889       1  AC016747.2\n",
       "ENSG00000259474       1  AC027228.1\n",
       "ENSG00000224745       1  AC063965.1\n",
       "ENSG00000107796       1       ACTA2\n",
       "ENSG00000228124       3  AL096678.1\n",
       "ENSG00000226455       2  AL121787.1\n",
       "ENSG00000260271       2  AL132996.1\n",
       "ENSG00000273797       1  AL133445.2\n",
       "ENSG00000229664       2  AL137186.1\n",
       "ENSG00000231128       1  AL137856.1\n",
       "ENSG00000213994       2  AL157395.1\n",
       "ENSG00000259097       1  AL163932.1\n",
       "ENSG00000273483       1  AL354760.1\n",
       "ENSG00000272703       1  AP005137.2\n",
       "ENSG00000150637       1       CD226\n",
       "ENSG00000203760       2       CENPW\n",
       "ENSG00000136982       1       DSCC1\n",
       "ENSG00000188996       1       HUS1B\n",
       "ENSG00000229414       1   KCNQ1-AS1\n",
       "ENSG00000224034       1   LINC02561\n",
       "ENSG00000173239       1        LIPM\n",
       "ENSG00000196436       1     NPIPB15\n",
       "ENSG00000090104       1        RGS1\n",
       "ENSG00000197728       3       RPS26\n",
       "ENSG00000254827       1  SLC22A18AS\n",
       "ENSG00000160183       2     TMPRSS3\n",
       "ENSG00000183439       1      TRIM61"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_genes = sgl_data.loc[(sgl_data.is_ld_snp == 0)].geneid.value_counts().to_frame()\n",
    "unique_gnames = [gene_dict[x] if x in gene_dict else x for x in unique_genes.index.tolist()]\n",
    "unique_genes.loc[:, 'genename'] = unique_gnames\n",
    "unique_genes.sort_values('genename', inplace=True)\n",
    "unique_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "f0e21823",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(28, 2)"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_genes.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "a1c656d8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>geneid</th>\n",
       "      <th>genename</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000230918</th>\n",
       "      <td>9</td>\n",
       "      <td>AC008063.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000271889</th>\n",
       "      <td>77</td>\n",
       "      <td>AC016747.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000259474</th>\n",
       "      <td>28</td>\n",
       "      <td>AC027228.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000224745</th>\n",
       "      <td>20</td>\n",
       "      <td>AC063965.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000107796</th>\n",
       "      <td>20</td>\n",
       "      <td>ACTA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000228124</th>\n",
       "      <td>107</td>\n",
       "      <td>AL096678.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000226455</th>\n",
       "      <td>64</td>\n",
       "      <td>AL121787.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000260271</th>\n",
       "      <td>89</td>\n",
       "      <td>AL132996.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000273797</th>\n",
       "      <td>32</td>\n",
       "      <td>AL133445.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000229664</th>\n",
       "      <td>7</td>\n",
       "      <td>AL137186.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000231128</th>\n",
       "      <td>4</td>\n",
       "      <td>AL137856.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000213994</th>\n",
       "      <td>8</td>\n",
       "      <td>AL157395.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000259097</th>\n",
       "      <td>62</td>\n",
       "      <td>AL163932.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000273483</th>\n",
       "      <td>4</td>\n",
       "      <td>AL354760.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000272703</th>\n",
       "      <td>34</td>\n",
       "      <td>AP005137.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000150637</th>\n",
       "      <td>49</td>\n",
       "      <td>CD226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000203760</th>\n",
       "      <td>489</td>\n",
       "      <td>CENPW</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000136982</th>\n",
       "      <td>27</td>\n",
       "      <td>DSCC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000188996</th>\n",
       "      <td>8</td>\n",
       "      <td>HUS1B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000229414</th>\n",
       "      <td>14</td>\n",
       "      <td>KCNQ1-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000224034</th>\n",
       "      <td>4</td>\n",
       "      <td>LINC02561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000173239</th>\n",
       "      <td>1</td>\n",
       "      <td>LIPM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000196436</th>\n",
       "      <td>17</td>\n",
       "      <td>NPIPB15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000090104</th>\n",
       "      <td>19</td>\n",
       "      <td>RGS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000197728</th>\n",
       "      <td>108</td>\n",
       "      <td>RPS26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000254827</th>\n",
       "      <td>14</td>\n",
       "      <td>SLC22A18AS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000160183</th>\n",
       "      <td>46</td>\n",
       "      <td>TMPRSS3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000183439</th>\n",
       "      <td>83</td>\n",
       "      <td>TRIM61</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 geneid    genename\n",
       "ENSG00000230918       9  AC008063.1\n",
       "ENSG00000271889      77  AC016747.2\n",
       "ENSG00000259474      28  AC027228.1\n",
       "ENSG00000224745      20  AC063965.1\n",
       "ENSG00000107796      20       ACTA2\n",
       "ENSG00000228124     107  AL096678.1\n",
       "ENSG00000226455      64  AL121787.1\n",
       "ENSG00000260271      89  AL132996.1\n",
       "ENSG00000273797      32  AL133445.2\n",
       "ENSG00000229664       7  AL137186.1\n",
       "ENSG00000231128       4  AL137856.1\n",
       "ENSG00000213994       8  AL157395.1\n",
       "ENSG00000259097      62  AL163932.1\n",
       "ENSG00000273483       4  AL354760.1\n",
       "ENSG00000272703      34  AP005137.2\n",
       "ENSG00000150637      49       CD226\n",
       "ENSG00000203760     489       CENPW\n",
       "ENSG00000136982      27       DSCC1\n",
       "ENSG00000188996       8       HUS1B\n",
       "ENSG00000229414      14   KCNQ1-AS1\n",
       "ENSG00000224034       4   LINC02561\n",
       "ENSG00000173239       1        LIPM\n",
       "ENSG00000196436      17     NPIPB15\n",
       "ENSG00000090104      19        RGS1\n",
       "ENSG00000197728     108       RPS26\n",
       "ENSG00000254827      14  SLC22A18AS\n",
       "ENSG00000160183      46     TMPRSS3\n",
       "ENSG00000183439      83      TRIM61"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_genes = sgl_data.loc[(sgl_data.is_ld_snp == 1)].geneid.value_counts().to_frame()\n",
    "unique_gnames = [gene_dict[x] if x in gene_dict else x for x in unique_genes.index.tolist()]\n",
    "unique_genes.loc[:, 'genename'] = unique_gnames\n",
    "unique_genes.sort_values('genename', inplace=True)\n",
    "unique_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "cbd9a92a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(28, 2)"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_genes.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "245dac97",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0935e2b9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ece137f3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "403e2309",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "749a39fb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9be346e5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "43671d0e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "80eb0c5d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "169ab3d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# find the unique SNPs\n",
    "uniq_snps_by_cells = sgl_data.groupby('ge_source').sid.nunique()\n",
    "uniq_snps_by_cells = uniq_snps_by_cells.to_frame()\n",
    "\n",
    "# find the unique genes \n",
    "uniq_genes_by_cells = sgl_data.groupby('ge_source').geneid.nunique()\n",
    "uniq_genes_by_cells = uniq_genes_by_cells.to_frame()\n",
    "\n",
    "# merge snps and genes\n",
    "uniq_counts_by_cells = pd.merge(uniq_snps_by_cells, uniq_genes_by_cells, left_index=True, right_index=True)\n",
    "uniq_counts_by_cells.columns = ['Number of Unique SNPs', 'Number of Unique Genes']\n",
    "uniq_counts_by_cells.index.name = 'Cell Line'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c7067546",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save the file\n",
    "excel_analysis = os.path.join(outdir, 'Unique_Counts_By_Cell_Line.xlsx')\n",
    "uniq_counts_by_cells.to_excel(excel_analysis, sheet_name='pieqtls')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "975c6e63",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>geneid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ge_source</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NK-cell_naive</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T-cell</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tfh_memory</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Th1-17_memory</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Th17_memory</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Th1_memory</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Th2_memory</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Treg_memory</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Treg_naive</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>monocyte</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>monocyte_CD16_naive</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>monocyte_IAV</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>monocyte_Pam3CSK4</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>monocyte_R848</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     geneid\n",
       "ge_source                  \n",
       "NK-cell_naive             1\n",
       "T-cell                    5\n",
       "Tfh_memory                7\n",
       "Th1-17_memory             2\n",
       "Th17_memory               5\n",
       "Th1_memory                3\n",
       "Th2_memory                2\n",
       "Treg_memory               1\n",
       "Treg_naive                5\n",
       "monocyte                  3\n",
       "monocyte_CD16_naive       1\n",
       "monocyte_IAV              2\n",
       "monocyte_Pam3CSK4         1\n",
       "monocyte_R848             1"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uniq_genes_by_cells"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "71a378db",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'results/main/sgls/combined/Unique_Counts_By_Cell_Line.xlsx'"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "excel_analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "49166bd3",
   "metadata": {},
   "source": [
    "#### Write the gene list as well"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "39ab2750",
   "metadata": {},
   "outputs": [],
   "source": [
    "gh_list = sorted(unique_genes_df.geneid.unique().tolist())\n",
    "gh_fn = os.path.join(outdir, 'gene_list.txt')\n",
    "with open(gh_fn, 'w') as fw:\n",
    "    for x in gh_list:\n",
    "        fw.write('{}\\n'.format(x))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e0d245c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'results/main/sgls/combined/gene_list.txt'"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gh_fn"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "19fe9be8",
   "metadata": {},
   "source": [
    "### Analyzing the Number of eQTLs, loops, colocaled SNP-Gene (per DataSet)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "22fd9194",
   "metadata": {},
   "outputs": [],
   "source": [
    "cells_with_loops = sgl_data[sgl_data.has_fithichip_loop == 1]\n",
    "uniq_cells = cells_with_loops[['eqtl_source', 'ge_source', 'loop_source']].values.tolist()\n",
    "uniq_cells = set([tuple(x) for x in uniq_cells])\n",
    "uniq_cells = list(uniq_cells)\n",
    "uniq_cells = pd.DataFrame(uniq_cells)\n",
    "uniq_cells.columns = ['eqtl_source', 'ge_source', 'loop_source']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "a7adb581",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>eqtl_source</th>\n",
       "      <th>ge_source</th>\n",
       "      <th>loop_source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Schmiedel_2018</td>\n",
       "      <td>NK-cell_naive</td>\n",
       "      <td>NK-cell_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GENCORD</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>BLUEPRINT</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Schmiedel_2018</td>\n",
       "      <td>Tfh_memory</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Schmiedel_2018</td>\n",
       "      <td>Th1-17_memory</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Schmiedel_2018</td>\n",
       "      <td>Th17_memory</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Schmiedel_2018</td>\n",
       "      <td>Th1_memory</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Schmiedel_2018</td>\n",
       "      <td>Th2_memory</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Schmiedel_2018</td>\n",
       "      <td>Treg_memory</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Schmiedel_2018</td>\n",
       "      <td>Treg_naive</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>BLUEPRINT</td>\n",
       "      <td>monocyte</td>\n",
       "      <td>monocyte_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Schmiedel_2018</td>\n",
       "      <td>monocyte_CD16_naive</td>\n",
       "      <td>monocyte_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Quach_2016</td>\n",
       "      <td>monocyte_IAV</td>\n",
       "      <td>monocyte_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Quach_2016</td>\n",
       "      <td>monocyte_Pam3CSK4</td>\n",
       "      <td>monocyte_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Quach_2016</td>\n",
       "      <td>monocyte_R848</td>\n",
       "      <td>monocyte_naive</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       eqtl_source            ge_source       loop_source\n",
       "6   Schmiedel_2018        NK-cell_naive     NK-cell_naive\n",
       "0          GENCORD               T-cell  CD4_T-cell_naive\n",
       "5        BLUEPRINT               T-cell  CD4_T-cell_naive\n",
       "11  Schmiedel_2018           Tfh_memory  CD4_T-cell_naive\n",
       "2   Schmiedel_2018        Th1-17_memory  CD4_T-cell_naive\n",
       "10  Schmiedel_2018          Th17_memory  CD4_T-cell_naive\n",
       "3   Schmiedel_2018           Th1_memory  CD4_T-cell_naive\n",
       "14  Schmiedel_2018           Th2_memory  CD4_T-cell_naive\n",
       "12  Schmiedel_2018          Treg_memory  CD4_T-cell_naive\n",
       "8   Schmiedel_2018           Treg_naive  CD4_T-cell_naive\n",
       "13       BLUEPRINT             monocyte    monocyte_naive\n",
       "7   Schmiedel_2018  monocyte_CD16_naive    monocyte_naive\n",
       "9       Quach_2016         monocyte_IAV    monocyte_naive\n",
       "1       Quach_2016    monocyte_Pam3CSK4    monocyte_naive\n",
       "4       Quach_2016        monocyte_R848    monocyte_naive"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uniq_cells.sort_values('ge_source')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "3f7a4d64",
   "metadata": {},
   "outputs": [],
   "source": [
    "# group by the dataset combos\n",
    "eqtl_ge_grps = sgl_data.groupby(['eqtl_source', 'ge_source', 'loop_source'])\n",
    "\n",
    "# calculate the number of sg pairs \n",
    "def count_uniq(x):\n",
    "    v = x[['sid', 'geneid']].values.tolist()\n",
    "    v = [tuple(x) for x in v]\n",
    "    v = set(v)\n",
    "    return(len(v))\n",
    "eqtl_ge_sg_pairs = eqtl_ge_grps.apply(count_uniq).to_frame()\n",
    "eqtl_ge_sg_pairs.columns = ['num_sg_pairs']\n",
    "\n",
    "# calculate the number of sg pairs with loops\n",
    "eqtl_ge_loops = eqtl_ge_grps.has_fithichip_loop.sum().to_frame()\n",
    "\n",
    "# calculate the number of sg pairs with significant eQTL signal\n",
    "eqtl_ge_eqtl = eqtl_ge_grps.is_eqtl_pair.sum().to_frame()\n",
    "\n",
    "# calculate the number of sg pairs with significant coloc \n",
    "eqtl_ge_coloc = eqtl_ge_grps.is_coloc_pair.sum().to_frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "a8b61d4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "eqtl_ge_master = eqtl_ge_sg_pairs.merge(eqtl_ge_coloc, left_index=True, right_index=True)\n",
    "eqtl_ge_master = eqtl_ge_master.merge(eqtl_ge_eqtl, left_index=True, right_index=True)\n",
    "eqtl_ge_master = eqtl_ge_master.merge(eqtl_ge_loops, left_index=True, right_index=True)\n",
    "\n",
    "eqtl_ge_master.columns = ['Number of\\\\nSNP-Gene Pairs',\n",
    "                          'Number of\\\\nColoc Pairs',\n",
    "                          'Number of\\\\neQTL Pairs',\n",
    "                          'Number of\\\\nPairs with a Loop']\n",
    "eqtl_ge_master.index.names = ['eQTL Source', 'GE Source', 'Loop Source']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "3f0ed392",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Number of<br>SNP-Gene Pairs</th>\n",
       "      <th>Number of<br>Coloc Pairs</th>\n",
       "      <th>Number of<br>eQTL Pairs</th>\n",
       "      <th>Number of<br>Pairs with a Loop</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>eQTL Source</th>\n",
       "      <th>GE Source</th>\n",
       "      <th>Loop Source</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">BLUEPRINT</th>\n",
       "      <th>T-cell</th>\n",
       "      <th>CD4_T-cell_naive</th>\n",
       "      <td>131</td>\n",
       "      <td>142</td>\n",
       "      <td>0</td>\n",
       "      <td>142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>monocyte</th>\n",
       "      <th>monocyte_naive</th>\n",
       "      <td>102</td>\n",
       "      <td>102</td>\n",
       "      <td>0</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GENCORD</th>\n",
       "      <th>T-cell</th>\n",
       "      <th>CD4_T-cell_naive</th>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>0</td>\n",
       "      <td>110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">Quach_2016</th>\n",
       "      <th>monocyte_IAV</th>\n",
       "      <th>monocyte_naive</th>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "      <td>0</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>monocyte_Pam3CSK4</th>\n",
       "      <th>monocyte_naive</th>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>monocyte_R848</th>\n",
       "      <th>monocyte_naive</th>\n",
       "      <td>245</td>\n",
       "      <td>245</td>\n",
       "      <td>0</td>\n",
       "      <td>245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"9\" valign=\"top\">Schmiedel_2018</th>\n",
       "      <th>NK-cell_naive</th>\n",
       "      <th>NK-cell_naive</th>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "      <td>0</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tfh_memory</th>\n",
       "      <th>CD4_T-cell_naive</th>\n",
       "      <td>369</td>\n",
       "      <td>399</td>\n",
       "      <td>0</td>\n",
       "      <td>399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Th1-17_memory</th>\n",
       "      <th>CD4_T-cell_naive</th>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "      <td>0</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Th17_memory</th>\n",
       "      <th>CD4_T-cell_naive</th>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>0</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Th1_memory</th>\n",
       "      <th>CD4_T-cell_naive</th>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "      <td>0</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Th2_memory</th>\n",
       "      <th>CD4_T-cell_naive</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Treg_memory</th>\n",
       "      <th>CD4_T-cell_naive</th>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Treg_naive</th>\n",
       "      <th>CD4_T-cell_naive</th>\n",
       "      <td>171</td>\n",
       "      <td>171</td>\n",
       "      <td>0</td>\n",
       "      <td>171</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>monocyte_CD16_naive</th>\n",
       "      <th>monocyte_naive</th>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "HTML(eqtl_ge_master.to_html().replace('\\\\n', '<br>'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "759269d9",
   "metadata": {},
   "source": [
    "#### Extract all pairs with a significant eQTL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "5cbc1c8c",
   "metadata": {},
   "outputs": [],
   "source": [
    "sg_with_eqtl = sgl_data.loc[(sgl_data.is_eqtl_pair == 1), major_cols]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "57cc5b6f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 0 SNP-Gene pairs with an eQTL.\n"
     ]
    }
   ],
   "source": [
    "print('There are {} SNP-Gene pairs with an eQTL.'.format(len(sg_with_eqtl)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "c515cfae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gwas_source</th>\n",
       "      <th>ge_source</th>\n",
       "      <th>loop_source</th>\n",
       "      <th>sid</th>\n",
       "      <th>rsid</th>\n",
       "      <th>geneid</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>tss_start</th>\n",
       "      <th>tss_end</th>\n",
       "      <th>is_eqtl_pair</th>\n",
       "      <th>is_coloc_pair</th>\n",
       "      <th>is_closest_gene</th>\n",
       "      <th>has_fithichip_loop</th>\n",
       "      <th>eqtl_pval</th>\n",
       "      <th>dist</th>\n",
       "      <th>ppH4</th>\n",
       "      <th>gene_start</th>\n",
       "      <th>gene_end</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gwas_pval</th>\n",
       "      <th>gene_strand</th>\n",
       "      <th>eqtl_source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [gwas_source, ge_source, loop_source, sid, rsid, geneid, gene_name, chrom, snp_pos, tss_start, tss_end, is_eqtl_pair, is_coloc_pair, is_closest_gene, has_fithichip_loop, eqtl_pval, dist, ppH4, gene_start, gene_end, ref, alt, AC, AF, AN, gwas_pval, gene_strand, eqtl_source]\n",
       "Index: []"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sg_with_eqtl"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "10a08a57",
   "metadata": {},
   "source": [
    "#### Extract all pairs with a colocalization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "c048455f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sid</th>\n",
       "      <th>rsid</th>\n",
       "      <th>geneid</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>tss_start</th>\n",
       "      <th>tss_end</th>\n",
       "      <th>is_eqtl_pair</th>\n",
       "      <th>is_coloc_pair</th>\n",
       "      <th>is_closest_gene</th>\n",
       "      <th>has_fithichip_loop</th>\n",
       "      <th>has_colocSNP_anchor</th>\n",
       "      <th>is_ld_snp</th>\n",
       "      <th>eqtl_pval</th>\n",
       "      <th>eqtl_beta</th>\n",
       "      <th>dist</th>\n",
       "      <th>ppH0</th>\n",
       "      <th>ppH1</th>\n",
       "      <th>ppH2</th>\n",
       "      <th>ppH3</th>\n",
       "      <th>ppH4</th>\n",
       "      <th>gene_start</th>\n",
       "      <th>gene_end</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gwas_slope</th>\n",
       "      <th>gwas_slope_se</th>\n",
       "      <th>gwas_pval</th>\n",
       "      <th>gene_start.1</th>\n",
       "      <th>gene_end.1</th>\n",
       "      <th>gene_strand</th>\n",
       "      <th>gwas_source</th>\n",
       "      <th>eqtl_source</th>\n",
       "      <th>ge_source</th>\n",
       "      <th>loop_source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>6:90936894</td>\n",
       "      <td>rs206913</td>\n",
       "      <td>ENSG00000260271</td>\n",
       "      <td>AL132996.1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>91940000</td>\n",
       "      <td>91030161</td>\n",
       "      <td>91030162</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.097200e-145</td>\n",
       "      <td>2.458868e-144</td>\n",
       "      <td>7.771106e-03</td>\n",
       "      <td>0.173335</td>\n",
       "      <td>0.818894</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3488.0</td>\n",
       "      <td>0.696486</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>-0.129995</td>\n",
       "      <td>0.018999</td>\n",
       "      <td>7.803700e-12</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>+</td>\n",
       "      <td>T1D_25751624</td>\n",
       "      <td>BLUEPRINT</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>6:90939178</td>\n",
       "      <td>rs58516003</td>\n",
       "      <td>ENSG00000260271</td>\n",
       "      <td>AL132996.1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>91940000</td>\n",
       "      <td>91030161</td>\n",
       "      <td>91030162</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.097200e-145</td>\n",
       "      <td>2.458868e-144</td>\n",
       "      <td>7.771106e-03</td>\n",
       "      <td>0.173335</td>\n",
       "      <td>0.818894</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3488.0</td>\n",
       "      <td>0.696486</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>-0.129995</td>\n",
       "      <td>0.018999</td>\n",
       "      <td>7.803700e-12</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>+</td>\n",
       "      <td>T1D_25751624</td>\n",
       "      <td>BLUEPRINT</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>6:90950736</td>\n",
       "      <td>rs2256672</td>\n",
       "      <td>ENSG00000260271</td>\n",
       "      <td>AL132996.1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>91955000</td>\n",
       "      <td>91030161</td>\n",
       "      <td>91030162</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.097200e-145</td>\n",
       "      <td>2.458868e-144</td>\n",
       "      <td>7.771106e-03</td>\n",
       "      <td>0.173335</td>\n",
       "      <td>0.818894</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3488.0</td>\n",
       "      <td>0.696486</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>-0.129995</td>\n",
       "      <td>0.018999</td>\n",
       "      <td>7.803700e-12</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>+</td>\n",
       "      <td>T1D_25751624</td>\n",
       "      <td>BLUEPRINT</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>6:90960329</td>\n",
       "      <td>rs673780</td>\n",
       "      <td>ENSG00000260271</td>\n",
       "      <td>AL132996.1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>91965000</td>\n",
       "      <td>91030161</td>\n",
       "      <td>91030162</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.097200e-145</td>\n",
       "      <td>2.458868e-144</td>\n",
       "      <td>7.771106e-03</td>\n",
       "      <td>0.173335</td>\n",
       "      <td>0.818894</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3488.0</td>\n",
       "      <td>0.696486</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>-0.129995</td>\n",
       "      <td>0.018999</td>\n",
       "      <td>7.803700e-12</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>+</td>\n",
       "      <td>T1D_25751624</td>\n",
       "      <td>BLUEPRINT</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>6:90968949</td>\n",
       "      <td>rs619192</td>\n",
       "      <td>ENSG00000260271</td>\n",
       "      <td>AL132996.1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>91970000</td>\n",
       "      <td>91030161</td>\n",
       "      <td>91030162</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.097200e-145</td>\n",
       "      <td>2.458868e-144</td>\n",
       "      <td>7.771106e-03</td>\n",
       "      <td>0.173335</td>\n",
       "      <td>0.818894</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3488.0</td>\n",
       "      <td>0.696486</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>-0.129995</td>\n",
       "      <td>0.018999</td>\n",
       "      <td>7.803700e-12</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>+</td>\n",
       "      <td>T1D_25751624</td>\n",
       "      <td>BLUEPRINT</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>749</th>\n",
       "      <td>11:2182224</td>\n",
       "      <td>rs689</td>\n",
       "      <td>ENSG00000229414</td>\n",
       "      <td>KCNQ1-AS1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>3185000</td>\n",
       "      <td>2882797</td>\n",
       "      <td>2882798</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.138070e-23</td>\n",
       "      <td>0.060991</td>\n",
       "      <td>0.939009</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3253.0</td>\n",
       "      <td>0.649561</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>0.308500</td>\n",
       "      <td>0.022200</td>\n",
       "      <td>9.360000e-44</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>-</td>\n",
       "      <td>T1D_34594039_GCST90018925</td>\n",
       "      <td>Schmiedel_2018</td>\n",
       "      <td>monocyte_CD16_naive</td>\n",
       "      <td>monocyte_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1118</th>\n",
       "      <td>11:2187855</td>\n",
       "      <td>rs12419447</td>\n",
       "      <td>ENSG00000229414</td>\n",
       "      <td>KCNQ1-AS1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>3190000</td>\n",
       "      <td>2882797</td>\n",
       "      <td>2882798</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.138070e-23</td>\n",
       "      <td>0.060991</td>\n",
       "      <td>0.939009</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3253.0</td>\n",
       "      <td>0.649561</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>0.308500</td>\n",
       "      <td>0.022200</td>\n",
       "      <td>9.360000e-44</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>-</td>\n",
       "      <td>T1D_34594039_GCST90018925</td>\n",
       "      <td>Schmiedel_2018</td>\n",
       "      <td>monocyte_CD16_naive</td>\n",
       "      <td>monocyte_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1119</th>\n",
       "      <td>11:2189983</td>\n",
       "      <td>rs4930043</td>\n",
       "      <td>ENSG00000229414</td>\n",
       "      <td>KCNQ1-AS1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>3190000</td>\n",
       "      <td>2882797</td>\n",
       "      <td>2882798</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.138070e-23</td>\n",
       "      <td>0.060991</td>\n",
       "      <td>0.939009</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3253.0</td>\n",
       "      <td>0.649561</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>0.308500</td>\n",
       "      <td>0.022200</td>\n",
       "      <td>9.360000e-44</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>-</td>\n",
       "      <td>T1D_34594039_GCST90018925</td>\n",
       "      <td>Schmiedel_2018</td>\n",
       "      <td>monocyte_CD16_naive</td>\n",
       "      <td>monocyte_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1120</th>\n",
       "      <td>11:2188238</td>\n",
       "      <td>rs6357</td>\n",
       "      <td>ENSG00000229414</td>\n",
       "      <td>KCNQ1-AS1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>3190000</td>\n",
       "      <td>2882797</td>\n",
       "      <td>2882798</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.138070e-23</td>\n",
       "      <td>0.060991</td>\n",
       "      <td>0.939009</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3253.0</td>\n",
       "      <td>0.649561</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>0.308500</td>\n",
       "      <td>0.022200</td>\n",
       "      <td>9.360000e-44</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>-</td>\n",
       "      <td>T1D_34594039_GCST90018925</td>\n",
       "      <td>Schmiedel_2018</td>\n",
       "      <td>monocyte_CD16_naive</td>\n",
       "      <td>monocyte_naive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1121</th>\n",
       "      <td>11:2188436</td>\n",
       "      <td>rs67482701</td>\n",
       "      <td>ENSG00000229414</td>\n",
       "      <td>KCNQ1-AS1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>3190000</td>\n",
       "      <td>2882797</td>\n",
       "      <td>2882798</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.138070e-23</td>\n",
       "      <td>0.060991</td>\n",
       "      <td>0.939009</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3253.0</td>\n",
       "      <td>0.649561</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>0.308500</td>\n",
       "      <td>0.022200</td>\n",
       "      <td>9.360000e-44</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>-</td>\n",
       "      <td>T1D_34594039_GCST90018925</td>\n",
       "      <td>Schmiedel_2018</td>\n",
       "      <td>monocyte_CD16_naive</td>\n",
       "      <td>monocyte_naive</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1525 rows  39 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             sid        rsid           geneid   gene_name  chrom   snp_pos  \\\n",
       "52    6:90936894    rs206913  ENSG00000260271  AL132996.1   chr6  91940000   \n",
       "53    6:90939178  rs58516003  ENSG00000260271  AL132996.1   chr6  91940000   \n",
       "83    6:90950736   rs2256672  ENSG00000260271  AL132996.1   chr6  91955000   \n",
       "111   6:90960329    rs673780  ENSG00000260271  AL132996.1   chr6  91965000   \n",
       "139   6:90968949    rs619192  ENSG00000260271  AL132996.1   chr6  91970000   \n",
       "...          ...         ...              ...         ...    ...       ...   \n",
       "749   11:2182224       rs689  ENSG00000229414   KCNQ1-AS1  chr11   3185000   \n",
       "1118  11:2187855  rs12419447  ENSG00000229414   KCNQ1-AS1  chr11   3190000   \n",
       "1119  11:2189983   rs4930043  ENSG00000229414   KCNQ1-AS1  chr11   3190000   \n",
       "1120  11:2188238      rs6357  ENSG00000229414   KCNQ1-AS1  chr11   3190000   \n",
       "1121  11:2188436  rs67482701  ENSG00000229414   KCNQ1-AS1  chr11   3190000   \n",
       "\n",
       "      tss_start   tss_end  is_eqtl_pair  is_coloc_pair  is_closest_gene  \\\n",
       "52     91030161  91030162             0              1                0   \n",
       "53     91030161  91030162             0              1                0   \n",
       "83     91030161  91030162             0              1                0   \n",
       "111    91030161  91030162             0              1                0   \n",
       "139    91030161  91030162             0              1                0   \n",
       "...         ...       ...           ...            ...              ...   \n",
       "749     2882797   2882798             0              1                0   \n",
       "1118    2882797   2882798             0              1                0   \n",
       "1119    2882797   2882798             0              1                0   \n",
       "1120    2882797   2882798             0              1                0   \n",
       "1121    2882797   2882798             0              1                0   \n",
       "\n",
       "      has_fithichip_loop  has_colocSNP_anchor  is_ld_snp  eqtl_pval  \\\n",
       "52                     1                    1          1        NaN   \n",
       "53                     1                    1          1        NaN   \n",
       "83                     1                    1          1        NaN   \n",
       "111                    1                    1          1        NaN   \n",
       "139                    1                    1          1        NaN   \n",
       "...                  ...                  ...        ...        ...   \n",
       "749                    1                    1          0        NaN   \n",
       "1118                   1                    1          1        NaN   \n",
       "1119                   1                    1          1        NaN   \n",
       "1120                   1                    1          1        NaN   \n",
       "1121                   1                    1          1        NaN   \n",
       "\n",
       "      eqtl_beta  dist           ppH0           ppH1          ppH2      ppH3  \\\n",
       "52          NaN   NaN  1.097200e-145  2.458868e-144  7.771106e-03  0.173335   \n",
       "53          NaN   NaN  1.097200e-145  2.458868e-144  7.771106e-03  0.173335   \n",
       "83          NaN   NaN  1.097200e-145  2.458868e-144  7.771106e-03  0.173335   \n",
       "111         NaN   NaN  1.097200e-145  2.458868e-144  7.771106e-03  0.173335   \n",
       "139         NaN   NaN  1.097200e-145  2.458868e-144  7.771106e-03  0.173335   \n",
       "...         ...   ...            ...            ...           ...       ...   \n",
       "749         NaN   NaN   0.000000e+00   0.000000e+00  2.138070e-23  0.060991   \n",
       "1118        NaN   NaN   0.000000e+00   0.000000e+00  2.138070e-23  0.060991   \n",
       "1119        NaN   NaN   0.000000e+00   0.000000e+00  2.138070e-23  0.060991   \n",
       "1120        NaN   NaN   0.000000e+00   0.000000e+00  2.138070e-23  0.060991   \n",
       "1121        NaN   NaN   0.000000e+00   0.000000e+00  2.138070e-23  0.060991   \n",
       "\n",
       "          ppH4  gene_start  gene_end  ref  alt      AC        AF      AN  \\\n",
       "52    0.818894    91030162  91072351  NaN  NaN  3488.0  0.696486  5008.0   \n",
       "53    0.818894    91030162  91072351  NaN  NaN  3488.0  0.696486  5008.0   \n",
       "83    0.818894    91030162  91072351  NaN  NaN  3488.0  0.696486  5008.0   \n",
       "111   0.818894    91030162  91072351  NaN  NaN  3488.0  0.696486  5008.0   \n",
       "139   0.818894    91030162  91072351  NaN  NaN  3488.0  0.696486  5008.0   \n",
       "...        ...         ...       ...  ...  ...     ...       ...     ...   \n",
       "749   0.939009     2861365   2882798  NaN  NaN  3253.0  0.649561  5008.0   \n",
       "1118  0.939009     2861365   2882798  NaN  NaN  3253.0  0.649561  5008.0   \n",
       "1119  0.939009     2861365   2882798  NaN  NaN  3253.0  0.649561  5008.0   \n",
       "1120  0.939009     2861365   2882798  NaN  NaN  3253.0  0.649561  5008.0   \n",
       "1121  0.939009     2861365   2882798  NaN  NaN  3253.0  0.649561  5008.0   \n",
       "\n",
       "      gwas_slope  gwas_slope_se     gwas_pval  gene_start.1  gene_end.1  \\\n",
       "52     -0.129995       0.018999  7.803700e-12      91030162    91072351   \n",
       "53     -0.129995       0.018999  7.803700e-12      91030162    91072351   \n",
       "83     -0.129995       0.018999  7.803700e-12      91030162    91072351   \n",
       "111    -0.129995       0.018999  7.803700e-12      91030162    91072351   \n",
       "139    -0.129995       0.018999  7.803700e-12      91030162    91072351   \n",
       "...          ...            ...           ...           ...         ...   \n",
       "749     0.308500       0.022200  9.360000e-44       2861365     2882798   \n",
       "1118    0.308500       0.022200  9.360000e-44       2861365     2882798   \n",
       "1119    0.308500       0.022200  9.360000e-44       2861365     2882798   \n",
       "1120    0.308500       0.022200  9.360000e-44       2861365     2882798   \n",
       "1121    0.308500       0.022200  9.360000e-44       2861365     2882798   \n",
       "\n",
       "     gene_strand                gwas_source     eqtl_source  \\\n",
       "52             +               T1D_25751624       BLUEPRINT   \n",
       "53             +               T1D_25751624       BLUEPRINT   \n",
       "83             +               T1D_25751624       BLUEPRINT   \n",
       "111            +               T1D_25751624       BLUEPRINT   \n",
       "139            +               T1D_25751624       BLUEPRINT   \n",
       "...          ...                        ...             ...   \n",
       "749            -  T1D_34594039_GCST90018925  Schmiedel_2018   \n",
       "1118           -  T1D_34594039_GCST90018925  Schmiedel_2018   \n",
       "1119           -  T1D_34594039_GCST90018925  Schmiedel_2018   \n",
       "1120           -  T1D_34594039_GCST90018925  Schmiedel_2018   \n",
       "1121           -  T1D_34594039_GCST90018925  Schmiedel_2018   \n",
       "\n",
       "                ge_source       loop_source  \n",
       "52                 T-cell  CD4_T-cell_naive  \n",
       "53                 T-cell  CD4_T-cell_naive  \n",
       "83                 T-cell  CD4_T-cell_naive  \n",
       "111                T-cell  CD4_T-cell_naive  \n",
       "139                T-cell  CD4_T-cell_naive  \n",
       "...                   ...               ...  \n",
       "749   monocyte_CD16_naive    monocyte_naive  \n",
       "1118  monocyte_CD16_naive    monocyte_naive  \n",
       "1119  monocyte_CD16_naive    monocyte_naive  \n",
       "1120  monocyte_CD16_naive    monocyte_naive  \n",
       "1121  monocyte_CD16_naive    monocyte_naive  \n",
       "\n",
       "[1525 rows x 39 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sgl_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "c551040a",
   "metadata": {},
   "outputs": [],
   "source": [
    "sg_with_coloc = sgl_data.loc[(sgl_data.is_coloc_pair == 1), major_cols]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "a9a43239",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 1525 SNP-Gene pairs with a coloc.\n"
     ]
    }
   ],
   "source": [
    "print('There are {} SNP-Gene pairs with a coloc.'.format(len(sg_with_coloc)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "3af25581",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gwas_source</th>\n",
       "      <th>ge_source</th>\n",
       "      <th>loop_source</th>\n",
       "      <th>sid</th>\n",
       "      <th>rsid</th>\n",
       "      <th>geneid</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>tss_start</th>\n",
       "      <th>tss_end</th>\n",
       "      <th>is_eqtl_pair</th>\n",
       "      <th>is_coloc_pair</th>\n",
       "      <th>is_closest_gene</th>\n",
       "      <th>has_fithichip_loop</th>\n",
       "      <th>eqtl_pval</th>\n",
       "      <th>dist</th>\n",
       "      <th>ppH4</th>\n",
       "      <th>gene_start</th>\n",
       "      <th>gene_end</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gwas_pval</th>\n",
       "      <th>gene_strand</th>\n",
       "      <th>eqtl_source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>T1D_25751624</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "      <td>6:90936894</td>\n",
       "      <td>rs206913</td>\n",
       "      <td>ENSG00000260271</td>\n",
       "      <td>AL132996.1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>91940000</td>\n",
       "      <td>91030161</td>\n",
       "      <td>91030162</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.818894</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3488.0</td>\n",
       "      <td>0.696486</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>7.803700e-12</td>\n",
       "      <td>+</td>\n",
       "      <td>BLUEPRINT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>T1D_25751624</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "      <td>6:90939178</td>\n",
       "      <td>rs58516003</td>\n",
       "      <td>ENSG00000260271</td>\n",
       "      <td>AL132996.1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>91940000</td>\n",
       "      <td>91030161</td>\n",
       "      <td>91030162</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.818894</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3488.0</td>\n",
       "      <td>0.696486</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>7.803700e-12</td>\n",
       "      <td>+</td>\n",
       "      <td>BLUEPRINT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>T1D_25751624</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "      <td>6:90950736</td>\n",
       "      <td>rs2256672</td>\n",
       "      <td>ENSG00000260271</td>\n",
       "      <td>AL132996.1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>91955000</td>\n",
       "      <td>91030161</td>\n",
       "      <td>91030162</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.818894</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3488.0</td>\n",
       "      <td>0.696486</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>7.803700e-12</td>\n",
       "      <td>+</td>\n",
       "      <td>BLUEPRINT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>T1D_25751624</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "      <td>6:90960329</td>\n",
       "      <td>rs673780</td>\n",
       "      <td>ENSG00000260271</td>\n",
       "      <td>AL132996.1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>91965000</td>\n",
       "      <td>91030161</td>\n",
       "      <td>91030162</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.818894</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3488.0</td>\n",
       "      <td>0.696486</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>7.803700e-12</td>\n",
       "      <td>+</td>\n",
       "      <td>BLUEPRINT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>T1D_25751624</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "      <td>6:90968949</td>\n",
       "      <td>rs619192</td>\n",
       "      <td>ENSG00000260271</td>\n",
       "      <td>AL132996.1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>91970000</td>\n",
       "      <td>91030161</td>\n",
       "      <td>91030162</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.818894</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3488.0</td>\n",
       "      <td>0.696486</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>7.803700e-12</td>\n",
       "      <td>+</td>\n",
       "      <td>BLUEPRINT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>749</th>\n",
       "      <td>T1D_34594039_GCST90018925</td>\n",
       "      <td>monocyte_CD16_naive</td>\n",
       "      <td>monocyte_naive</td>\n",
       "      <td>11:2182224</td>\n",
       "      <td>rs689</td>\n",
       "      <td>ENSG00000229414</td>\n",
       "      <td>KCNQ1-AS1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>3185000</td>\n",
       "      <td>2882797</td>\n",
       "      <td>2882798</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.939009</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3253.0</td>\n",
       "      <td>0.649561</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>9.360000e-44</td>\n",
       "      <td>-</td>\n",
       "      <td>Schmiedel_2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1118</th>\n",
       "      <td>T1D_34594039_GCST90018925</td>\n",
       "      <td>monocyte_CD16_naive</td>\n",
       "      <td>monocyte_naive</td>\n",
       "      <td>11:2187855</td>\n",
       "      <td>rs12419447</td>\n",
       "      <td>ENSG00000229414</td>\n",
       "      <td>KCNQ1-AS1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>3190000</td>\n",
       "      <td>2882797</td>\n",
       "      <td>2882798</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.939009</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3253.0</td>\n",
       "      <td>0.649561</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>9.360000e-44</td>\n",
       "      <td>-</td>\n",
       "      <td>Schmiedel_2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1119</th>\n",
       "      <td>T1D_34594039_GCST90018925</td>\n",
       "      <td>monocyte_CD16_naive</td>\n",
       "      <td>monocyte_naive</td>\n",
       "      <td>11:2189983</td>\n",
       "      <td>rs4930043</td>\n",
       "      <td>ENSG00000229414</td>\n",
       "      <td>KCNQ1-AS1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>3190000</td>\n",
       "      <td>2882797</td>\n",
       "      <td>2882798</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.939009</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3253.0</td>\n",
       "      <td>0.649561</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>9.360000e-44</td>\n",
       "      <td>-</td>\n",
       "      <td>Schmiedel_2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1120</th>\n",
       "      <td>T1D_34594039_GCST90018925</td>\n",
       "      <td>monocyte_CD16_naive</td>\n",
       "      <td>monocyte_naive</td>\n",
       "      <td>11:2188238</td>\n",
       "      <td>rs6357</td>\n",
       "      <td>ENSG00000229414</td>\n",
       "      <td>KCNQ1-AS1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>3190000</td>\n",
       "      <td>2882797</td>\n",
       "      <td>2882798</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.939009</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3253.0</td>\n",
       "      <td>0.649561</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>9.360000e-44</td>\n",
       "      <td>-</td>\n",
       "      <td>Schmiedel_2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1121</th>\n",
       "      <td>T1D_34594039_GCST90018925</td>\n",
       "      <td>monocyte_CD16_naive</td>\n",
       "      <td>monocyte_naive</td>\n",
       "      <td>11:2188436</td>\n",
       "      <td>rs67482701</td>\n",
       "      <td>ENSG00000229414</td>\n",
       "      <td>KCNQ1-AS1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>3190000</td>\n",
       "      <td>2882797</td>\n",
       "      <td>2882798</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.939009</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3253.0</td>\n",
       "      <td>0.649561</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>9.360000e-44</td>\n",
       "      <td>-</td>\n",
       "      <td>Schmiedel_2018</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1525 rows  28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    gwas_source            ge_source       loop_source  \\\n",
       "52                 T1D_25751624               T-cell  CD4_T-cell_naive   \n",
       "53                 T1D_25751624               T-cell  CD4_T-cell_naive   \n",
       "83                 T1D_25751624               T-cell  CD4_T-cell_naive   \n",
       "111                T1D_25751624               T-cell  CD4_T-cell_naive   \n",
       "139                T1D_25751624               T-cell  CD4_T-cell_naive   \n",
       "...                         ...                  ...               ...   \n",
       "749   T1D_34594039_GCST90018925  monocyte_CD16_naive    monocyte_naive   \n",
       "1118  T1D_34594039_GCST90018925  monocyte_CD16_naive    monocyte_naive   \n",
       "1119  T1D_34594039_GCST90018925  monocyte_CD16_naive    monocyte_naive   \n",
       "1120  T1D_34594039_GCST90018925  monocyte_CD16_naive    monocyte_naive   \n",
       "1121  T1D_34594039_GCST90018925  monocyte_CD16_naive    monocyte_naive   \n",
       "\n",
       "             sid        rsid           geneid   gene_name  chrom   snp_pos  \\\n",
       "52    6:90936894    rs206913  ENSG00000260271  AL132996.1   chr6  91940000   \n",
       "53    6:90939178  rs58516003  ENSG00000260271  AL132996.1   chr6  91940000   \n",
       "83    6:90950736   rs2256672  ENSG00000260271  AL132996.1   chr6  91955000   \n",
       "111   6:90960329    rs673780  ENSG00000260271  AL132996.1   chr6  91965000   \n",
       "139   6:90968949    rs619192  ENSG00000260271  AL132996.1   chr6  91970000   \n",
       "...          ...         ...              ...         ...    ...       ...   \n",
       "749   11:2182224       rs689  ENSG00000229414   KCNQ1-AS1  chr11   3185000   \n",
       "1118  11:2187855  rs12419447  ENSG00000229414   KCNQ1-AS1  chr11   3190000   \n",
       "1119  11:2189983   rs4930043  ENSG00000229414   KCNQ1-AS1  chr11   3190000   \n",
       "1120  11:2188238      rs6357  ENSG00000229414   KCNQ1-AS1  chr11   3190000   \n",
       "1121  11:2188436  rs67482701  ENSG00000229414   KCNQ1-AS1  chr11   3190000   \n",
       "\n",
       "      tss_start   tss_end  is_eqtl_pair  is_coloc_pair  is_closest_gene  \\\n",
       "52     91030161  91030162             0              1                0   \n",
       "53     91030161  91030162             0              1                0   \n",
       "83     91030161  91030162             0              1                0   \n",
       "111    91030161  91030162             0              1                0   \n",
       "139    91030161  91030162             0              1                0   \n",
       "...         ...       ...           ...            ...              ...   \n",
       "749     2882797   2882798             0              1                0   \n",
       "1118    2882797   2882798             0              1                0   \n",
       "1119    2882797   2882798             0              1                0   \n",
       "1120    2882797   2882798             0              1                0   \n",
       "1121    2882797   2882798             0              1                0   \n",
       "\n",
       "      has_fithichip_loop  eqtl_pval  dist      ppH4  gene_start  gene_end  \\\n",
       "52                     1        NaN   NaN  0.818894    91030162  91072351   \n",
       "53                     1        NaN   NaN  0.818894    91030162  91072351   \n",
       "83                     1        NaN   NaN  0.818894    91030162  91072351   \n",
       "111                    1        NaN   NaN  0.818894    91030162  91072351   \n",
       "139                    1        NaN   NaN  0.818894    91030162  91072351   \n",
       "...                  ...        ...   ...       ...         ...       ...   \n",
       "749                    1        NaN   NaN  0.939009     2861365   2882798   \n",
       "1118                   1        NaN   NaN  0.939009     2861365   2882798   \n",
       "1119                   1        NaN   NaN  0.939009     2861365   2882798   \n",
       "1120                   1        NaN   NaN  0.939009     2861365   2882798   \n",
       "1121                   1        NaN   NaN  0.939009     2861365   2882798   \n",
       "\n",
       "      ref  alt      AC        AF      AN     gwas_pval gene_strand  \\\n",
       "52    NaN  NaN  3488.0  0.696486  5008.0  7.803700e-12           +   \n",
       "53    NaN  NaN  3488.0  0.696486  5008.0  7.803700e-12           +   \n",
       "83    NaN  NaN  3488.0  0.696486  5008.0  7.803700e-12           +   \n",
       "111   NaN  NaN  3488.0  0.696486  5008.0  7.803700e-12           +   \n",
       "139   NaN  NaN  3488.0  0.696486  5008.0  7.803700e-12           +   \n",
       "...   ...  ...     ...       ...     ...           ...         ...   \n",
       "749   NaN  NaN  3253.0  0.649561  5008.0  9.360000e-44           -   \n",
       "1118  NaN  NaN  3253.0  0.649561  5008.0  9.360000e-44           -   \n",
       "1119  NaN  NaN  3253.0  0.649561  5008.0  9.360000e-44           -   \n",
       "1120  NaN  NaN  3253.0  0.649561  5008.0  9.360000e-44           -   \n",
       "1121  NaN  NaN  3253.0  0.649561  5008.0  9.360000e-44           -   \n",
       "\n",
       "         eqtl_source  \n",
       "52         BLUEPRINT  \n",
       "53         BLUEPRINT  \n",
       "83         BLUEPRINT  \n",
       "111        BLUEPRINT  \n",
       "139        BLUEPRINT  \n",
       "...              ...  \n",
       "749   Schmiedel_2018  \n",
       "1118  Schmiedel_2018  \n",
       "1119  Schmiedel_2018  \n",
       "1120  Schmiedel_2018  \n",
       "1121  Schmiedel_2018  \n",
       "\n",
       "[1525 rows x 28 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sg_with_coloc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3d600b3f",
   "metadata": {},
   "source": [
    "#### Extract all pairs with a FitHiChIP Loop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "37116dc8",
   "metadata": {},
   "outputs": [],
   "source": [
    "sg_with_loops = sgl_data.loc[(sgl_data.has_fithichip_loop == 1), major_cols]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "5fefc9c2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 1525 SNP-Gene pairs with a loop.\n"
     ]
    }
   ],
   "source": [
    "print('There are {} SNP-Gene pairs with a loop.'.format(len(sg_with_loops)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "a072cc14",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gwas_source</th>\n",
       "      <th>ge_source</th>\n",
       "      <th>loop_source</th>\n",
       "      <th>sid</th>\n",
       "      <th>rsid</th>\n",
       "      <th>geneid</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>tss_start</th>\n",
       "      <th>tss_end</th>\n",
       "      <th>is_eqtl_pair</th>\n",
       "      <th>is_coloc_pair</th>\n",
       "      <th>is_closest_gene</th>\n",
       "      <th>has_fithichip_loop</th>\n",
       "      <th>eqtl_pval</th>\n",
       "      <th>dist</th>\n",
       "      <th>ppH4</th>\n",
       "      <th>gene_start</th>\n",
       "      <th>gene_end</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gwas_pval</th>\n",
       "      <th>gene_strand</th>\n",
       "      <th>eqtl_source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>T1D_25751624</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "      <td>6:90936894</td>\n",
       "      <td>rs206913</td>\n",
       "      <td>ENSG00000260271</td>\n",
       "      <td>AL132996.1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>91940000</td>\n",
       "      <td>91030161</td>\n",
       "      <td>91030162</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.818894</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3488.0</td>\n",
       "      <td>0.696486</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>7.803700e-12</td>\n",
       "      <td>+</td>\n",
       "      <td>BLUEPRINT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>T1D_25751624</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "      <td>6:90939178</td>\n",
       "      <td>rs58516003</td>\n",
       "      <td>ENSG00000260271</td>\n",
       "      <td>AL132996.1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>91940000</td>\n",
       "      <td>91030161</td>\n",
       "      <td>91030162</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.818894</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3488.0</td>\n",
       "      <td>0.696486</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>7.803700e-12</td>\n",
       "      <td>+</td>\n",
       "      <td>BLUEPRINT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>T1D_25751624</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "      <td>6:90950736</td>\n",
       "      <td>rs2256672</td>\n",
       "      <td>ENSG00000260271</td>\n",
       "      <td>AL132996.1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>91955000</td>\n",
       "      <td>91030161</td>\n",
       "      <td>91030162</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.818894</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3488.0</td>\n",
       "      <td>0.696486</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>7.803700e-12</td>\n",
       "      <td>+</td>\n",
       "      <td>BLUEPRINT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>T1D_25751624</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "      <td>6:90960329</td>\n",
       "      <td>rs673780</td>\n",
       "      <td>ENSG00000260271</td>\n",
       "      <td>AL132996.1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>91965000</td>\n",
       "      <td>91030161</td>\n",
       "      <td>91030162</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.818894</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3488.0</td>\n",
       "      <td>0.696486</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>7.803700e-12</td>\n",
       "      <td>+</td>\n",
       "      <td>BLUEPRINT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>T1D_25751624</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>CD4_T-cell_naive</td>\n",
       "      <td>6:90968949</td>\n",
       "      <td>rs619192</td>\n",
       "      <td>ENSG00000260271</td>\n",
       "      <td>AL132996.1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>91970000</td>\n",
       "      <td>91030161</td>\n",
       "      <td>91030162</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.818894</td>\n",
       "      <td>91030162</td>\n",
       "      <td>91072351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3488.0</td>\n",
       "      <td>0.696486</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>7.803700e-12</td>\n",
       "      <td>+</td>\n",
       "      <td>BLUEPRINT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>749</th>\n",
       "      <td>T1D_34594039_GCST90018925</td>\n",
       "      <td>monocyte_CD16_naive</td>\n",
       "      <td>monocyte_naive</td>\n",
       "      <td>11:2182224</td>\n",
       "      <td>rs689</td>\n",
       "      <td>ENSG00000229414</td>\n",
       "      <td>KCNQ1-AS1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>3185000</td>\n",
       "      <td>2882797</td>\n",
       "      <td>2882798</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.939009</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3253.0</td>\n",
       "      <td>0.649561</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>9.360000e-44</td>\n",
       "      <td>-</td>\n",
       "      <td>Schmiedel_2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1118</th>\n",
       "      <td>T1D_34594039_GCST90018925</td>\n",
       "      <td>monocyte_CD16_naive</td>\n",
       "      <td>monocyte_naive</td>\n",
       "      <td>11:2187855</td>\n",
       "      <td>rs12419447</td>\n",
       "      <td>ENSG00000229414</td>\n",
       "      <td>KCNQ1-AS1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>3190000</td>\n",
       "      <td>2882797</td>\n",
       "      <td>2882798</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.939009</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3253.0</td>\n",
       "      <td>0.649561</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>9.360000e-44</td>\n",
       "      <td>-</td>\n",
       "      <td>Schmiedel_2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1119</th>\n",
       "      <td>T1D_34594039_GCST90018925</td>\n",
       "      <td>monocyte_CD16_naive</td>\n",
       "      <td>monocyte_naive</td>\n",
       "      <td>11:2189983</td>\n",
       "      <td>rs4930043</td>\n",
       "      <td>ENSG00000229414</td>\n",
       "      <td>KCNQ1-AS1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>3190000</td>\n",
       "      <td>2882797</td>\n",
       "      <td>2882798</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.939009</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3253.0</td>\n",
       "      <td>0.649561</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>9.360000e-44</td>\n",
       "      <td>-</td>\n",
       "      <td>Schmiedel_2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1120</th>\n",
       "      <td>T1D_34594039_GCST90018925</td>\n",
       "      <td>monocyte_CD16_naive</td>\n",
       "      <td>monocyte_naive</td>\n",
       "      <td>11:2188238</td>\n",
       "      <td>rs6357</td>\n",
       "      <td>ENSG00000229414</td>\n",
       "      <td>KCNQ1-AS1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>3190000</td>\n",
       "      <td>2882797</td>\n",
       "      <td>2882798</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.939009</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3253.0</td>\n",
       "      <td>0.649561</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>9.360000e-44</td>\n",
       "      <td>-</td>\n",
       "      <td>Schmiedel_2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1121</th>\n",
       "      <td>T1D_34594039_GCST90018925</td>\n",
       "      <td>monocyte_CD16_naive</td>\n",
       "      <td>monocyte_naive</td>\n",
       "      <td>11:2188436</td>\n",
       "      <td>rs67482701</td>\n",
       "      <td>ENSG00000229414</td>\n",
       "      <td>KCNQ1-AS1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>3190000</td>\n",
       "      <td>2882797</td>\n",
       "      <td>2882798</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.939009</td>\n",
       "      <td>2861365</td>\n",
       "      <td>2882798</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3253.0</td>\n",
       "      <td>0.649561</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>9.360000e-44</td>\n",
       "      <td>-</td>\n",
       "      <td>Schmiedel_2018</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1525 rows  28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    gwas_source            ge_source       loop_source  \\\n",
       "52                 T1D_25751624               T-cell  CD4_T-cell_naive   \n",
       "53                 T1D_25751624               T-cell  CD4_T-cell_naive   \n",
       "83                 T1D_25751624               T-cell  CD4_T-cell_naive   \n",
       "111                T1D_25751624               T-cell  CD4_T-cell_naive   \n",
       "139                T1D_25751624               T-cell  CD4_T-cell_naive   \n",
       "...                         ...                  ...               ...   \n",
       "749   T1D_34594039_GCST90018925  monocyte_CD16_naive    monocyte_naive   \n",
       "1118  T1D_34594039_GCST90018925  monocyte_CD16_naive    monocyte_naive   \n",
       "1119  T1D_34594039_GCST90018925  monocyte_CD16_naive    monocyte_naive   \n",
       "1120  T1D_34594039_GCST90018925  monocyte_CD16_naive    monocyte_naive   \n",
       "1121  T1D_34594039_GCST90018925  monocyte_CD16_naive    monocyte_naive   \n",
       "\n",
       "             sid        rsid           geneid   gene_name  chrom   snp_pos  \\\n",
       "52    6:90936894    rs206913  ENSG00000260271  AL132996.1   chr6  91940000   \n",
       "53    6:90939178  rs58516003  ENSG00000260271  AL132996.1   chr6  91940000   \n",
       "83    6:90950736   rs2256672  ENSG00000260271  AL132996.1   chr6  91955000   \n",
       "111   6:90960329    rs673780  ENSG00000260271  AL132996.1   chr6  91965000   \n",
       "139   6:90968949    rs619192  ENSG00000260271  AL132996.1   chr6  91970000   \n",
       "...          ...         ...              ...         ...    ...       ...   \n",
       "749   11:2182224       rs689  ENSG00000229414   KCNQ1-AS1  chr11   3185000   \n",
       "1118  11:2187855  rs12419447  ENSG00000229414   KCNQ1-AS1  chr11   3190000   \n",
       "1119  11:2189983   rs4930043  ENSG00000229414   KCNQ1-AS1  chr11   3190000   \n",
       "1120  11:2188238      rs6357  ENSG00000229414   KCNQ1-AS1  chr11   3190000   \n",
       "1121  11:2188436  rs67482701  ENSG00000229414   KCNQ1-AS1  chr11   3190000   \n",
       "\n",
       "      tss_start   tss_end  is_eqtl_pair  is_coloc_pair  is_closest_gene  \\\n",
       "52     91030161  91030162             0              1                0   \n",
       "53     91030161  91030162             0              1                0   \n",
       "83     91030161  91030162             0              1                0   \n",
       "111    91030161  91030162             0              1                0   \n",
       "139    91030161  91030162             0              1                0   \n",
       "...         ...       ...           ...            ...              ...   \n",
       "749     2882797   2882798             0              1                0   \n",
       "1118    2882797   2882798             0              1                0   \n",
       "1119    2882797   2882798             0              1                0   \n",
       "1120    2882797   2882798             0              1                0   \n",
       "1121    2882797   2882798             0              1                0   \n",
       "\n",
       "      has_fithichip_loop  eqtl_pval  dist      ppH4  gene_start  gene_end  \\\n",
       "52                     1        NaN   NaN  0.818894    91030162  91072351   \n",
       "53                     1        NaN   NaN  0.818894    91030162  91072351   \n",
       "83                     1        NaN   NaN  0.818894    91030162  91072351   \n",
       "111                    1        NaN   NaN  0.818894    91030162  91072351   \n",
       "139                    1        NaN   NaN  0.818894    91030162  91072351   \n",
       "...                  ...        ...   ...       ...         ...       ...   \n",
       "749                    1        NaN   NaN  0.939009     2861365   2882798   \n",
       "1118                   1        NaN   NaN  0.939009     2861365   2882798   \n",
       "1119                   1        NaN   NaN  0.939009     2861365   2882798   \n",
       "1120                   1        NaN   NaN  0.939009     2861365   2882798   \n",
       "1121                   1        NaN   NaN  0.939009     2861365   2882798   \n",
       "\n",
       "      ref  alt      AC        AF      AN     gwas_pval gene_strand  \\\n",
       "52    NaN  NaN  3488.0  0.696486  5008.0  7.803700e-12           +   \n",
       "53    NaN  NaN  3488.0  0.696486  5008.0  7.803700e-12           +   \n",
       "83    NaN  NaN  3488.0  0.696486  5008.0  7.803700e-12           +   \n",
       "111   NaN  NaN  3488.0  0.696486  5008.0  7.803700e-12           +   \n",
       "139   NaN  NaN  3488.0  0.696486  5008.0  7.803700e-12           +   \n",
       "...   ...  ...     ...       ...     ...           ...         ...   \n",
       "749   NaN  NaN  3253.0  0.649561  5008.0  9.360000e-44           -   \n",
       "1118  NaN  NaN  3253.0  0.649561  5008.0  9.360000e-44           -   \n",
       "1119  NaN  NaN  3253.0  0.649561  5008.0  9.360000e-44           -   \n",
       "1120  NaN  NaN  3253.0  0.649561  5008.0  9.360000e-44           -   \n",
       "1121  NaN  NaN  3253.0  0.649561  5008.0  9.360000e-44           -   \n",
       "\n",
       "         eqtl_source  \n",
       "52         BLUEPRINT  \n",
       "53         BLUEPRINT  \n",
       "83         BLUEPRINT  \n",
       "111        BLUEPRINT  \n",
       "139        BLUEPRINT  \n",
       "...              ...  \n",
       "749   Schmiedel_2018  \n",
       "1118  Schmiedel_2018  \n",
       "1119  Schmiedel_2018  \n",
       "1120  Schmiedel_2018  \n",
       "1121  Schmiedel_2018  \n",
       "\n",
       "[1525 rows x 28 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sg_with_loops"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1c964b2",
   "metadata": {},
   "source": [
    "## Finalizing the super master table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "ec786699",
   "metadata": {},
   "outputs": [],
   "source": [
    "#data = data.loc[:, new_order]\n",
    "sgl_data.drop_duplicates(subset=['sid', 'geneid', 'eqtl_source', 'ge_source', 'loop_source'], inplace=True)\n",
    "master_fn = os.path.join(outdir, 'super_master.snp_gene_loop.analysis.tsv')\n",
    "sgl_data[major_cols].to_csv(master_fn, sep='\\t', index=False, na_rep='nan')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "74e3a543",
   "metadata": {},
   "outputs": [],
   "source": [
    "master_fn = os.path.join(outdir, 'super_master.snp_gene_loop.analysis.xlsx')\n",
    "xdata = sgl_data.sort_values(['rsid', 'gwas_source','eqtl_source', 'ge_source', 'loop_source', 'geneid'])\n",
    "xdata = xdata[major_cols].set_index(['rsid', 'gwas_source','eqtl_source', 'ge_source', 'loop_source', 'geneid'])\n",
    "xdata.to_excel(master_fn, na_rep='nan')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8a601330",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06ac6070",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "26396b70",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d25e9dfb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0cef6821",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5aa157aa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f0e28fa3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cbbfe0dc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2aae39f5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "28e6dc10",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3db32419",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6a9c15bc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "547a7996",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3fd019d3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71ef4cf0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d2b3bad5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5ff275eb",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "HiChIP-DB",
   "language": "python",
   "name": "hichip-db"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
