{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "7b5ad180",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os \n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import subprocess\n",
    "import glob\n",
    "import pybedtools as pbt \n",
    "from IPython.display import HTML\n",
    "\n",
    "pbt.set_bedtools_path('/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/')\n",
    "os.chdir('/mnt/BioHome/jreyna/jreyna/projects/dchallenge/')\n",
    "\n",
    "gsizes = 'results/refs/hg19/hg19.chrom.sizes'\n",
    "res = 5000\n",
    "\n",
    "# make the directory to save our data\n",
    "outdir = 'results/main/Intersect_T1D_Finemap_GWAS_SNPs_with_HiChIP'\n",
    "os.makedirs(outdir, exist_ok=True)\n",
    "bedpe_cols = ['chrA', 'startA', 'endA', 'chrB', 'startB', 'endB']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "25ec296e",
   "metadata": {},
   "source": [
    "## Load Fine Mapped GWAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "46298a0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "gwas = 'results/main/finemapping/T1D_34012112_Gaulton/GRCh37/offset_1000000/Summary/sss/FINAL_top_snp_credible_set.txt'\n",
    "gwas_df = pd.read_table(gwas)\n",
    "gwas_df.loc[:, 'bin_start'] = np.floor(gwas_df.loc[:, 'position'] / res).astype(int) * res\n",
    "gwas_df.loc[:, 'bin_end'] = gwas_df.loc[:, 'bin_start'] + res\n",
    "\n",
    "gwas_bed = gwas_df.iloc[:, [4,5,5]]\n",
    "gwas_bed.columns = ['chrom', 'start', 'end']\n",
    "gwas_bed['start'] = gwas_bed['start'] - 1\n",
    "gwas_pbt = pbt.BedTool.from_dataframe(gwas_bed)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8b1e3652",
   "metadata": {},
   "source": [
    "## Load HiChIP Loops"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "b49da1dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_seB(x): \n",
    "    s,e = x.split(':')[1].split('-')\n",
    "    e = e.split(',')[0]\n",
    "    return((s,e))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "6cd23b57",
   "metadata": {},
   "outputs": [],
   "source": [
    "loops = 'results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/'\n",
    "loops += '*/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01_WashU.bed.gz'\n",
    "loops = glob.glob(loops)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "6edc6f89",
   "metadata": {},
   "outputs": [],
   "source": [
    "loops = ['results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/NCM/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01_WashU.bed.gz',\n",
    " 'results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CD4N/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01_WashU.bed.gz',\n",
    " 'results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CD8N/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01_WashU.bed.gz',\n",
    " 'results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/NB/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01_WashU.bed.gz',\n",
    " 'results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CM/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01_WashU.bed.gz',\n",
    " 'results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/NK/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01_WashU.bed.gz']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "1c387b45",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/NCM/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01_WashU.bed.gz\n",
      "results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CD4N/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01_WashU.bed.gz\n",
      "results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CD8N/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01_WashU.bed.gz\n",
      "results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/NB/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01_WashU.bed.gz\n",
      "results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CM/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01_WashU.bed.gz\n",
      "results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/NK/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01_WashU.bed.gz\n"
     ]
    }
   ],
   "source": [
    "loop_data = []\n",
    "for loop in loops:\n",
    "    print(loop)\n",
    "    \n",
    "    cline = loop.split('/')[5]\n",
    "    df = pd.read_table(loop, header=None)    \n",
    "    df.columns = ['chrom', 'startA', 'endA', 'seB', 'e1', 'e2']\n",
    "    \n",
    "    df['chrom'] = df['chrom'].str.replace('chr', '')\n",
    "    \n",
    "    df['startB'], df['endB'] = zip(*df['seB'].apply(parse_seB))\n",
    "    df['startB'] = df['startB'].astype(int)\n",
    "    \n",
    "    df['startA'] = df['startA'] + 1 \n",
    "    df['endA'] = df['startA'] + res\n",
    "\n",
    "    df['startB'] = df['startB'] + 1 \n",
    "    df['endB'] = df['startB'] + res\n",
    "    \n",
    "    # re-organize the data into bedpe-like\n",
    "    df = df.iloc[:, [0,1,2,0,6,7,3,4,5]]\n",
    "\n",
    "    # add cell type\n",
    "    df['cline'] = cline\n",
    "\n",
    "    loop_data.append(df) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "8ebb79cd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>startA</th>\n",
       "      <th>endA</th>\n",
       "      <th>chrom</th>\n",
       "      <th>startB</th>\n",
       "      <th>endB</th>\n",
       "      <th>seB</th>\n",
       "      <th>e1</th>\n",
       "      <th>e2</th>\n",
       "      <th>cline</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>712500</td>\n",
       "      <td>717500</td>\n",
       "      <td>1</td>\n",
       "      <td>757500</td>\n",
       "      <td>762500</td>\n",
       "      <td>chr1:757499-757501,39.9592</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>NK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>712500</td>\n",
       "      <td>717500</td>\n",
       "      <td>1</td>\n",
       "      <td>762500</td>\n",
       "      <td>767500</td>\n",
       "      <td>chr1:762499-762501,18.9118</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>NK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>712500</td>\n",
       "      <td>717500</td>\n",
       "      <td>1</td>\n",
       "      <td>892500</td>\n",
       "      <td>897500</td>\n",
       "      <td>chr1:892499-892501,2.57933</td>\n",
       "      <td>3</td>\n",
       "      <td>.</td>\n",
       "      <td>NK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>757500</td>\n",
       "      <td>762500</td>\n",
       "      <td>1</td>\n",
       "      <td>782500</td>\n",
       "      <td>787500</td>\n",
       "      <td>chr1:782499-782501,3.26331</td>\n",
       "      <td>4</td>\n",
       "      <td>.</td>\n",
       "      <td>NK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>757500</td>\n",
       "      <td>762500</td>\n",
       "      <td>1</td>\n",
       "      <td>947500</td>\n",
       "      <td>952500</td>\n",
       "      <td>chr1:947499-947501,3.43179</td>\n",
       "      <td>5</td>\n",
       "      <td>.</td>\n",
       "      <td>NK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129885</th>\n",
       "      <td>9</td>\n",
       "      <td>140602500</td>\n",
       "      <td>140607500</td>\n",
       "      <td>9</td>\n",
       "      <td>140652500</td>\n",
       "      <td>140657500</td>\n",
       "      <td>chr9:140652499-140652501,2.79712</td>\n",
       "      <td>129886</td>\n",
       "      <td>.</td>\n",
       "      <td>NK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129886</th>\n",
       "      <td>9</td>\n",
       "      <td>140612500</td>\n",
       "      <td>140617500</td>\n",
       "      <td>9</td>\n",
       "      <td>140627500</td>\n",
       "      <td>140632500</td>\n",
       "      <td>chr9:140627499-140627501,2.12146</td>\n",
       "      <td>129887</td>\n",
       "      <td>.</td>\n",
       "      <td>NK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129887</th>\n",
       "      <td>9</td>\n",
       "      <td>140612500</td>\n",
       "      <td>140617500</td>\n",
       "      <td>9</td>\n",
       "      <td>140632500</td>\n",
       "      <td>140637500</td>\n",
       "      <td>chr9:140632499-140632501,2.63493</td>\n",
       "      <td>129888</td>\n",
       "      <td>.</td>\n",
       "      <td>NK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129888</th>\n",
       "      <td>9</td>\n",
       "      <td>140612500</td>\n",
       "      <td>140617500</td>\n",
       "      <td>9</td>\n",
       "      <td>140792500</td>\n",
       "      <td>140797500</td>\n",
       "      <td>chr9:140792499-140792501,3.83825</td>\n",
       "      <td>129889</td>\n",
       "      <td>.</td>\n",
       "      <td>NK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129889</th>\n",
       "      <td>9</td>\n",
       "      <td>140627500</td>\n",
       "      <td>140632500</td>\n",
       "      <td>9</td>\n",
       "      <td>140652500</td>\n",
       "      <td>140657500</td>\n",
       "      <td>chr9:140652499-140652501,2.5259</td>\n",
       "      <td>129890</td>\n",
       "      <td>.</td>\n",
       "      <td>NK</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>129890 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chrom     startA       endA chrom     startB       endB  \\\n",
       "0          1     712500     717500     1     757500     762500   \n",
       "1          1     712500     717500     1     762500     767500   \n",
       "2          1     712500     717500     1     892500     897500   \n",
       "3          1     757500     762500     1     782500     787500   \n",
       "4          1     757500     762500     1     947500     952500   \n",
       "...      ...        ...        ...   ...        ...        ...   \n",
       "129885     9  140602500  140607500     9  140652500  140657500   \n",
       "129886     9  140612500  140617500     9  140627500  140632500   \n",
       "129887     9  140612500  140617500     9  140632500  140637500   \n",
       "129888     9  140612500  140617500     9  140792500  140797500   \n",
       "129889     9  140627500  140632500     9  140652500  140657500   \n",
       "\n",
       "                                     seB      e1 e2 cline  \n",
       "0             chr1:757499-757501,39.9592       1  .    NK  \n",
       "1             chr1:762499-762501,18.9118       2  .    NK  \n",
       "2             chr1:892499-892501,2.57933       3  .    NK  \n",
       "3             chr1:782499-782501,3.26331       4  .    NK  \n",
       "4             chr1:947499-947501,3.43179       5  .    NK  \n",
       "...                                  ...     ... ..   ...  \n",
       "129885  chr9:140652499-140652501,2.79712  129886  .    NK  \n",
       "129886  chr9:140627499-140627501,2.12146  129887  .    NK  \n",
       "129887  chr9:140632499-140632501,2.63493  129888  .    NK  \n",
       "129888  chr9:140792499-140792501,3.83825  129889  .    NK  \n",
       "129889   chr9:140652499-140652501,2.5259  129890  .    NK  \n",
       "\n",
       "[129890 rows x 10 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21cafb1e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0d1ef71d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db3efc8a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "541b020f",
   "metadata": {},
   "outputs": [],
   "source": [
    "loop_df = pd.concat(loop_data)\n",
    "loop_bed = loop_df.iloc[:, [0,1,2,3,4,5,-1]]\n",
    "loop_pbt = pbt.BedTool.from_dataframe(loop_bed)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8bf47849",
   "metadata": {},
   "source": [
    "## Intersect Fine Mapped GWAS and loops"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c7b3cadc",
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_slop = 5000\n",
    "intersect_pbt = loop_pbt.pair_to_bed(gwas_pbt.slop(b=snp_slop, g=gsizes), type='either')\n",
    "#intersect_pbt = loop_pbt.pair_to_bed(gwas_pbt.slop(l=snp_slop, r=0, g=gsizes), type='either')\n",
    "#intersect_pbt = loop_pbt.pair_to_bed(gwas_pbt.slop(b=100000, g=gsizes), type='either')\n",
    "gwas_hichip = intersect_pbt.to_dataframe()\n",
    "\n",
    "gwas_hichip = gwas_hichip.iloc[:, [7,8,9,0,1,2,3,4,5,6]]\n",
    "\n",
    "loop_cols = ['{}_loop'.format(x) for x in bedpe_cols]\n",
    "gwas_hichip.columns = ['chr_snp', 'start_snp', 'end_snp'] + loop_cols + ['cline_loop']\n",
    "gwas_hichip['start_snp'] += snp_slop\n",
    "gwas_hichip['end_snp'] -= snp_slop\n",
    "gwas_hichip = gwas_hichip.merge(gwas_df, left_on=['chr_snp', 'end_snp'], right_on=['chromosome', 'position'])\n",
    "gwas_hichip.drop('start_snp', axis=1, inplace=True)\n",
    "gwas_hichip.rename(columns={'end_snp': 'position_snp'}, inplace=True)\n",
    "gwas_hichip['sid'] = 'chr' +  gwas_hichip['chr_snp'].astype(str) + ':' + gwas_hichip['position_snp'].astype(str)\n",
    "\n",
    "# add loop ids\n",
    "def make_lid(sr, cols):\n",
    "    lid = sr[cols].tolist()\n",
    "    lid = [str(x) for x in lid]\n",
    "    lid = ':'.join(lid)\n",
    "    return(lid)\n",
    "\n",
    "lid_cols = [2,3,4,5,6,7]\n",
    "lids = []\n",
    "for sr in gwas_hichip.values: \n",
    "    new_lid = make_lid(sr, lid_cols)\n",
    "    lids.append(new_lid)\n",
    "gwas_hichip['loop_id'] = lids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "2defcbde",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr_snp</th>\n",
       "      <th>position_snp</th>\n",
       "      <th>chrA_loop</th>\n",
       "      <th>startA_loop</th>\n",
       "      <th>endA_loop</th>\n",
       "      <th>chrB_loop</th>\n",
       "      <th>startB_loop</th>\n",
       "      <th>endB_loop</th>\n",
       "      <th>cline_loop</th>\n",
       "      <th>regionID</th>\n",
       "      <th>...</th>\n",
       "      <th>z</th>\n",
       "      <th>prob</th>\n",
       "      <th>log10bf</th>\n",
       "      <th>mean</th>\n",
       "      <th>sd</th>\n",
       "      <th>mean_incl</th>\n",
       "      <th>sd_incl</th>\n",
       "      <th>pval</th>\n",
       "      <th>sid</th>\n",
       "      <th>loop_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11</td>\n",
       "      <td>128624171</td>\n",
       "      <td>11</td>\n",
       "      <td>127902500</td>\n",
       "      <td>127912500</td>\n",
       "      <td>11</td>\n",
       "      <td>128627500</td>\n",
       "      <td>128637500</td>\n",
       "      <td>NCM</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>2.15558</td>\n",
       "      <td>0.952259</td>\n",
       "      <td>4.80804</td>\n",
       "      <td>0.278615</td>\n",
       "      <td>0.548181</td>\n",
       "      <td>0.292583</td>\n",
       "      <td>0.558105</td>\n",
       "      <td>0.015558</td>\n",
       "      <td>chr11:128624171</td>\n",
       "      <td>11:127902500:127912500:11:128627500:128637500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>11</td>\n",
       "      <td>128624171</td>\n",
       "      <td>11</td>\n",
       "      <td>128142500</td>\n",
       "      <td>128152500</td>\n",
       "      <td>11</td>\n",
       "      <td>128617500</td>\n",
       "      <td>128627500</td>\n",
       "      <td>NCM</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>2.15558</td>\n",
       "      <td>0.952259</td>\n",
       "      <td>4.80804</td>\n",
       "      <td>0.278615</td>\n",
       "      <td>0.548181</td>\n",
       "      <td>0.292583</td>\n",
       "      <td>0.558105</td>\n",
       "      <td>0.015558</td>\n",
       "      <td>chr11:128624171</td>\n",
       "      <td>11:128142500:128152500:11:128617500:128627500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>11</td>\n",
       "      <td>128624171</td>\n",
       "      <td>11</td>\n",
       "      <td>128142500</td>\n",
       "      <td>128152500</td>\n",
       "      <td>11</td>\n",
       "      <td>128627500</td>\n",
       "      <td>128637500</td>\n",
       "      <td>NCM</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>2.15558</td>\n",
       "      <td>0.952259</td>\n",
       "      <td>4.80804</td>\n",
       "      <td>0.278615</td>\n",
       "      <td>0.548181</td>\n",
       "      <td>0.292583</td>\n",
       "      <td>0.558105</td>\n",
       "      <td>0.015558</td>\n",
       "      <td>chr11:128624171</td>\n",
       "      <td>11:128142500:128152500:11:128627500:128637500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>11</td>\n",
       "      <td>128624171</td>\n",
       "      <td>11</td>\n",
       "      <td>128162500</td>\n",
       "      <td>128172500</td>\n",
       "      <td>11</td>\n",
       "      <td>128622500</td>\n",
       "      <td>128632500</td>\n",
       "      <td>NCM</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>2.15558</td>\n",
       "      <td>0.952259</td>\n",
       "      <td>4.80804</td>\n",
       "      <td>0.278615</td>\n",
       "      <td>0.548181</td>\n",
       "      <td>0.292583</td>\n",
       "      <td>0.558105</td>\n",
       "      <td>0.015558</td>\n",
       "      <td>chr11:128624171</td>\n",
       "      <td>11:128162500:128172500:11:128622500:128632500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11</td>\n",
       "      <td>128624171</td>\n",
       "      <td>11</td>\n",
       "      <td>128392500</td>\n",
       "      <td>128402500</td>\n",
       "      <td>11</td>\n",
       "      <td>128617500</td>\n",
       "      <td>128627500</td>\n",
       "      <td>NCM</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>2.15558</td>\n",
       "      <td>0.952259</td>\n",
       "      <td>4.80804</td>\n",
       "      <td>0.278615</td>\n",
       "      <td>0.548181</td>\n",
       "      <td>0.292583</td>\n",
       "      <td>0.558105</td>\n",
       "      <td>0.015558</td>\n",
       "      <td>chr11:128624171</td>\n",
       "      <td>11:128392500:128402500:11:128617500:128627500</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   chr_snp  position_snp  chrA_loop  startA_loop  endA_loop  chrB_loop  \\\n",
       "0       11     128624171         11    127902500  127912500         11   \n",
       "1       11     128624171         11    128142500  128152500         11   \n",
       "2       11     128624171         11    128142500  128152500         11   \n",
       "3       11     128624171         11    128162500  128172500         11   \n",
       "4       11     128624171         11    128392500  128402500         11   \n",
       "\n",
       "   startB_loop  endB_loop cline_loop  regionID  ...        z      prob  \\\n",
       "0    128627500  128637500        NCM        20  ...  2.15558  0.952259   \n",
       "1    128617500  128627500        NCM        20  ...  2.15558  0.952259   \n",
       "2    128627500  128637500        NCM        20  ...  2.15558  0.952259   \n",
       "3    128622500  128632500        NCM        20  ...  2.15558  0.952259   \n",
       "4    128617500  128627500        NCM        20  ...  2.15558  0.952259   \n",
       "\n",
       "   log10bf      mean        sd mean_incl   sd_incl      pval              sid  \\\n",
       "0  4.80804  0.278615  0.548181  0.292583  0.558105  0.015558  chr11:128624171   \n",
       "1  4.80804  0.278615  0.548181  0.292583  0.558105  0.015558  chr11:128624171   \n",
       "2  4.80804  0.278615  0.548181  0.292583  0.558105  0.015558  chr11:128624171   \n",
       "3  4.80804  0.278615  0.548181  0.292583  0.558105  0.015558  chr11:128624171   \n",
       "4  4.80804  0.278615  0.548181  0.292583  0.558105  0.015558  chr11:128624171   \n",
       "\n",
       "                                         loop_id  \n",
       "0  11:127902500:127912500:11:128627500:128637500  \n",
       "1  11:128142500:128152500:11:128617500:128627500  \n",
       "2  11:128142500:128152500:11:128627500:128637500  \n",
       "3  11:128162500:128172500:11:128622500:128632500  \n",
       "4  11:128392500:128402500:11:128617500:128627500  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_hichip.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "74c4bd66",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2033, 30)"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_hichip.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3ec0240f",
   "metadata": {},
   "source": [
    "## Integrate genes "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1804679b",
   "metadata": {},
   "source": [
    "### Load the gene data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "8c7520b4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Load the gene data\n",
      "# make a genes pbt for intersection\n",
      "   chrom  start    end        gname          gene_id strand\n",
      "0      1  11868  11869      DDX11L1  ENSG00000223972      +\n",
      "12     1  29569  29570       WASH7P  ENSG00000227232      -\n",
      "25     1  17435  17436    MIR6859-1  ENSG00000278267      -\n",
      "28     1  29553  29554  MIR1302-2HG  ENSG00000243485      +\n",
      "36     1  30365  30366    MIR1302-2  ENSG00000284332      +\n",
      "There are 58825 genes in this GTF-derived file.\n"
     ]
    }
   ],
   "source": [
    "print('# Load the gene data')\n",
    "\n",
    "genes_fn = 'results/refs/gencode/v30/gencode.v30.annotation.bed'\n",
    "\n",
    "# load the gencode coords\n",
    "cols = ['chrom', 'start', 'end', 'strand', 'type', 'gene_id', 'gname']\n",
    "gencode = pd.read_table(genes_fn, header=None, names=cols)\n",
    "\n",
    "# extract just the genes\n",
    "genes_df = gencode.loc[gencode['type'].isin(['gene'])]\n",
    "genes_df = genes_df.loc[~genes_df.duplicated(subset='gene_id'), :]\n",
    "genes_df.loc[:, 'chrom'] = genes_df['chrom'].astype(str)\n",
    "genes_df = genes_df.iloc[:, [0,1,2,6,5,3]]\n",
    "\n",
    "# create a copy of the original gene bed before coordinate shrinking\n",
    "orig_genes_df = genes_df.copy()\n",
    "\n",
    "# convert the start/end position into start/end for the TSS\n",
    "# if the gene is + then the start is uses as the tss otherwise\n",
    "# the end is used as the tss\n",
    "genes_df.loc[(genes_df.strand == '+'), 'end'] = genes_df.loc[(genes_df.strand == '+'), 'start']\n",
    "genes_df.loc[(genes_df.strand == '+'), 'start'] = genes_df.loc[(genes_df.strand == '+'), 'start'] - 1\n",
    "genes_df.loc[(genes_df.strand == '-'), 'end'] = genes_df.loc[(genes_df.strand == '-'), 'end']\n",
    "genes_df.loc[(genes_df.strand == '-'), 'start'] = genes_df.loc[(genes_df.strand == '-'), 'end'] - 1\n",
    "genes_df.loc[:, 'chrom'] = genes_df.loc[:, 'chrom'].str.replace('chr', '')\n",
    "\n",
    "# make a genes pbt for intersection\n",
    "print(\"# make a genes pbt for intersection\")\n",
    "print(genes_df.head())\n",
    "genes_pbt = pbt.BedTool.from_dataframe(genes_df).sort()\n",
    "\n",
    "print('There are {} genes in this GTF-derived file.'.format(genes_df.shape[0]))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c0c92c29",
   "metadata": {},
   "source": [
    "### Extract anchors opposite of a SNP anchor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "ca94edfe",
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_anchor = []\n",
    "for i, sr in gwas_hichip.iterrows():\n",
    "    if (sr.startA_loop - snp_slop <= sr.position_snp) & (sr.position_snp <= sr.endA_loop + snp_slop):\n",
    "        snp_anchor.append('AnchorA')\n",
    "    elif (sr.startB_loop - snp_slop <= sr.position_snp) & (sr.position_snp <= sr.endB_loop + snp_slop):\n",
    "        snp_anchor.append('AnchorB')\n",
    "    else:\n",
    "        snp_anchor.append('bug')\n",
    "        print('bug')\n",
    "        break\n",
    "gwas_hichip.loc[:, 'snp_anchor'] = snp_anchor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "526d8af4",
   "metadata": {},
   "outputs": [],
   "source": [
    "nonsnp_anchorsA = gwas_hichip.loc[gwas_hichip['snp_anchor'] == 'AnchorA', \n",
    "                                  ['chrB_loop', 'startB_loop', 'endB_loop', 'sid', 'cline_loop']]\n",
    "nonsnp_anchorsB = gwas_hichip.loc[gwas_hichip['snp_anchor'] == 'AnchorB',\n",
    "                                  ['chrA_loop', 'startA_loop', 'endA_loop', 'sid', 'cline_loop']]\n",
    "nonsnp_anchorsA.columns = ['chr', 'start', 'end', 'sid', 'cline_loop']\n",
    "nonsnp_anchorsB.columns = ['chr', 'start', 'end', 'sid', 'cline_loop']\n",
    "nonsnp_anchors = pd.concat([nonsnp_anchorsA, nonsnp_anchorsB], axis=0)\n",
    "nonsnp_anchors_pbt = pbt.BedTool.from_dataframe(nonsnp_anchors)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "17a16801",
   "metadata": {},
   "source": [
    "### Intersecting genes on anchors opposing a SNP anchor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "ccec3c96",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_overlaps = nonsnp_anchors_pbt.slop(l=5000, r=0, g=gsizes).intersect(genes_pbt, wa=True, wb=True)\n",
    "gene_overlaps = gene_overlaps.to_dataframe()\n",
    "\n",
    "gene_overlaps.columns = ['chrSNP', 'startSNP', 'endSNP', 'sid', 'cline_loop',\n",
    "                         'chrGene', 'startGene', 'endGene',\n",
    "                         'genename', 'geneid', 'other']\n",
    "\n",
    "gwas_hichip_genes = gwas_hichip.merge(gene_overlaps, how='left', on=['sid', 'cline_loop'])\n",
    "gwas_hichip_genes = gwas_hichip_genes.loc[~gwas_hichip_genes.chrSNP.isna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "f92b0365",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr_snp</th>\n",
       "      <th>position_snp</th>\n",
       "      <th>chrA_loop</th>\n",
       "      <th>startA_loop</th>\n",
       "      <th>endA_loop</th>\n",
       "      <th>chrB_loop</th>\n",
       "      <th>startB_loop</th>\n",
       "      <th>endB_loop</th>\n",
       "      <th>cline_loop</th>\n",
       "      <th>regionID</th>\n",
       "      <th>...</th>\n",
       "      <th>snp_anchor</th>\n",
       "      <th>chrSNP</th>\n",
       "      <th>startSNP</th>\n",
       "      <th>endSNP</th>\n",
       "      <th>chrGene</th>\n",
       "      <th>startGene</th>\n",
       "      <th>endGene</th>\n",
       "      <th>genename</th>\n",
       "      <th>geneid</th>\n",
       "      <th>other</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11</td>\n",
       "      <td>128624171</td>\n",
       "      <td>11</td>\n",
       "      <td>127902500</td>\n",
       "      <td>127912500</td>\n",
       "      <td>11</td>\n",
       "      <td>128627500</td>\n",
       "      <td>128637500</td>\n",
       "      <td>NCM</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>AnchorB</td>\n",
       "      <td>11.0</td>\n",
       "      <td>128507500.0</td>\n",
       "      <td>128522500.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>128522389.0</td>\n",
       "      <td>128522390.0</td>\n",
       "      <td>MIR6090</td>\n",
       "      <td>ENSG00000276176</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>11</td>\n",
       "      <td>128624171</td>\n",
       "      <td>11</td>\n",
       "      <td>127902500</td>\n",
       "      <td>127912500</td>\n",
       "      <td>11</td>\n",
       "      <td>128627500</td>\n",
       "      <td>128637500</td>\n",
       "      <td>NCM</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>AnchorB</td>\n",
       "      <td>11.0</td>\n",
       "      <td>128582500.0</td>\n",
       "      <td>128597500.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>128587557.0</td>\n",
       "      <td>128587558.0</td>\n",
       "      <td>ETS1</td>\n",
       "      <td>ENSG00000134954</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>11</td>\n",
       "      <td>128624171</td>\n",
       "      <td>11</td>\n",
       "      <td>127902500</td>\n",
       "      <td>127912500</td>\n",
       "      <td>11</td>\n",
       "      <td>128627500</td>\n",
       "      <td>128637500</td>\n",
       "      <td>NCM</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>AnchorB</td>\n",
       "      <td>11.0</td>\n",
       "      <td>128582500.0</td>\n",
       "      <td>128597500.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>128587557.0</td>\n",
       "      <td>128587558.0</td>\n",
       "      <td>ETS1</td>\n",
       "      <td>ENSG00000134954</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>11</td>\n",
       "      <td>128624171</td>\n",
       "      <td>11</td>\n",
       "      <td>127902500</td>\n",
       "      <td>127912500</td>\n",
       "      <td>11</td>\n",
       "      <td>128627500</td>\n",
       "      <td>128637500</td>\n",
       "      <td>NCM</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>AnchorB</td>\n",
       "      <td>11.0</td>\n",
       "      <td>128587500.0</td>\n",
       "      <td>128602500.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>128587557.0</td>\n",
       "      <td>128587558.0</td>\n",
       "      <td>ETS1</td>\n",
       "      <td>ENSG00000134954</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11</td>\n",
       "      <td>128624171</td>\n",
       "      <td>11</td>\n",
       "      <td>128142500</td>\n",
       "      <td>128152500</td>\n",
       "      <td>11</td>\n",
       "      <td>128617500</td>\n",
       "      <td>128627500</td>\n",
       "      <td>NCM</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>AnchorB</td>\n",
       "      <td>11.0</td>\n",
       "      <td>128507500.0</td>\n",
       "      <td>128522500.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>128522389.0</td>\n",
       "      <td>128522390.0</td>\n",
       "      <td>MIR6090</td>\n",
       "      <td>ENSG00000276176</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8846</th>\n",
       "      <td>6</td>\n",
       "      <td>137963070</td>\n",
       "      <td>6</td>\n",
       "      <td>137542500</td>\n",
       "      <td>137552500</td>\n",
       "      <td>6</td>\n",
       "      <td>137962500</td>\n",
       "      <td>137972500</td>\n",
       "      <td>NB</td>\n",
       "      <td>72</td>\n",
       "      <td>...</td>\n",
       "      <td>AnchorB</td>\n",
       "      <td>6.0</td>\n",
       "      <td>137537500.0</td>\n",
       "      <td>137552500.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>137544371.0</td>\n",
       "      <td>137544372.0</td>\n",
       "      <td>BTF3L4P3</td>\n",
       "      <td>ENSG00000213108</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8847</th>\n",
       "      <td>6</td>\n",
       "      <td>137963070</td>\n",
       "      <td>6</td>\n",
       "      <td>137542500</td>\n",
       "      <td>137552500</td>\n",
       "      <td>6</td>\n",
       "      <td>137957500</td>\n",
       "      <td>137967500</td>\n",
       "      <td>CM</td>\n",
       "      <td>72</td>\n",
       "      <td>...</td>\n",
       "      <td>AnchorB</td>\n",
       "      <td>6.0</td>\n",
       "      <td>137537500.0</td>\n",
       "      <td>137552500.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>137544371.0</td>\n",
       "      <td>137544372.0</td>\n",
       "      <td>BTF3L4P3</td>\n",
       "      <td>ENSG00000213108</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8848</th>\n",
       "      <td>6</td>\n",
       "      <td>137963070</td>\n",
       "      <td>6</td>\n",
       "      <td>137542500</td>\n",
       "      <td>137552500</td>\n",
       "      <td>6</td>\n",
       "      <td>137957500</td>\n",
       "      <td>137967500</td>\n",
       "      <td>CM</td>\n",
       "      <td>72</td>\n",
       "      <td>...</td>\n",
       "      <td>AnchorB</td>\n",
       "      <td>6.0</td>\n",
       "      <td>137537500.0</td>\n",
       "      <td>137552500.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>137544371.0</td>\n",
       "      <td>137544372.0</td>\n",
       "      <td>BTF3L4P3</td>\n",
       "      <td>ENSG00000213108</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8849</th>\n",
       "      <td>6</td>\n",
       "      <td>137963070</td>\n",
       "      <td>6</td>\n",
       "      <td>137542500</td>\n",
       "      <td>137552500</td>\n",
       "      <td>6</td>\n",
       "      <td>137962500</td>\n",
       "      <td>137972500</td>\n",
       "      <td>CM</td>\n",
       "      <td>72</td>\n",
       "      <td>...</td>\n",
       "      <td>AnchorB</td>\n",
       "      <td>6.0</td>\n",
       "      <td>137537500.0</td>\n",
       "      <td>137552500.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>137544371.0</td>\n",
       "      <td>137544372.0</td>\n",
       "      <td>BTF3L4P3</td>\n",
       "      <td>ENSG00000213108</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8850</th>\n",
       "      <td>6</td>\n",
       "      <td>137963070</td>\n",
       "      <td>6</td>\n",
       "      <td>137542500</td>\n",
       "      <td>137552500</td>\n",
       "      <td>6</td>\n",
       "      <td>137962500</td>\n",
       "      <td>137972500</td>\n",
       "      <td>CM</td>\n",
       "      <td>72</td>\n",
       "      <td>...</td>\n",
       "      <td>AnchorB</td>\n",
       "      <td>6.0</td>\n",
       "      <td>137537500.0</td>\n",
       "      <td>137552500.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>137544371.0</td>\n",
       "      <td>137544372.0</td>\n",
       "      <td>BTF3L4P3</td>\n",
       "      <td>ENSG00000213108</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8460 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr_snp  position_snp  chrA_loop  startA_loop  endA_loop  chrB_loop  \\\n",
       "0          11     128624171         11    127902500  127912500         11   \n",
       "1          11     128624171         11    127902500  127912500         11   \n",
       "2          11     128624171         11    127902500  127912500         11   \n",
       "3          11     128624171         11    127902500  127912500         11   \n",
       "4          11     128624171         11    128142500  128152500         11   \n",
       "...       ...           ...        ...          ...        ...        ...   \n",
       "8846        6     137963070          6    137542500  137552500          6   \n",
       "8847        6     137963070          6    137542500  137552500          6   \n",
       "8848        6     137963070          6    137542500  137552500          6   \n",
       "8849        6     137963070          6    137542500  137552500          6   \n",
       "8850        6     137963070          6    137542500  137552500          6   \n",
       "\n",
       "      startB_loop  endB_loop cline_loop  regionID  ... snp_anchor  chrSNP  \\\n",
       "0       128627500  128637500        NCM        20  ...    AnchorB    11.0   \n",
       "1       128627500  128637500        NCM        20  ...    AnchorB    11.0   \n",
       "2       128627500  128637500        NCM        20  ...    AnchorB    11.0   \n",
       "3       128627500  128637500        NCM        20  ...    AnchorB    11.0   \n",
       "4       128617500  128627500        NCM        20  ...    AnchorB    11.0   \n",
       "...           ...        ...        ...       ...  ...        ...     ...   \n",
       "8846    137962500  137972500         NB        72  ...    AnchorB     6.0   \n",
       "8847    137957500  137967500         CM        72  ...    AnchorB     6.0   \n",
       "8848    137957500  137967500         CM        72  ...    AnchorB     6.0   \n",
       "8849    137962500  137972500         CM        72  ...    AnchorB     6.0   \n",
       "8850    137962500  137972500         CM        72  ...    AnchorB     6.0   \n",
       "\n",
       "         startSNP       endSNP  chrGene    startGene      endGene  genename  \\\n",
       "0     128507500.0  128522500.0     11.0  128522389.0  128522390.0   MIR6090   \n",
       "1     128582500.0  128597500.0     11.0  128587557.0  128587558.0      ETS1   \n",
       "2     128582500.0  128597500.0     11.0  128587557.0  128587558.0      ETS1   \n",
       "3     128587500.0  128602500.0     11.0  128587557.0  128587558.0      ETS1   \n",
       "4     128507500.0  128522500.0     11.0  128522389.0  128522390.0   MIR6090   \n",
       "...           ...          ...      ...          ...          ...       ...   \n",
       "8846  137537500.0  137552500.0      6.0  137544371.0  137544372.0  BTF3L4P3   \n",
       "8847  137537500.0  137552500.0      6.0  137544371.0  137544372.0  BTF3L4P3   \n",
       "8848  137537500.0  137552500.0      6.0  137544371.0  137544372.0  BTF3L4P3   \n",
       "8849  137537500.0  137552500.0      6.0  137544371.0  137544372.0  BTF3L4P3   \n",
       "8850  137537500.0  137552500.0      6.0  137544371.0  137544372.0  BTF3L4P3   \n",
       "\n",
       "               geneid  other  \n",
       "0     ENSG00000276176      +  \n",
       "1     ENSG00000134954      -  \n",
       "2     ENSG00000134954      -  \n",
       "3     ENSG00000134954      -  \n",
       "4     ENSG00000276176      +  \n",
       "...               ...    ...  \n",
       "8846  ENSG00000213108      -  \n",
       "8847  ENSG00000213108      -  \n",
       "8848  ENSG00000213108      -  \n",
       "8849  ENSG00000213108      -  \n",
       "8850  ENSG00000213108      -  \n",
       "\n",
       "[8460 rows x 40 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_hichip_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "66d8b2f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# find the unique SNPs\n",
    "uniq_snps_by_cells = gwas_hichip_genes.groupby('cline_loop').sid.nunique()\n",
    "uniq_snps_by_cells = uniq_snps_by_cells.to_frame()\n",
    "\n",
    "# find the unique genes \n",
    "uniq_genes_by_cells = gwas_hichip_genes.groupby('cline_loop').genename.nunique()\n",
    "uniq_genes_by_cells = uniq_genes_by_cells.to_frame()\n",
    "\n",
    "# merge snps and genes\n",
    "uniq_counts_by_cells = pd.merge(uniq_snps_by_cells, uniq_genes_by_cells, left_index=True, right_index=True)\n",
    "uniq_counts_by_cells.columns = ['Number of Unique SNPs', 'Number of Unique Genes']\n",
    "uniq_counts_by_cells.index.name = 'Cell Line'\n",
    "\n",
    "# save the file\n",
    "excel_analysis = os.path.join(outdir, 'Unique_Counts_By_Cell_Line.xlsx')\n",
    "uniq_counts_by_cells.to_excel(excel_analysis, sheet_name='finemapping')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "c57c71c9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genename</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cline_loop</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CD4N</th>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD8N</th>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CM</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB</th>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCM</th>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NK</th>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            genename\n",
       "cline_loop          \n",
       "CD4N              31\n",
       "CD8N              30\n",
       "CM                 7\n",
       "NB                38\n",
       "NCM               30\n",
       "NK                32"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uniq_genes_by_cells"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "121499ee",
   "metadata": {},
   "source": [
    "#### Write the gene list as well"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "d5a69fa5",
   "metadata": {},
   "outputs": [],
   "source": [
    "gh_list = gwas_hichip_genes.geneid.unique()\n",
    "gh_fn = os.path.join(outdir, 'gene_list.txt')\n",
    "with open(gh_fn, 'w') as fw:\n",
    "    for x in gh_list:\n",
    "        fw.write('{}\\n'.format(x))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9c05028a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ca640c34",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "a512ec64",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrSNP</th>\n",
       "      <th>startSNP</th>\n",
       "      <th>endSNP</th>\n",
       "      <th>sid</th>\n",
       "      <th>cline_loop</th>\n",
       "      <th>chrGene</th>\n",
       "      <th>startGene</th>\n",
       "      <th>endGene</th>\n",
       "      <th>genename</th>\n",
       "      <th>geneid</th>\n",
       "      <th>other</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11</td>\n",
       "      <td>128582500</td>\n",
       "      <td>128597500</td>\n",
       "      <td>chr11:128138867</td>\n",
       "      <td>NCM</td>\n",
       "      <td>11</td>\n",
       "      <td>128587557</td>\n",
       "      <td>128587558</td>\n",
       "      <td>ETS1</td>\n",
       "      <td>ENSG00000134954</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>11</td>\n",
       "      <td>128587500</td>\n",
       "      <td>128602500</td>\n",
       "      <td>chr11:128138867</td>\n",
       "      <td>NCM</td>\n",
       "      <td>11</td>\n",
       "      <td>128587557</td>\n",
       "      <td>128587558</td>\n",
       "      <td>ETS1</td>\n",
       "      <td>ENSG00000134954</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>11</td>\n",
       "      <td>128582500</td>\n",
       "      <td>128597500</td>\n",
       "      <td>chr11:128138867</td>\n",
       "      <td>NB</td>\n",
       "      <td>11</td>\n",
       "      <td>128587557</td>\n",
       "      <td>128587558</td>\n",
       "      <td>ETS1</td>\n",
       "      <td>ENSG00000134954</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>11</td>\n",
       "      <td>128582500</td>\n",
       "      <td>128597500</td>\n",
       "      <td>chr11:128138867</td>\n",
       "      <td>CM</td>\n",
       "      <td>11</td>\n",
       "      <td>128587557</td>\n",
       "      <td>128587558</td>\n",
       "      <td>ETS1</td>\n",
       "      <td>ENSG00000134954</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11</td>\n",
       "      <td>128582500</td>\n",
       "      <td>128597500</td>\n",
       "      <td>chr11:128131013</td>\n",
       "      <td>NB</td>\n",
       "      <td>11</td>\n",
       "      <td>128587557</td>\n",
       "      <td>128587558</td>\n",
       "      <td>ETS1</td>\n",
       "      <td>ENSG00000134954</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>706</th>\n",
       "      <td>6</td>\n",
       "      <td>137537500</td>\n",
       "      <td>137552500</td>\n",
       "      <td>chr6:137963367</td>\n",
       "      <td>CM</td>\n",
       "      <td>6</td>\n",
       "      <td>137544371</td>\n",
       "      <td>137544372</td>\n",
       "      <td>BTF3L4P3</td>\n",
       "      <td>ENSG00000213108</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>707</th>\n",
       "      <td>6</td>\n",
       "      <td>137537500</td>\n",
       "      <td>137552500</td>\n",
       "      <td>chr6:137963367</td>\n",
       "      <td>CM</td>\n",
       "      <td>6</td>\n",
       "      <td>137544371</td>\n",
       "      <td>137544372</td>\n",
       "      <td>BTF3L4P3</td>\n",
       "      <td>ENSG00000213108</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>708</th>\n",
       "      <td>6</td>\n",
       "      <td>137537500</td>\n",
       "      <td>137552500</td>\n",
       "      <td>chr6:137963070</td>\n",
       "      <td>NB</td>\n",
       "      <td>6</td>\n",
       "      <td>137544371</td>\n",
       "      <td>137544372</td>\n",
       "      <td>BTF3L4P3</td>\n",
       "      <td>ENSG00000213108</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>709</th>\n",
       "      <td>6</td>\n",
       "      <td>137537500</td>\n",
       "      <td>137552500</td>\n",
       "      <td>chr6:137963070</td>\n",
       "      <td>CM</td>\n",
       "      <td>6</td>\n",
       "      <td>137544371</td>\n",
       "      <td>137544372</td>\n",
       "      <td>BTF3L4P3</td>\n",
       "      <td>ENSG00000213108</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>710</th>\n",
       "      <td>6</td>\n",
       "      <td>137537500</td>\n",
       "      <td>137552500</td>\n",
       "      <td>chr6:137963070</td>\n",
       "      <td>CM</td>\n",
       "      <td>6</td>\n",
       "      <td>137544371</td>\n",
       "      <td>137544372</td>\n",
       "      <td>BTF3L4P3</td>\n",
       "      <td>ENSG00000213108</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>711 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrSNP   startSNP     endSNP              sid cline_loop  chrGene  \\\n",
       "0        11  128582500  128597500  chr11:128138867        NCM       11   \n",
       "1        11  128587500  128602500  chr11:128138867        NCM       11   \n",
       "2        11  128582500  128597500  chr11:128138867         NB       11   \n",
       "3        11  128582500  128597500  chr11:128138867         CM       11   \n",
       "4        11  128582500  128597500  chr11:128131013         NB       11   \n",
       "..      ...        ...        ...              ...        ...      ...   \n",
       "706       6  137537500  137552500   chr6:137963367         CM        6   \n",
       "707       6  137537500  137552500   chr6:137963367         CM        6   \n",
       "708       6  137537500  137552500   chr6:137963070         NB        6   \n",
       "709       6  137537500  137552500   chr6:137963070         CM        6   \n",
       "710       6  137537500  137552500   chr6:137963070         CM        6   \n",
       "\n",
       "     startGene    endGene  genename           geneid other  \n",
       "0    128587557  128587558      ETS1  ENSG00000134954     -  \n",
       "1    128587557  128587558      ETS1  ENSG00000134954     -  \n",
       "2    128587557  128587558      ETS1  ENSG00000134954     -  \n",
       "3    128587557  128587558      ETS1  ENSG00000134954     -  \n",
       "4    128587557  128587558      ETS1  ENSG00000134954     -  \n",
       "..         ...        ...       ...              ...   ...  \n",
       "706  137544371  137544372  BTF3L4P3  ENSG00000213108     -  \n",
       "707  137544371  137544372  BTF3L4P3  ENSG00000213108     -  \n",
       "708  137544371  137544372  BTF3L4P3  ENSG00000213108     -  \n",
       "709  137544371  137544372  BTF3L4P3  ENSG00000213108     -  \n",
       "710  137544371  137544372  BTF3L4P3  ENSG00000213108     -  \n",
       "\n",
       "[711 rows x 11 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_overlaps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bb2e9fa3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "754172cd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "da9def6f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5d3d9da5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "a87355e2",
   "metadata": {},
   "source": [
    "## Summarize the SNPs, Loops and Intersection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "822d1071",
   "metadata": {},
   "outputs": [],
   "source": [
    "total_gwas = gwas_df.shape[0]\n",
    "# # summarize the total number of GWAS loops per cell # DOESN't MAKE SENSE\n",
    "# cell_summary['total_gwas'] = gwas_hichip.groupby('cline_loop').nunique('sid')['chr_snp']\n",
    "# cell_summary['total_gwas'] = cell_summary['total_gwas'].to_frame()\n",
    "# cell_summary['total_gwas'].columns = ['Total GWAS SNPs']\n",
    "# cell_summary['total_gwas']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "1e91d3cd",
   "metadata": {},
   "outputs": [],
   "source": [
    "cell_summary = {}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6452a2b8",
   "metadata": {},
   "source": [
    "### Summarize the Number of Loops per Cell (pre-intersection)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "2b34797f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>total_hichip</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cline</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CD4N</th>\n",
       "      <td>114421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD8N</th>\n",
       "      <td>84599</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CM</th>\n",
       "      <td>84298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB</th>\n",
       "      <td>128288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCM</th>\n",
       "      <td>103342</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NK</th>\n",
       "      <td>129890</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       total_hichip\n",
       "cline              \n",
       "CD4N         114421\n",
       "CD8N          84599\n",
       "CM            84298\n",
       "NB           128288\n",
       "NCM          103342\n",
       "NK           129890"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cell_summary['total_loops'] = loop_df.groupby('cline').count()['startA'].to_frame()\n",
    "cell_summary['total_loops'].columns = ['total_hichip']\n",
    "cell_summary['total_loops']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "08ae84d7",
   "metadata": {},
   "source": [
    "### Summarize the Number of SNP-Loop (SL) Pairs per Cell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "1d116fc4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sl_pairs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NB</th>\n",
       "      <td>624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NK</th>\n",
       "      <td>452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD4N</th>\n",
       "      <td>317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD8N</th>\n",
       "      <td>304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCM</th>\n",
       "      <td>217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CM</th>\n",
       "      <td>119</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      sl_pairs\n",
       "NB         624\n",
       "NK         452\n",
       "CD4N       317\n",
       "CD8N       304\n",
       "NCM        217\n",
       "CM         119"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cell_summary['sl_pairs'] = gwas_hichip['cline_loop'].value_counts().to_frame()\n",
    "cell_summary['sl_pairs'].columns = ['sl_pairs']\n",
    "cell_summary['sl_pairs']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c468bc58",
   "metadata": {},
   "source": [
    "### Summarize the Number of Unique GWAS SNPs which Overlap a HiChIP Loop Cell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "42cc7348",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniq_gwas_in_slpairs</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cline_loop</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CD4N</th>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD8N</th>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CM</th>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB</th>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCM</th>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NK</th>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            uniq_gwas_in_slpairs\n",
       "cline_loop                      \n",
       "CD4N                          46\n",
       "CD8N                          39\n",
       "CM                            44\n",
       "NB                            74\n",
       "NCM                           44\n",
       "NK                            53"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cell_summary['uniq_gwas'] = gwas_hichip.groupby('cline_loop')['sid'].nunique().to_frame()\n",
    "cell_summary['uniq_gwas'].columns = ['uniq_gwas_in_slpairs']\n",
    "cell_summary['uniq_gwas']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eec14a11",
   "metadata": {},
   "source": [
    "### Summarize the Number of Loops with GWAS Overlaps (per cell)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "3a8f01e3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniq_loops_in_slpairs</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cline_loop</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CD4N</th>\n",
       "      <td>222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD8N</th>\n",
       "      <td>186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CM</th>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB</th>\n",
       "      <td>314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCM</th>\n",
       "      <td>122</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NK</th>\n",
       "      <td>289</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            uniq_loops_in_slpairs\n",
       "cline_loop                       \n",
       "CD4N                          222\n",
       "CD8N                          186\n",
       "CM                             71\n",
       "NB                            314\n",
       "NCM                           122\n",
       "NK                            289"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loop_cols = ['chrA_loop', 'startA_loop', 'endA_loop', 'chrB_loop', 'startB_loop', 'endB_loop']\n",
    "cell_summary['uniq_loops'] = gwas_hichip.groupby('cline_loop')['loop_id'].nunique().to_frame()\n",
    "cell_summary['uniq_loops'].columns = ['uniq_loops_in_slpairs']\n",
    "cell_summary['uniq_loops']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "091f4b50",
   "metadata": {},
   "outputs": [],
   "source": [
    "concat_list = [cell_summary['total_loops'], cell_summary['sl_pairs'],\n",
    "               cell_summary['uniq_gwas'], cell_summary['uniq_loops']]\n",
    "summary = pd.concat(concat_list, axis=1)\n",
    "summary['pct_uniq_gwas_in_slpairs'] = summary['uniq_gwas_in_slpairs'] / total_gwas * 100\n",
    "summary['pct_uniq_loops_in_slpairs'] = summary['uniq_loops_in_slpairs'] / summary['total_hichip'] * 100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "371814c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>total_hichip</th>\n",
       "      <th>sl_pairs</th>\n",
       "      <th>uniq_gwas_in_slpairs</th>\n",
       "      <th>uniq_loops_in_slpairs</th>\n",
       "      <th>pct_uniq_gwas_in_slpairs</th>\n",
       "      <th>pct_uniq_loops_in_slpairs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CD4N</th>\n",
       "      <td>114421</td>\n",
       "      <td>317</td>\n",
       "      <td>46</td>\n",
       "      <td>222</td>\n",
       "      <td>46.464646</td>\n",
       "      <td>0.194020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD8N</th>\n",
       "      <td>84599</td>\n",
       "      <td>304</td>\n",
       "      <td>39</td>\n",
       "      <td>186</td>\n",
       "      <td>39.393939</td>\n",
       "      <td>0.219861</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CM</th>\n",
       "      <td>84298</td>\n",
       "      <td>119</td>\n",
       "      <td>44</td>\n",
       "      <td>71</td>\n",
       "      <td>44.444444</td>\n",
       "      <td>0.084225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB</th>\n",
       "      <td>128288</td>\n",
       "      <td>624</td>\n",
       "      <td>74</td>\n",
       "      <td>314</td>\n",
       "      <td>74.747475</td>\n",
       "      <td>0.244762</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCM</th>\n",
       "      <td>103342</td>\n",
       "      <td>217</td>\n",
       "      <td>44</td>\n",
       "      <td>122</td>\n",
       "      <td>44.444444</td>\n",
       "      <td>0.118055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NK</th>\n",
       "      <td>129890</td>\n",
       "      <td>452</td>\n",
       "      <td>53</td>\n",
       "      <td>289</td>\n",
       "      <td>53.535354</td>\n",
       "      <td>0.222496</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      total_hichip  sl_pairs  uniq_gwas_in_slpairs  uniq_loops_in_slpairs  \\\n",
       "CD4N        114421       317                    46                    222   \n",
       "CD8N         84599       304                    39                    186   \n",
       "CM           84298       119                    44                     71   \n",
       "NB          128288       624                    74                    314   \n",
       "NCM         103342       217                    44                    122   \n",
       "NK          129890       452                    53                    289   \n",
       "\n",
       "      pct_uniq_gwas_in_slpairs  pct_uniq_loops_in_slpairs  \n",
       "CD4N                 46.464646                   0.194020  \n",
       "CD8N                 39.393939                   0.219861  \n",
       "CM                   44.444444                   0.084225  \n",
       "NB                   74.747475                   0.244762  \n",
       "NCM                  44.444444                   0.118055  \n",
       "NK                   53.535354                   0.222496  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "bf5dbdb0",
   "metadata": {},
   "outputs": [],
   "source": [
    "final_summary = summary.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "cbb2cb48",
   "metadata": {},
   "outputs": [],
   "source": [
    "final_colnames = ['Total\\\\nHiChIP Loops', \n",
    "                  'Number of\\\\nGWAS-Loop Pairs',\n",
    "                  'Number of\\\\nUnique GWAS SNPs in GL Pairs', \n",
    "                  'Number of\\\\nUnique loops in GL Pairs',\n",
    "                  'Percentage of\\\\nUnique GWAS SNPs in GL Pairs', \n",
    "                  'Percentage of\\\\nUnique loops in GL Pairs']\n",
    "final_colnames = ['Total HiChIP Loops', \n",
    "                  'Number of GWAS-Loop Pairs',\n",
    "                  'Number of Unique GWAS SNPs in GL Pairs', \n",
    "                  'Number of Unique loops in GL Pairs',\n",
    "                  'Percentage of Unique GWAS SNPs in GL Pairs', \n",
    "                  'Percentage of Unique loops in GL Pairs']\n",
    "final_summary.columns = final_colnames"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "a2180e2a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Total HiChIP Loops</th>\n",
       "      <th>Number of GWAS-Loop Pairs</th>\n",
       "      <th>Number of Unique GWAS SNPs in GL Pairs</th>\n",
       "      <th>Number of Unique loops in GL Pairs</th>\n",
       "      <th>Percentage of Unique GWAS SNPs in GL Pairs</th>\n",
       "      <th>Percentage of Unique loops in GL Pairs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CD4N</th>\n",
       "      <td>114421</td>\n",
       "      <td>317</td>\n",
       "      <td>46</td>\n",
       "      <td>222</td>\n",
       "      <td>46.464646</td>\n",
       "      <td>0.194020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD8N</th>\n",
       "      <td>84599</td>\n",
       "      <td>304</td>\n",
       "      <td>39</td>\n",
       "      <td>186</td>\n",
       "      <td>39.393939</td>\n",
       "      <td>0.219861</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CM</th>\n",
       "      <td>84298</td>\n",
       "      <td>119</td>\n",
       "      <td>44</td>\n",
       "      <td>71</td>\n",
       "      <td>44.444444</td>\n",
       "      <td>0.084225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB</th>\n",
       "      <td>128288</td>\n",
       "      <td>624</td>\n",
       "      <td>74</td>\n",
       "      <td>314</td>\n",
       "      <td>74.747475</td>\n",
       "      <td>0.244762</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCM</th>\n",
       "      <td>103342</td>\n",
       "      <td>217</td>\n",
       "      <td>44</td>\n",
       "      <td>122</td>\n",
       "      <td>44.444444</td>\n",
       "      <td>0.118055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NK</th>\n",
       "      <td>129890</td>\n",
       "      <td>452</td>\n",
       "      <td>53</td>\n",
       "      <td>289</td>\n",
       "      <td>53.535354</td>\n",
       "      <td>0.222496</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(HTML(final_summary.to_html().replace(\"\\\\n\",\"<br>\")))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "54b8ec78",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f8c6d9ca",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ab4085da",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f66bfe3d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eb4d6f5b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9578328e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "HiChIP DB",
   "language": "python",
   "name": "hichip-db"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
