{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 146,
   "id": "9a59a3f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# %load Find_SGLs.eQTL_Catalogue_Format.py\n",
    "#!/usr/bin/env python\n",
    "\n",
    "# %load ./scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops.py\n",
    "#!/usr/bin/env python\n",
    "\n",
    "# # Make the master table\n",
    "import os \n",
    "import sys\n",
    "import pybedtools as pbt\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import subprocess as sp\n",
    "import json\n",
    "pbt.helpers.set_tempdir('/mnt/BioHome/jreyna/tmp/')\n",
    "pbt.set_bedtools_path('/mnt/BioApps/bedtools/bin/')\n",
    "bgzip = '/mnt/BioApps/tabix/tabix-0.2.6/bgzip'\n",
    "tabix = '/mnt/BioApps/tabix/tabix-0.2.6/tabix'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "id": "f640a094",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('/mnt/BioAdHoc/Groups/vd-ay/jreyna/projects/dchallenge/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "id": "df9fda4a",
   "metadata": {},
   "outputs": [],
   "source": [
    "bedpe_6cols = ['chrA', 'startA', 'endA', 'chrB', 'startB', 'endB']\n",
    "bedpe_10cols = ['chrA', 'startA', 'endA', 'chrB', 'startB', 'endB', 'name', 'score', 'strand1', 'strand2']\n",
    "\n",
    "# ## default values for the command line\n",
    "sys.argv = [0] * 100\n",
    "# sys.argv[1] =  'results/main/sgls/T1D_34012112_Gaulton/Schmiedel_2018/monocyte_naive/eqtls.coloc_filtered.tsv.gz'\n",
    "# sys.argv[2] = 'results/main/coloc/Results/Colocalization_SMKN/T1D_34012112_Gaulton/Schmiedel_2018/monocyte_naive/'\n",
    "# sys.argv[2] += 'FINAL_Summary_Coloc_Gene_SNP_Pairs.bed'\n",
    "# sys.argv[3] = 'results/main/h3k27ac_hichip/monocyte_naive/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed'\n",
    "# sys.argv[4] = 'monocyte_naive'\n",
    "# sys.argv[5] = 'results/refs/hg38/hg38.chrom.sizes'\n",
    "# sys.argv[6] = 'results/refs/gencode/v30/gencode.v30.annotation.bed'\n",
    "# sys.argv[7] = 'results/main/sgls/T1D_34012112_Gaulton/Schmiedel_2018/monocyte_naive/monocyte_naive'\n",
    "\n",
    "sys.argv[1] = 'results/main/sgls/T1D_34012112_Gaulton/Quach_2016/monocyte_IAV/eqtls.coloc_filtered.tsv.gz'\n",
    "sys.argv[2] = 'results/main/coloc/Results/Colocalization_SMKN/T1D_34012112_Gaulton/Quach_2016/monocyte_IAV/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed'\n",
    "sys.argv[3] = 'results/main/h3k27ac_hichip/monocyte_naive/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.grch38.bed'\n",
    "sys.argv[4] = 'monocyte_IAV'\n",
    "sys.argv[5] = 'results/refs/hg38/hg38.chrom.sizes'\n",
    "sys.argv[6] = 'results/refs/gencode/v30/gencode.v30.annotation.bed'\n",
    "sys.argv[7] = 'results/main/sgls/T1D_34012112_Gaulton/Quach_2016/monocyte_IAV/monocyte_naive'\n",
    "\n",
    "\n",
    "#########################################################################################\n",
    "# parsing the commandline arguments\n",
    "#########################################################################################\n",
    "\n",
    "eqtl_fn = sys.argv[1]\n",
    "coloc_fn = sys.argv[2]\n",
    "loop_fn = sys.argv[3]\n",
    "celltype = sys.argv[4]\n",
    "gs_fn = sys.argv[5]\n",
    "genes_fn = sys.argv[6]\n",
    "outdir = sys.argv[7]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "id": "4ea951ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "#########################################################################################\n",
    "# Load the colocalization data\n",
    "#########################################################################################\n",
    "\n",
    "# load the colocalization data\n",
    "coloc = pd.read_table(coloc_fn)\n",
    "coloc['sid'] = coloc['chr'].str.replace('chr', '')  + ':' + coloc['pos'].astype(str)\n",
    "#coloc.rename(columns={'rs_id'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "111b015e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>pp_H0_Coloc_Summary</th>\n",
       "      <th>pp_H1_Coloc_Summary</th>\n",
       "      <th>pp_H2_Coloc_Summary</th>\n",
       "      <th>pp_H3_Coloc_Summary</th>\n",
       "      <th>pp_H4_Coloc_Summary</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>geneName</th>\n",
       "      <th>...</th>\n",
       "      <th>alt</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>slope_se_snp</th>\n",
       "      <th>slope_gwas</th>\n",
       "      <th>slope_se_gwas</th>\n",
       "      <th>pval_nominal</th>\n",
       "      <th>SampleSize</th>\n",
       "      <th>sid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>162368011</td>\n",
       "      <td>6.793362e-75</td>\n",
       "      <td>1.005820e-73</td>\n",
       "      <td>0.008265</td>\n",
       "      <td>0.121498</td>\n",
       "      <td>0.870237</td>\n",
       "      <td>rs984971</td>\n",
       "      <td>2:162368011:A:G</td>\n",
       "      <td>ENSG00000115263</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>2970</td>\n",
       "      <td>0.593051</td>\n",
       "      <td>5008</td>\n",
       "      <td>935884.206798</td>\n",
       "      <td>-0.118349</td>\n",
       "      <td>0.014721</td>\n",
       "      <td>9.030000e-16</td>\n",
       "      <td>520580</td>\n",
       "      <td>2:162368011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>162368011</td>\n",
       "      <td>6.793362e-75</td>\n",
       "      <td>1.005820e-73</td>\n",
       "      <td>0.008265</td>\n",
       "      <td>0.121498</td>\n",
       "      <td>0.870237</td>\n",
       "      <td>rs984971</td>\n",
       "      <td>2:162368011:A:G</td>\n",
       "      <td>ENSG00000197635</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>2970</td>\n",
       "      <td>0.593051</td>\n",
       "      <td>5008</td>\n",
       "      <td>0.388094</td>\n",
       "      <td>-0.118349</td>\n",
       "      <td>0.014721</td>\n",
       "      <td>9.030000e-16</td>\n",
       "      <td>520580</td>\n",
       "      <td>2:162368011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr2</td>\n",
       "      <td>162368011</td>\n",
       "      <td>6.793362e-75</td>\n",
       "      <td>1.005820e-73</td>\n",
       "      <td>0.008265</td>\n",
       "      <td>0.121498</td>\n",
       "      <td>0.870237</td>\n",
       "      <td>rs984971</td>\n",
       "      <td>2:162368011:A:G</td>\n",
       "      <td>ENSG00000115271</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>2970</td>\n",
       "      <td>0.593051</td>\n",
       "      <td>5008</td>\n",
       "      <td>3.428822</td>\n",
       "      <td>-0.118349</td>\n",
       "      <td>0.014721</td>\n",
       "      <td>9.030000e-16</td>\n",
       "      <td>520580</td>\n",
       "      <td>2:162368011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>162368011</td>\n",
       "      <td>6.793362e-75</td>\n",
       "      <td>1.005820e-73</td>\n",
       "      <td>0.008265</td>\n",
       "      <td>0.121498</td>\n",
       "      <td>0.870237</td>\n",
       "      <td>rs984971</td>\n",
       "      <td>2:162368011:A:G</td>\n",
       "      <td>ENSG00000236841</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>2970</td>\n",
       "      <td>0.593051</td>\n",
       "      <td>5008</td>\n",
       "      <td>5594.125846</td>\n",
       "      <td>-0.118349</td>\n",
       "      <td>0.014721</td>\n",
       "      <td>9.030000e-16</td>\n",
       "      <td>520580</td>\n",
       "      <td>2:162368011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr2</td>\n",
       "      <td>162368011</td>\n",
       "      <td>6.793362e-75</td>\n",
       "      <td>1.005820e-73</td>\n",
       "      <td>0.008265</td>\n",
       "      <td>0.121498</td>\n",
       "      <td>0.870237</td>\n",
       "      <td>rs984971</td>\n",
       "      <td>2:162368011:A:G</td>\n",
       "      <td>ENSG00000184611</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>2970</td>\n",
       "      <td>0.593051</td>\n",
       "      <td>5008</td>\n",
       "      <td>0.460995</td>\n",
       "      <td>-0.118349</td>\n",
       "      <td>0.014721</td>\n",
       "      <td>9.030000e-16</td>\n",
       "      <td>520580</td>\n",
       "      <td>2:162368011</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr        pos  pp_H0_Coloc_Summary  pp_H1_Coloc_Summary  \\\n",
       "0  chr2  162368011         6.793362e-75         1.005820e-73   \n",
       "1  chr2  162368011         6.793362e-75         1.005820e-73   \n",
       "2  chr2  162368011         6.793362e-75         1.005820e-73   \n",
       "3  chr2  162368011         6.793362e-75         1.005820e-73   \n",
       "4  chr2  162368011         6.793362e-75         1.005820e-73   \n",
       "\n",
       "   pp_H2_Coloc_Summary  pp_H3_Coloc_Summary  pp_H4_Coloc_Summary     rs_id  \\\n",
       "0             0.008265             0.121498             0.870237  rs984971   \n",
       "1             0.008265             0.121498             0.870237  rs984971   \n",
       "2             0.008265             0.121498             0.870237  rs984971   \n",
       "3             0.008265             0.121498             0.870237  rs984971   \n",
       "4             0.008265             0.121498             0.870237  rs984971   \n",
       "\n",
       "        variant_id         geneName  ...  alt    AC        AF    AN  \\\n",
       "0  2:162368011:A:G  ENSG00000115263  ...    G  2970  0.593051  5008   \n",
       "1  2:162368011:A:G  ENSG00000197635  ...    G  2970  0.593051  5008   \n",
       "2  2:162368011:A:G  ENSG00000115271  ...    G  2970  0.593051  5008   \n",
       "3  2:162368011:A:G  ENSG00000236841  ...    G  2970  0.593051  5008   \n",
       "4  2:162368011:A:G  ENSG00000184611  ...    G  2970  0.593051  5008   \n",
       "\n",
       "    slope_se_snp slope_gwas  slope_se_gwas  pval_nominal  SampleSize  \\\n",
       "0  935884.206798  -0.118349       0.014721  9.030000e-16      520580   \n",
       "1       0.388094  -0.118349       0.014721  9.030000e-16      520580   \n",
       "2       3.428822  -0.118349       0.014721  9.030000e-16      520580   \n",
       "3    5594.125846  -0.118349       0.014721  9.030000e-16      520580   \n",
       "4       0.460995  -0.118349       0.014721  9.030000e-16      520580   \n",
       "\n",
       "           sid  \n",
       "0  2:162368011  \n",
       "1  2:162368011  \n",
       "2  2:162368011  \n",
       "3  2:162368011  \n",
       "4  2:162368011  \n",
       "\n",
       "[5 rows x 25 columns]"
      ]
     },
     "execution_count": 190,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coloc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "id": "fb7ef519",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(32, 25)"
      ]
     },
     "execution_count": 191,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coloc.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "id": "332f67a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# extract the most significant according the H4 \n",
    "coloc_sgpairs = coloc[coloc['pp_H4_Coloc_Summary'] > 0.75]\n",
    "coloc_sgpairs = coloc_sgpairs.loc[~coloc_sgpairs.duplicated(subset=['sid', 'geneName']),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "id": "b4554d24",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>pp_H0_Coloc_Summary</th>\n",
       "      <th>pp_H1_Coloc_Summary</th>\n",
       "      <th>pp_H2_Coloc_Summary</th>\n",
       "      <th>pp_H3_Coloc_Summary</th>\n",
       "      <th>pp_H4_Coloc_Summary</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>geneName</th>\n",
       "      <th>...</th>\n",
       "      <th>alt</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>slope_se_snp</th>\n",
       "      <th>slope_gwas</th>\n",
       "      <th>slope_se_gwas</th>\n",
       "      <th>pval_nominal</th>\n",
       "      <th>SampleSize</th>\n",
       "      <th>sid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>162368011</td>\n",
       "      <td>6.793362e-75</td>\n",
       "      <td>1.005820e-73</td>\n",
       "      <td>0.008265</td>\n",
       "      <td>0.121498</td>\n",
       "      <td>0.870237</td>\n",
       "      <td>rs984971</td>\n",
       "      <td>2:162368011:A:G</td>\n",
       "      <td>ENSG00000115263</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>2970</td>\n",
       "      <td>0.593051</td>\n",
       "      <td>5008</td>\n",
       "      <td>935884.206798</td>\n",
       "      <td>-0.118349</td>\n",
       "      <td>0.014721</td>\n",
       "      <td>9.030000e-16</td>\n",
       "      <td>520580</td>\n",
       "      <td>2:162368011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>162368011</td>\n",
       "      <td>6.793362e-75</td>\n",
       "      <td>1.005820e-73</td>\n",
       "      <td>0.008265</td>\n",
       "      <td>0.121498</td>\n",
       "      <td>0.870237</td>\n",
       "      <td>rs984971</td>\n",
       "      <td>2:162368011:A:G</td>\n",
       "      <td>ENSG00000197635</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>2970</td>\n",
       "      <td>0.593051</td>\n",
       "      <td>5008</td>\n",
       "      <td>0.388094</td>\n",
       "      <td>-0.118349</td>\n",
       "      <td>0.014721</td>\n",
       "      <td>9.030000e-16</td>\n",
       "      <td>520580</td>\n",
       "      <td>2:162368011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr2</td>\n",
       "      <td>162368011</td>\n",
       "      <td>6.793362e-75</td>\n",
       "      <td>1.005820e-73</td>\n",
       "      <td>0.008265</td>\n",
       "      <td>0.121498</td>\n",
       "      <td>0.870237</td>\n",
       "      <td>rs984971</td>\n",
       "      <td>2:162368011:A:G</td>\n",
       "      <td>ENSG00000115271</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>2970</td>\n",
       "      <td>0.593051</td>\n",
       "      <td>5008</td>\n",
       "      <td>3.428822</td>\n",
       "      <td>-0.118349</td>\n",
       "      <td>0.014721</td>\n",
       "      <td>9.030000e-16</td>\n",
       "      <td>520580</td>\n",
       "      <td>2:162368011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>162368011</td>\n",
       "      <td>6.793362e-75</td>\n",
       "      <td>1.005820e-73</td>\n",
       "      <td>0.008265</td>\n",
       "      <td>0.121498</td>\n",
       "      <td>0.870237</td>\n",
       "      <td>rs984971</td>\n",
       "      <td>2:162368011:A:G</td>\n",
       "      <td>ENSG00000236841</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>2970</td>\n",
       "      <td>0.593051</td>\n",
       "      <td>5008</td>\n",
       "      <td>5594.125846</td>\n",
       "      <td>-0.118349</td>\n",
       "      <td>0.014721</td>\n",
       "      <td>9.030000e-16</td>\n",
       "      <td>520580</td>\n",
       "      <td>2:162368011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr2</td>\n",
       "      <td>162368011</td>\n",
       "      <td>6.793362e-75</td>\n",
       "      <td>1.005820e-73</td>\n",
       "      <td>0.008265</td>\n",
       "      <td>0.121498</td>\n",
       "      <td>0.870237</td>\n",
       "      <td>rs984971</td>\n",
       "      <td>2:162368011:A:G</td>\n",
       "      <td>ENSG00000184611</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>2970</td>\n",
       "      <td>0.593051</td>\n",
       "      <td>5008</td>\n",
       "      <td>0.460995</td>\n",
       "      <td>-0.118349</td>\n",
       "      <td>0.014721</td>\n",
       "      <td>9.030000e-16</td>\n",
       "      <td>520580</td>\n",
       "      <td>2:162368011</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr        pos  pp_H0_Coloc_Summary  pp_H1_Coloc_Summary  \\\n",
       "0  chr2  162368011         6.793362e-75         1.005820e-73   \n",
       "1  chr2  162368011         6.793362e-75         1.005820e-73   \n",
       "2  chr2  162368011         6.793362e-75         1.005820e-73   \n",
       "3  chr2  162368011         6.793362e-75         1.005820e-73   \n",
       "4  chr2  162368011         6.793362e-75         1.005820e-73   \n",
       "\n",
       "   pp_H2_Coloc_Summary  pp_H3_Coloc_Summary  pp_H4_Coloc_Summary     rs_id  \\\n",
       "0             0.008265             0.121498             0.870237  rs984971   \n",
       "1             0.008265             0.121498             0.870237  rs984971   \n",
       "2             0.008265             0.121498             0.870237  rs984971   \n",
       "3             0.008265             0.121498             0.870237  rs984971   \n",
       "4             0.008265             0.121498             0.870237  rs984971   \n",
       "\n",
       "        variant_id         geneName  ...  alt    AC        AF    AN  \\\n",
       "0  2:162368011:A:G  ENSG00000115263  ...    G  2970  0.593051  5008   \n",
       "1  2:162368011:A:G  ENSG00000197635  ...    G  2970  0.593051  5008   \n",
       "2  2:162368011:A:G  ENSG00000115271  ...    G  2970  0.593051  5008   \n",
       "3  2:162368011:A:G  ENSG00000236841  ...    G  2970  0.593051  5008   \n",
       "4  2:162368011:A:G  ENSG00000184611  ...    G  2970  0.593051  5008   \n",
       "\n",
       "    slope_se_snp slope_gwas  slope_se_gwas  pval_nominal  SampleSize  \\\n",
       "0  935884.206798  -0.118349       0.014721  9.030000e-16      520580   \n",
       "1       0.388094  -0.118349       0.014721  9.030000e-16      520580   \n",
       "2       3.428822  -0.118349       0.014721  9.030000e-16      520580   \n",
       "3    5594.125846  -0.118349       0.014721  9.030000e-16      520580   \n",
       "4       0.460995  -0.118349       0.014721  9.030000e-16      520580   \n",
       "\n",
       "           sid  \n",
       "0  2:162368011  \n",
       "1  2:162368011  \n",
       "2  2:162368011  \n",
       "3  2:162368011  \n",
       "4  2:162368011  \n",
       "\n",
       "[5 rows x 25 columns]"
      ]
     },
     "execution_count": 193,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coloc_sgpairs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "id": "fe5abd73",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 22 colocalized SNP-gene pairs\n",
      "There are 2 colocalized SNPs\n",
      "There are 22 colocalized genes\n"
     ]
    }
   ],
   "source": [
    "coloc_sig_full = coloc_sgpairs.copy(deep=True)\n",
    "coloc_sgpairs.rename(columns={'pos': 'end'}, inplace=True)\n",
    "coloc_sgpairs.loc[:, 'start'] = coloc_sgpairs.loc[:, 'end'] - 1\n",
    "\n",
    "print('There are {} colocalized SNP-gene pairs'.format(coloc_sgpairs.shape[0]))\n",
    "print('There are {} colocalized SNPs'.format(coloc_sgpairs.sid.nunique()))\n",
    "print('There are {} colocalized genes'.format(coloc_sgpairs.geneName.nunique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "id": "3da6d585",
   "metadata": {},
   "outputs": [],
   "source": [
    "coloc_snps = coloc_sgpairs[['chr', 'start', 'end', 'rs_id', 'variant_id', 'sid']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "id": "d0fcd075",
   "metadata": {},
   "outputs": [],
   "source": [
    "coloc_snps = coloc_snps.loc[~coloc_snps.duplicated(subset='sid'), :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "id": "e222e306",
   "metadata": {},
   "outputs": [],
   "source": [
    "coloc_snps_pbt = pbt.BedTool.from_dataframe(coloc_snps.iloc[:, [0,1,2,5]]).sort()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "id": "f6e536aa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Load the gene data\n",
      "# make a genes pbt for intersection\n",
      "   chrom  start    end        gname          gene_id strand\n",
      "0   chr1  11868  11869      DDX11L1  ENSG00000223972      +\n",
      "12  chr1  29569  29570       WASH7P  ENSG00000227232      -\n",
      "25  chr1  17435  17436    MIR6859-1  ENSG00000278267      -\n",
      "28  chr1  29553  29554  MIR1302-2HG  ENSG00000243485      +\n",
      "36  chr1  30365  30366    MIR1302-2  ENSG00000284332      +\n",
      "There are 58825 genes in this GTF-derived file.\n"
     ]
    }
   ],
   "source": [
    "#########################################################################################\n",
    "# ## Load the gene data\n",
    "#########################################################################################\n",
    "print('# Load the gene data')\n",
    "\n",
    "# load the gencode coords\n",
    "cols = ['chrom', 'start', 'end', 'strand', 'type', 'gene_id', 'gname']\n",
    "gencode = pd.read_table(genes_fn, header=None, names=cols)\n",
    "\n",
    "# extract just the genes\n",
    "genes_df = gencode.loc[gencode.type.isin(['gene'])]\n",
    "genes_df = genes_df.loc[~genes_df.duplicated(subset='gene_id'), :]\n",
    "genes_df.loc[:, 'chrom'] = genes_df['chrom'].astype(str)\n",
    "genes_df = genes_df.iloc[:, [0,1,2,6,5,3]]\n",
    "\n",
    "# create a copy of the original gene bed before coordinate shrinking\n",
    "orig_genes_df = genes_df.copy()\n",
    "\n",
    "# convert the start/end position into start/end for the TSS\n",
    "# if the gene is + then the start is uses as the tss otherwise\n",
    "# the end is used as the tss\n",
    "genes_df.loc[(genes_df.strand == '+'), 'end'] = genes_df.loc[(genes_df.strand == '+'), 'start']\n",
    "genes_df.loc[(genes_df.strand == '+'), 'start'] = genes_df.loc[(genes_df.strand == '+'), 'start'] - 1\n",
    "genes_df.loc[(genes_df.strand == '-'), 'end'] = genes_df.loc[(genes_df.strand == '-'), 'end']\n",
    "genes_df.loc[(genes_df.strand == '-'), 'start'] = genes_df.loc[(genes_df.strand == '-'), 'end'] - 1\n",
    "\n",
    "# make a genes pbt for intersection\n",
    "print(\"# make a genes pbt for intersection\")\n",
    "print(genes_df.head())\n",
    "genes_pbt = pbt.BedTool.from_dataframe(genes_df).sort()\n",
    "\n",
    "print('There are {} genes in this GTF-derived file.'.format(genes_df.shape[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "id": "860cf6a3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Find all genes +/- 500kb\n"
     ]
    }
   ],
   "source": [
    "#########################################################################################\n",
    "# Find all genes +/- 500kb of a colocalized SNP\n",
    "#########################################################################################\n",
    "print(\"# Find all genes +/- 500kb\")\n",
    "\n",
    "# get a list of gene names within +- 500kb of the SNPs\n",
    "fivekb_genes = coloc_snps_pbt.slop(b=500000, g=gs_fn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "id": "488920cf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>161868010</td>\n",
       "      <td>162868011</td>\n",
       "      <td>2:162368011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr9</td>\n",
       "      <td>3796429</td>\n",
       "      <td>4796430</td>\n",
       "      <td>9:4296430</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom      start        end         name\n",
       "0  chr2  161868010  162868011  2:162368011\n",
       "1  chr9    3796429    4796430    9:4296430"
      ]
     },
     "execution_count": 200,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fivekb_genes.to_dataframe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "id": "1e5892ac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 30 colocalized snp-gene pairs within +/- 500kb.\n"
     ]
    }
   ],
   "source": [
    "fivekb_genes = fivekb_genes.intersect(genes_pbt, wa=True, wb=True)\n",
    "fivekb_genes = fivekb_genes.to_dataframe()\n",
    "fivekb_genes = fivekb_genes.iloc[:, [0,1,2,4,5,6,3,7,8,9]]\n",
    "\n",
    "fivekb_genes.columns = bedpe_6cols + ['rs_id', 'gname', 'gid', 'strand2']\n",
    "fivekb_genes['strand1'] = '+'\n",
    "fivekb_genes['name'] = fivekb_genes['rs_id'] + '_' + fivekb_genes['gname']\n",
    "fivekb_genes['score'] = '.'\n",
    "new_order = bedpe_10cols + ['rs_id', 'gname', 'gid']\n",
    "fivekb_genes = fivekb_genes[new_order]\n",
    "\n",
    "fivekb_genes['startA'] += 500000\n",
    "fivekb_genes['endA'] -= 500000\n",
    "fivekb_genes['sid'] = fivekb_genes['chrA'].str.replace('chr', '') + ':' + fivekb_genes['endA'].astype(str)\n",
    "\n",
    "print('There are {} colocalized snp-gene pairs within +/- 500kb.'.format(fivekb_genes.shape[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "id": "47f3f097",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Find the closest gene\n"
     ]
    }
   ],
   "source": [
    "#########################################################################################\n",
    "# Find the closest gene\n",
    "#########################################################################################\n",
    "print('# Find the closest gene')\n",
    "\n",
    "closest_gene = coloc_sig_pbt.closest(genes_pbt, d=True)\n",
    "closest_gene = closest_gene.to_dataframe().iloc[:, [0,1,2,4,5,6,3,7,8,9]]\n",
    "closest_gene.columns = bedpe_6cols + ['rs_id', 'gname', 'gid', 'dist']\n",
    "closest_gene['sid'] = closest_gene['chrA'].str.replace('chr', '') + ':' + closest_gene['endA'].astype(str)\n",
    "closest_gene.set_index(['sid', 'gname'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "id": "80ac79e2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr1</th>\n",
       "      <th>s1</th>\n",
       "      <th>e1</th>\n",
       "      <th>chr2</th>\n",
       "      <th>s2</th>\n",
       "      <th>e2</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand1</th>\n",
       "      <th>strand2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999620</td>\n",
       "      <td>1004620</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1019620</td>\n",
       "      <td>1024620</td>\n",
       "      <td>.</td>\n",
       "      <td>4.851310e-06</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1778561</td>\n",
       "      <td>1783561</td>\n",
       "      <td>chr1</td>\n",
       "      <td>3793436</td>\n",
       "      <td>3798436</td>\n",
       "      <td>.</td>\n",
       "      <td>4.646481e-08</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26278509</td>\n",
       "      <td>26283509</td>\n",
       "      <td>chr1</td>\n",
       "      <td>26318509</td>\n",
       "      <td>26323509</td>\n",
       "      <td>.</td>\n",
       "      <td>3.598136e-06</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>44934552</td>\n",
       "      <td>44939552</td>\n",
       "      <td>chr10</td>\n",
       "      <td>44954552</td>\n",
       "      <td>44959552</td>\n",
       "      <td>.</td>\n",
       "      <td>5.488571e-06</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10</td>\n",
       "      <td>44944552</td>\n",
       "      <td>44949552</td>\n",
       "      <td>chr10</td>\n",
       "      <td>44959552</td>\n",
       "      <td>44964552</td>\n",
       "      <td>.</td>\n",
       "      <td>5.173212e-06</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84137</th>\n",
       "      <td>chr10</td>\n",
       "      <td>43434552</td>\n",
       "      <td>43439552</td>\n",
       "      <td>chr10</td>\n",
       "      <td>45379552</td>\n",
       "      <td>45384552</td>\n",
       "      <td>.</td>\n",
       "      <td>4.792100e-08</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84138</th>\n",
       "      <td>chr10</td>\n",
       "      <td>43434552</td>\n",
       "      <td>43439552</td>\n",
       "      <td>chr10</td>\n",
       "      <td>45714552</td>\n",
       "      <td>45719552</td>\n",
       "      <td>.</td>\n",
       "      <td>4.075525e-08</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84139</th>\n",
       "      <td>chr10</td>\n",
       "      <td>43434552</td>\n",
       "      <td>43439552</td>\n",
       "      <td>chr10</td>\n",
       "      <td>44954552</td>\n",
       "      <td>44959552</td>\n",
       "      <td>.</td>\n",
       "      <td>4.851310e-06</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84140</th>\n",
       "      <td>chr10</td>\n",
       "      <td>44934552</td>\n",
       "      <td>44939552</td>\n",
       "      <td>chr10</td>\n",
       "      <td>44959552</td>\n",
       "      <td>44964552</td>\n",
       "      <td>.</td>\n",
       "      <td>4.528059e-06</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84141</th>\n",
       "      <td>chr10</td>\n",
       "      <td>44934552</td>\n",
       "      <td>44939552</td>\n",
       "      <td>chr10</td>\n",
       "      <td>44964552</td>\n",
       "      <td>44969552</td>\n",
       "      <td>.</td>\n",
       "      <td>4.208650e-06</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>84142 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chr1        s1        e1   chr2        s2        e2 name  \\\n",
       "0       chr1    999620   1004620   chr1   1019620   1024620    .   \n",
       "1       chr1   1778561   1783561   chr1   3793436   3798436    .   \n",
       "2       chr1  26278509  26283509   chr1  26318509  26323509    .   \n",
       "3      chr10  44934552  44939552  chr10  44954552  44959552    .   \n",
       "4      chr10  44944552  44949552  chr10  44959552  44964552    .   \n",
       "...      ...       ...       ...    ...       ...       ...  ...   \n",
       "84137  chr10  43434552  43439552  chr10  45379552  45384552    .   \n",
       "84138  chr10  43434552  43439552  chr10  45714552  45719552    .   \n",
       "84139  chr10  43434552  43439552  chr10  44954552  44959552    .   \n",
       "84140  chr10  44934552  44939552  chr10  44959552  44964552    .   \n",
       "84141  chr10  44934552  44939552  chr10  44964552  44969552    .   \n",
       "\n",
       "              score strand1 strand2  \n",
       "0      4.851310e-06       .       .  \n",
       "1      4.646481e-08       .       .  \n",
       "2      3.598136e-06       .       .  \n",
       "3      5.488571e-06       .       .  \n",
       "4      5.173212e-06       .       .  \n",
       "...             ...     ...     ...  \n",
       "84137  4.792100e-08       .       .  \n",
       "84138  4.075525e-08       .       .  \n",
       "84139  4.851310e-06       .       .  \n",
       "84140  4.528059e-06       .       .  \n",
       "84141  4.208650e-06       .       .  \n",
       "\n",
       "[84142 rows x 10 columns]"
      ]
     },
     "execution_count": 203,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_loops"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "52575bf2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fe1309a0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "id": "e68e4ffc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Get the loops\n"
     ]
    }
   ],
   "source": [
    "#########################################################################################\n",
    "# Load the H3K27ac HiChIP loops\n",
    "#########################################################################################\n",
    "print('# Get the loops')\n",
    "\n",
    "# load the loop data\n",
    "loops = pd.read_table(loop_fn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "id": "fa030d41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr1</th>\n",
       "      <th>s1</th>\n",
       "      <th>e1</th>\n",
       "      <th>chr2</th>\n",
       "      <th>s2</th>\n",
       "      <th>e2</th>\n",
       "      <th>cc</th>\n",
       "      <th>Coverage1</th>\n",
       "      <th>isPeak1</th>\n",
       "      <th>Bias1</th>\n",
       "      <th>...</th>\n",
       "      <th>Bias2</th>\n",
       "      <th>Mapp2</th>\n",
       "      <th>GCContent2</th>\n",
       "      <th>RESites2</th>\n",
       "      <th>p</th>\n",
       "      <th>exp_cc_Bias</th>\n",
       "      <th>p_Bias</th>\n",
       "      <th>dbinom_Bias</th>\n",
       "      <th>P-Value_Bias</th>\n",
       "      <th>Q-Value_Bias</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999620</td>\n",
       "      <td>1004620</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1019620</td>\n",
       "      <td>1024620</td>\n",
       "      <td>31</td>\n",
       "      <td>590</td>\n",
       "      <td>1</td>\n",
       "      <td>0.442683</td>\n",
       "      <td>...</td>\n",
       "      <td>0.354897</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.851310e-06</td>\n",
       "      <td>9.241261</td>\n",
       "      <td>8.645438e-08</td>\n",
       "      <td>1.021450e-08</td>\n",
       "      <td>1.429445e-08</td>\n",
       "      <td>3.577668e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1778561</td>\n",
       "      <td>1783561</td>\n",
       "      <td>chr1</td>\n",
       "      <td>3793436</td>\n",
       "      <td>3798436</td>\n",
       "      <td>10</td>\n",
       "      <td>10203</td>\n",
       "      <td>1</td>\n",
       "      <td>7.655420</td>\n",
       "      <td>...</td>\n",
       "      <td>0.282117</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.646481e-08</td>\n",
       "      <td>1.985142</td>\n",
       "      <td>1.857152e-08</td>\n",
       "      <td>3.597630e-05</td>\n",
       "      <td>4.373368e-05</td>\n",
       "      <td>5.051998e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26278509</td>\n",
       "      <td>26283509</td>\n",
       "      <td>chr1</td>\n",
       "      <td>26318509</td>\n",
       "      <td>26323509</td>\n",
       "      <td>517</td>\n",
       "      <td>12458</td>\n",
       "      <td>1</td>\n",
       "      <td>9.347371</td>\n",
       "      <td>...</td>\n",
       "      <td>3.311872</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.598136e-06</td>\n",
       "      <td>392.886045</td>\n",
       "      <td>3.675550e-06</td>\n",
       "      <td>3.215832e-10</td>\n",
       "      <td>1.307941e-09</td>\n",
       "      <td>3.909070e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>44934552</td>\n",
       "      <td>44939552</td>\n",
       "      <td>chr10</td>\n",
       "      <td>44954552</td>\n",
       "      <td>44959552</td>\n",
       "      <td>124</td>\n",
       "      <td>658</td>\n",
       "      <td>1</td>\n",
       "      <td>0.493704</td>\n",
       "      <td>...</td>\n",
       "      <td>1.306291</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.488571e-06</td>\n",
       "      <td>31.570973</td>\n",
       "      <td>2.953546e-07</td>\n",
       "      <td>1.053721e-35</td>\n",
       "      <td>1.408527e-35</td>\n",
       "      <td>4.933339e-32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10</td>\n",
       "      <td>44944552</td>\n",
       "      <td>44949552</td>\n",
       "      <td>chr10</td>\n",
       "      <td>44959552</td>\n",
       "      <td>44964552</td>\n",
       "      <td>73</td>\n",
       "      <td>658</td>\n",
       "      <td>1</td>\n",
       "      <td>0.493704</td>\n",
       "      <td>...</td>\n",
       "      <td>0.741307</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.173212e-06</td>\n",
       "      <td>19.034114</td>\n",
       "      <td>1.780690e-07</td>\n",
       "      <td>3.084248e-21</td>\n",
       "      <td>4.145798e-21</td>\n",
       "      <td>4.803915e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84137</th>\n",
       "      <td>chr10</td>\n",
       "      <td>43434552</td>\n",
       "      <td>43439552</td>\n",
       "      <td>chr10</td>\n",
       "      <td>45379552</td>\n",
       "      <td>45384552</td>\n",
       "      <td>12</td>\n",
       "      <td>8964</td>\n",
       "      <td>1</td>\n",
       "      <td>6.725785</td>\n",
       "      <td>...</td>\n",
       "      <td>2.393491</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.792100e-08</td>\n",
       "      <td>2.755553</td>\n",
       "      <td>2.577891e-08</td>\n",
       "      <td>2.543585e-05</td>\n",
       "      <td>3.212350e-05</td>\n",
       "      <td>3.852477e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84138</th>\n",
       "      <td>chr10</td>\n",
       "      <td>43434552</td>\n",
       "      <td>43439552</td>\n",
       "      <td>chr10</td>\n",
       "      <td>45714552</td>\n",
       "      <td>45719552</td>\n",
       "      <td>11</td>\n",
       "      <td>8964</td>\n",
       "      <td>1</td>\n",
       "      <td>6.725785</td>\n",
       "      <td>...</td>\n",
       "      <td>0.951779</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.075525e-08</td>\n",
       "      <td>2.239503</td>\n",
       "      <td>2.095113e-08</td>\n",
       "      <td>1.896306e-05</td>\n",
       "      <td>2.322613e-05</td>\n",
       "      <td>2.893782e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84139</th>\n",
       "      <td>chr10</td>\n",
       "      <td>43434552</td>\n",
       "      <td>43439552</td>\n",
       "      <td>chr10</td>\n",
       "      <td>44954552</td>\n",
       "      <td>44959552</td>\n",
       "      <td>96</td>\n",
       "      <td>476</td>\n",
       "      <td>1</td>\n",
       "      <td>0.357148</td>\n",
       "      <td>...</td>\n",
       "      <td>1.306291</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.851310e-06</td>\n",
       "      <td>21.079813</td>\n",
       "      <td>1.972071e-07</td>\n",
       "      <td>8.709230e-33</td>\n",
       "      <td>1.111877e-32</td>\n",
       "      <td>3.184313e-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84140</th>\n",
       "      <td>chr10</td>\n",
       "      <td>44934552</td>\n",
       "      <td>44939552</td>\n",
       "      <td>chr10</td>\n",
       "      <td>44959552</td>\n",
       "      <td>44964552</td>\n",
       "      <td>36</td>\n",
       "      <td>476</td>\n",
       "      <td>1</td>\n",
       "      <td>0.357148</td>\n",
       "      <td>...</td>\n",
       "      <td>0.741307</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.528059e-06</td>\n",
       "      <td>12.629390</td>\n",
       "      <td>1.181512e-07</td>\n",
       "      <td>3.929659e-08</td>\n",
       "      <td>5.926719e-08</td>\n",
       "      <td>1.321116e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84141</th>\n",
       "      <td>chr10</td>\n",
       "      <td>44934552</td>\n",
       "      <td>44939552</td>\n",
       "      <td>chr10</td>\n",
       "      <td>44964552</td>\n",
       "      <td>44969552</td>\n",
       "      <td>39</td>\n",
       "      <td>476</td>\n",
       "      <td>1</td>\n",
       "      <td>0.357148</td>\n",
       "      <td>...</td>\n",
       "      <td>0.791578</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.208650e-06</td>\n",
       "      <td>12.235101</td>\n",
       "      <td>1.144625e-07</td>\n",
       "      <td>6.215428e-10</td>\n",
       "      <td>8.914132e-10</td>\n",
       "      <td>2.732105e-07</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>84142 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chr1        s1        e1   chr2        s2        e2   cc  Coverage1  \\\n",
       "0       chr1    999620   1004620   chr1   1019620   1024620   31        590   \n",
       "1       chr1   1778561   1783561   chr1   3793436   3798436   10      10203   \n",
       "2       chr1  26278509  26283509   chr1  26318509  26323509  517      12458   \n",
       "3      chr10  44934552  44939552  chr10  44954552  44959552  124        658   \n",
       "4      chr10  44944552  44949552  chr10  44959552  44964552   73        658   \n",
       "...      ...       ...       ...    ...       ...       ...  ...        ...   \n",
       "84137  chr10  43434552  43439552  chr10  45379552  45384552   12       8964   \n",
       "84138  chr10  43434552  43439552  chr10  45714552  45719552   11       8964   \n",
       "84139  chr10  43434552  43439552  chr10  44954552  44959552   96        476   \n",
       "84140  chr10  44934552  44939552  chr10  44959552  44964552   36        476   \n",
       "84141  chr10  44934552  44939552  chr10  44964552  44969552   39        476   \n",
       "\n",
       "       isPeak1     Bias1  ...     Bias2  Mapp2  GCContent2  RESites2  \\\n",
       "0            1  0.442683  ...  0.354897      0           0         0   \n",
       "1            1  7.655420  ...  0.282117      0           0         0   \n",
       "2            1  9.347371  ...  3.311872      0           0         0   \n",
       "3            1  0.493704  ...  1.306291      0           0         0   \n",
       "4            1  0.493704  ...  0.741307      0           0         0   \n",
       "...        ...       ...  ...       ...    ...         ...       ...   \n",
       "84137        1  6.725785  ...  2.393491      0           0         0   \n",
       "84138        1  6.725785  ...  0.951779      0           0         0   \n",
       "84139        1  0.357148  ...  1.306291      0           0         0   \n",
       "84140        1  0.357148  ...  0.741307      0           0         0   \n",
       "84141        1  0.357148  ...  0.791578      0           0         0   \n",
       "\n",
       "                  p  exp_cc_Bias        p_Bias   dbinom_Bias  P-Value_Bias  \\\n",
       "0      4.851310e-06     9.241261  8.645438e-08  1.021450e-08  1.429445e-08   \n",
       "1      4.646481e-08     1.985142  1.857152e-08  3.597630e-05  4.373368e-05   \n",
       "2      3.598136e-06   392.886045  3.675550e-06  3.215832e-10  1.307941e-09   \n",
       "3      5.488571e-06    31.570973  2.953546e-07  1.053721e-35  1.408527e-35   \n",
       "4      5.173212e-06    19.034114  1.780690e-07  3.084248e-21  4.145798e-21   \n",
       "...             ...          ...           ...           ...           ...   \n",
       "84137  4.792100e-08     2.755553  2.577891e-08  2.543585e-05  3.212350e-05   \n",
       "84138  4.075525e-08     2.239503  2.095113e-08  1.896306e-05  2.322613e-05   \n",
       "84139  4.851310e-06    21.079813  1.972071e-07  8.709230e-33  1.111877e-32   \n",
       "84140  4.528059e-06    12.629390  1.181512e-07  3.929659e-08  5.926719e-08   \n",
       "84141  4.208650e-06    12.235101  1.144625e-07  6.215428e-10  8.914132e-10   \n",
       "\n",
       "       Q-Value_Bias  \n",
       "0      3.577668e-06  \n",
       "1      5.051998e-03  \n",
       "2      3.909070e-07  \n",
       "3      4.933339e-32  \n",
       "4      4.803915e-18  \n",
       "...             ...  \n",
       "84137  3.852477e-03  \n",
       "84138  2.893782e-03  \n",
       "84139  3.184313e-29  \n",
       "84140  1.321116e-05  \n",
       "84141  2.732105e-07  \n",
       "\n",
       "[84142 rows x 25 columns]"
      ]
     },
     "execution_count": 206,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loops"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5072a64c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dd1e5212",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "1eed4340",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/mnt/BioHome/jreyna/tmp//ipykernel_61203/4187765372.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tmp_loops.rename(columns={'p': 'score'}, inplace=True)\n",
      "/mnt/BioHome/jreyna/tmp//ipykernel_61203/4187765372.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tmp_loops.loc[:, 'name'] = '.'\n",
      "/mnt/BioHome/jreyna/tmp//ipykernel_61203/4187765372.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tmp_loops.loc[:, 'score'] = loops['p']\n",
      "/mnt/BioHome/jreyna/tmp//ipykernel_61203/4187765372.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tmp_loops.loc[:, 'strand1'] = '.'\n",
      "/mnt/BioHome/jreyna/tmp//ipykernel_61203/4187765372.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tmp_loops.loc[:, 'strand2'] = '.'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(84142, 10)\n"
     ]
    }
   ],
   "source": [
    "tmp_loops = loops[['chr1', 's1', 'e1', 'chr2', 's2', 'e2']]\n",
    "tmp_loops.rename(columns={'p': 'score'}, inplace=True)\n",
    "tmp_loops.loc[:, 'name'] = '.'\n",
    "tmp_loops.loc[:, 'score'] = loops['p']\n",
    "tmp_loops.loc[:, 'strand1'] = '.'\n",
    "tmp_loops.loc[:, 'strand2'] = '.'\n",
    "loops = pbt.BedTool.from_dataframe(tmp_loops)\n",
    "print(tmp_loops.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "d0a9f5d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>sid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>162368010</td>\n",
       "      <td>162368011</td>\n",
       "      <td>rs984971</td>\n",
       "      <td>2:162368011:A:G</td>\n",
       "      <td>2:162368011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr9</td>\n",
       "      <td>4296429</td>\n",
       "      <td>4296430</td>\n",
       "      <td>rs10814917</td>\n",
       "      <td>9:4296430:A:G</td>\n",
       "      <td>9:4296430</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr      start        end       rs_id       variant_id          sid\n",
       "0   chr2  162368010  162368011    rs984971  2:162368011:A:G  2:162368011\n",
       "20  chr9    4296429    4296430  rs10814917    9:4296430:A:G    9:4296430"
      ]
     },
     "execution_count": 208,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coloc_snps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bb054167",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "008fdd72",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "642009ee",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 2 SNP-Gene pairs with a loop.\n"
     ]
    }
   ],
   "source": [
    "#########################################################################################\n",
    "# Find snp-gene pairs overlapping loops\n",
    "#########################################################################################\n",
    "\n",
    "# re-arranging to fit bedpe format\n",
    "fivekb_gloops = fivekb_genes.copy()\n",
    "\n",
    "# loading into pbt\n",
    "fivekb_gloops = pbt.BedTool.from_dataframe(fivekb_gloops)\n",
    "fivekb_gloops = fivekb_gloops.pair_to_pair(loops, type='both', slop=20000, **{'is':True})\n",
    "#fivekb_gloops = fivekb_gloops.pair_to_pair(loops, type='both', slop=7500, **{'is':True})\n",
    "\n",
    "fivekb_gloops = fivekb_gloops.to_dataframe(disable_auto_names=True, header=None)\n",
    "\n",
    "if len(fivekb_gloops) > 0:\n",
    "    fivekb_gloops_set = fivekb_gloops.iloc[:, [13,12]]\n",
    "    fivekb_gloops_uniq = set([tuple(x) for x in fivekb_gloops_set.values.tolist()])\n",
    "else:\n",
    "    print('WARNING: Found no overlap between the genes and loops.')\n",
    "    fivekb_gloops_uniq = set()\n",
    "\n",
    "print('There are {} SNP-Gene pairs with a loop.'.format(len(fivekb_gloops_uniq)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "9ad2faa7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Construct master table\n",
      "Master is starting with 942 snp-gene pairs.\n"
     ]
    }
   ],
   "source": [
    "#########################################################################################\n",
    "# Construct master table\n",
    "#########################################################################################\n",
    "print('# Construct master table')\n",
    "\n",
    "# begin making the master\n",
    "master = fivekb_genes.copy()\n",
    "master['sid'] = master['chrA'].str.replace('chr', '') + ':' + master['endA'].astype(str)\n",
    "\n",
    "print('Master is starting with {} snp-gene pairs.'.format(master.shape[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "975e968a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#coloc_sig_df['sid'] = coloc_sig_df['chr'].str.replace('chr', '') + ':'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "528d3371",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# #### Add eqtl results\n",
      "There are 10000 eQTLs.\n"
     ]
    }
   ],
   "source": [
    "# #### Add eqtl results \n",
    "print(\"# #### Add eqtl results\")\n",
    "\n",
    "# load the eQTL's\n",
    "eqtls = pd.read_table(eqtl_fn, header=0).head(10000)\n",
    "# old eqtls.columns = ['eqtl_gname', 'nvar', 'shape1', 'shape2', 'dummy',\n",
    "#                 'sid', 'dist', 'npval', 'slope', 'ppval', 'bpval', 'qval']\n",
    "# new eqtls.columns\n",
    "# molecular_trait_id, chromosome, position, ref, alt, variant, ma_samples,\n",
    "# maf, pvalue, beta, se, type, ac, an, r2, molecular_trait_object_id, gene_id, median_tpm, rsid, dist, fdr\n",
    "eqtls['sid'] = eqtls['chromosome'].astype(str) + ':' + eqtls['position'].astype(str)\n",
    "\n",
    "print('There are {} eQTLs.'.format(eqtls.shape[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "a050407c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# need to use outer or else you exclude some eQTL's\n",
    "master = master.merge(eqtls, left_on=['sid', 'gname'], right_on=['sid', 'molecular_trait_id'], how='outer')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "ee666f22",
   "metadata": {},
   "outputs": [],
   "source": [
    "# add column to filter on eqtl snp status\n",
    "master['is_eqtl_pair'] = (~master['pvalue'].isna()).astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "c6018b68",
   "metadata": {},
   "outputs": [],
   "source": [
    "# add gene names to entries with a missing name (after adding eQTL info)\n",
    "master.loc[master.gname.isna(), 'gname'] = master.loc[master.gname.isna(), 'molecular_trait_id']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "6802cf89",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrA</th>\n",
       "      <th>startA</th>\n",
       "      <th>endA</th>\n",
       "      <th>chrB</th>\n",
       "      <th>startB</th>\n",
       "      <th>endB</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand1</th>\n",
       "      <th>strand2</th>\n",
       "      <th>...</th>\n",
       "      <th>ac</th>\n",
       "      <th>an</th>\n",
       "      <th>r2</th>\n",
       "      <th>molecular_trait_object_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>median_tpm</th>\n",
       "      <th>rsid</th>\n",
       "      <th>dist</th>\n",
       "      <th>fdr</th>\n",
       "      <th>is_eqtl_pair</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr18</td>\n",
       "      <td>12780463.0</td>\n",
       "      <td>12780464.0</td>\n",
       "      <td>chr18</td>\n",
       "      <td>12288307.0</td>\n",
       "      <td>12288308.0</td>\n",
       "      <td>rs9952991_AP005264.1</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr18</td>\n",
       "      <td>12780463.0</td>\n",
       "      <td>12780464.0</td>\n",
       "      <td>chr18</td>\n",
       "      <td>12288307.0</td>\n",
       "      <td>12288308.0</td>\n",
       "      <td>rs9952991_AP005264.1</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr18</td>\n",
       "      <td>12780463.0</td>\n",
       "      <td>12780464.0</td>\n",
       "      <td>chr18</td>\n",
       "      <td>12288307.0</td>\n",
       "      <td>12288308.0</td>\n",
       "      <td>rs9952991_AP005264.1</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr18</td>\n",
       "      <td>12780463.0</td>\n",
       "      <td>12780464.0</td>\n",
       "      <td>chr18</td>\n",
       "      <td>12288307.0</td>\n",
       "      <td>12288308.0</td>\n",
       "      <td>rs9952991_AP005264.1</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr18</td>\n",
       "      <td>12780463.0</td>\n",
       "      <td>12780464.0</td>\n",
       "      <td>chr18</td>\n",
       "      <td>12288307.0</td>\n",
       "      <td>12288308.0</td>\n",
       "      <td>rs9952991_AP005264.1</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10937</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>7.0</td>\n",
       "      <td>396.0</td>\n",
       "      <td>0.89079</td>\n",
       "      <td>ENSG00000162542</td>\n",
       "      <td>ENSG00000162542</td>\n",
       "      <td>2.315</td>\n",
       "      <td>rs142679596</td>\n",
       "      <td>688135.0</td>\n",
       "      <td>0.007247</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10938</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>104.0</td>\n",
       "      <td>396.0</td>\n",
       "      <td>0.99999</td>\n",
       "      <td>ENSG00000127481</td>\n",
       "      <td>ENSG00000127481</td>\n",
       "      <td>69.666</td>\n",
       "      <td>rs6693149</td>\n",
       "      <td>90173.0</td>\n",
       "      <td>0.000315</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10939</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>96.0</td>\n",
       "      <td>396.0</td>\n",
       "      <td>0.99759</td>\n",
       "      <td>ENSG00000127481</td>\n",
       "      <td>ENSG00000127481</td>\n",
       "      <td>69.666</td>\n",
       "      <td>rs6696220</td>\n",
       "      <td>86426.0</td>\n",
       "      <td>0.002288</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10940</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>102.0</td>\n",
       "      <td>396.0</td>\n",
       "      <td>0.99582</td>\n",
       "      <td>ENSG00000127481</td>\n",
       "      <td>ENSG00000127481</td>\n",
       "      <td>69.666</td>\n",
       "      <td>rs6688770</td>\n",
       "      <td>86155.0</td>\n",
       "      <td>0.000315</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10941</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>96.0</td>\n",
       "      <td>396.0</td>\n",
       "      <td>0.99516</td>\n",
       "      <td>ENSG00000127481</td>\n",
       "      <td>ENSG00000127481</td>\n",
       "      <td>69.666</td>\n",
       "      <td>rs67443992</td>\n",
       "      <td>79026.0</td>\n",
       "      <td>0.002341</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10942 rows × 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrA      startA        endA   chrB      startB        endB  \\\n",
       "0      chr18  12780463.0  12780464.0  chr18  12288307.0  12288308.0   \n",
       "1      chr18  12780463.0  12780464.0  chr18  12288307.0  12288308.0   \n",
       "2      chr18  12780463.0  12780464.0  chr18  12288307.0  12288308.0   \n",
       "3      chr18  12780463.0  12780464.0  chr18  12288307.0  12288308.0   \n",
       "4      chr18  12780463.0  12780464.0  chr18  12288307.0  12288308.0   \n",
       "...      ...         ...         ...    ...         ...         ...   \n",
       "10937    NaN         NaN         NaN    NaN         NaN         NaN   \n",
       "10938    NaN         NaN         NaN    NaN         NaN         NaN   \n",
       "10939    NaN         NaN         NaN    NaN         NaN         NaN   \n",
       "10940    NaN         NaN         NaN    NaN         NaN         NaN   \n",
       "10941    NaN         NaN         NaN    NaN         NaN         NaN   \n",
       "\n",
       "                       name score strand1 strand2  ...     ac     an       r2  \\\n",
       "0      rs9952991_AP005264.1     .       +       +  ...    NaN    NaN      NaN   \n",
       "1      rs9952991_AP005264.1     .       +       +  ...    NaN    NaN      NaN   \n",
       "2      rs9952991_AP005264.1     .       +       +  ...    NaN    NaN      NaN   \n",
       "3      rs9952991_AP005264.1     .       +       +  ...    NaN    NaN      NaN   \n",
       "4      rs9952991_AP005264.1     .       +       +  ...    NaN    NaN      NaN   \n",
       "...                     ...   ...     ...     ...  ...    ...    ...      ...   \n",
       "10937                   NaN   NaN     NaN     NaN  ...    7.0  396.0  0.89079   \n",
       "10938                   NaN   NaN     NaN     NaN  ...  104.0  396.0  0.99999   \n",
       "10939                   NaN   NaN     NaN     NaN  ...   96.0  396.0  0.99759   \n",
       "10940                   NaN   NaN     NaN     NaN  ...  102.0  396.0  0.99582   \n",
       "10941                   NaN   NaN     NaN     NaN  ...   96.0  396.0  0.99516   \n",
       "\n",
       "      molecular_trait_object_id          gene_id  median_tpm         rsid  \\\n",
       "0                           NaN              NaN         NaN          NaN   \n",
       "1                           NaN              NaN         NaN          NaN   \n",
       "2                           NaN              NaN         NaN          NaN   \n",
       "3                           NaN              NaN         NaN          NaN   \n",
       "4                           NaN              NaN         NaN          NaN   \n",
       "...                         ...              ...         ...          ...   \n",
       "10937           ENSG00000162542  ENSG00000162542       2.315  rs142679596   \n",
       "10938           ENSG00000127481  ENSG00000127481      69.666    rs6693149   \n",
       "10939           ENSG00000127481  ENSG00000127481      69.666    rs6696220   \n",
       "10940           ENSG00000127481  ENSG00000127481      69.666    rs6688770   \n",
       "10941           ENSG00000127481  ENSG00000127481      69.666   rs67443992   \n",
       "\n",
       "           dist       fdr is_eqtl_pair  \n",
       "0           NaN       NaN            0  \n",
       "1           NaN       NaN            0  \n",
       "2           NaN       NaN            0  \n",
       "3           NaN       NaN            0  \n",
       "4           NaN       NaN            0  \n",
       "...         ...       ...          ...  \n",
       "10937  688135.0  0.007247            1  \n",
       "10938   90173.0  0.000315            1  \n",
       "10939   86426.0  0.002288            1  \n",
       "10940   86155.0  0.000315            1  \n",
       "10941   79026.0  0.002341            1  \n",
       "\n",
       "[10942 rows x 36 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "7c6e731d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "After outer merging with eqtls, master has 10942 rows.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/mnt/BioHome/jreyna/tmp//ipykernel_61203/3726262882.py:2: FutureWarning: The default value of regex will change from True to False in a future version.\n",
      "  master.loc[master.chrA.isna(), 'chrA'] = 'chr' + master.loc[master.chrA.isna(), 'sid'].str.replace(':[0-9]+', '')\n",
      "/mnt/BioHome/jreyna/tmp//ipykernel_61203/3726262882.py:3: FutureWarning: The default value of regex will change from True to False in a future version.\n",
      "  master.loc[master.chrB.isna(), 'chrB'] = 'chr' + master.loc[master.chrB.isna(), 'sid'].str.replace(':[0-9]+', '')\n",
      "/mnt/BioHome/jreyna/tmp//ipykernel_61203/3726262882.py:4: FutureWarning: The default value of regex will change from True to False in a future version.\n",
      "  master.loc[master.startA.isna(), 'startA'] = (master.loc[master.startA.isna(), 'sid'].str.replace('[0-9]+:', '')).astype(int)\n",
      "/mnt/BioHome/jreyna/tmp//ipykernel_61203/3726262882.py:6: FutureWarning: The default value of regex will change from True to False in a future version.\n",
      "  master.loc[master.endA.isna(), 'endA'] = master.loc[master.endA.isna(), 'sid'].str.replace('[0-9]+:', '')\n"
     ]
    }
   ],
   "source": [
    "# add missing chrA, chrB, startA and endA data for the eQTL rows\n",
    "master.loc[master.chrA.isna(), 'chrA'] = 'chr' + master.loc[master.chrA.isna(), 'sid'].str.replace(':[0-9]+', '')\n",
    "master.loc[master.chrB.isna(), 'chrB'] = 'chr' + master.loc[master.chrB.isna(), 'sid'].str.replace(':[0-9]+', '')\n",
    "master.loc[master.startA.isna(), 'startA'] = (master.loc[master.startA.isna(), 'sid'].str.replace('[0-9]+:', '')).astype(int)\n",
    "master.loc[master.startA.isna(), 'startA'] -= 1     \n",
    "master.loc[master.endA.isna(), 'endA'] = master.loc[master.endA.isna(), 'sid'].str.replace('[0-9]+:', '')\n",
    "\n",
    "print('After outer merging with eqtls, master has {} rows.'.format(master.shape[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "e53112b5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrA</th>\n",
       "      <th>startA</th>\n",
       "      <th>endA</th>\n",
       "      <th>chrB</th>\n",
       "      <th>startB</th>\n",
       "      <th>endB</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand1</th>\n",
       "      <th>strand2</th>\n",
       "      <th>...</th>\n",
       "      <th>ac</th>\n",
       "      <th>an</th>\n",
       "      <th>r2</th>\n",
       "      <th>molecular_trait_object_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>median_tpm</th>\n",
       "      <th>rsid</th>\n",
       "      <th>dist</th>\n",
       "      <th>fdr</th>\n",
       "      <th>is_eqtl_pair</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr18</td>\n",
       "      <td>12780463.0</td>\n",
       "      <td>12780464.0</td>\n",
       "      <td>chr18</td>\n",
       "      <td>12288307.0</td>\n",
       "      <td>12288308.0</td>\n",
       "      <td>rs9952991_AP005264.1</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr18</td>\n",
       "      <td>12780463.0</td>\n",
       "      <td>12780464.0</td>\n",
       "      <td>chr18</td>\n",
       "      <td>12288307.0</td>\n",
       "      <td>12288308.0</td>\n",
       "      <td>rs9952991_AP005264.1</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr18</td>\n",
       "      <td>12780463.0</td>\n",
       "      <td>12780464.0</td>\n",
       "      <td>chr18</td>\n",
       "      <td>12288307.0</td>\n",
       "      <td>12288308.0</td>\n",
       "      <td>rs9952991_AP005264.1</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr18</td>\n",
       "      <td>12780463.0</td>\n",
       "      <td>12780464.0</td>\n",
       "      <td>chr18</td>\n",
       "      <td>12288307.0</td>\n",
       "      <td>12288308.0</td>\n",
       "      <td>rs9952991_AP005264.1</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr18</td>\n",
       "      <td>12780463.0</td>\n",
       "      <td>12780464.0</td>\n",
       "      <td>chr18</td>\n",
       "      <td>12288307.0</td>\n",
       "      <td>12288308.0</td>\n",
       "      <td>rs9952991_AP005264.1</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10937</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19111810.0</td>\n",
       "      <td>19111810</td>\n",
       "      <td>chr1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>7.0</td>\n",
       "      <td>396.0</td>\n",
       "      <td>0.89079</td>\n",
       "      <td>ENSG00000162542</td>\n",
       "      <td>ENSG00000162542</td>\n",
       "      <td>2.315</td>\n",
       "      <td>rs142679596</td>\n",
       "      <td>688135.0</td>\n",
       "      <td>0.007247</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10938</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19120093.0</td>\n",
       "      <td>19120093</td>\n",
       "      <td>chr1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>104.0</td>\n",
       "      <td>396.0</td>\n",
       "      <td>0.99999</td>\n",
       "      <td>ENSG00000127481</td>\n",
       "      <td>ENSG00000127481</td>\n",
       "      <td>69.666</td>\n",
       "      <td>rs6693149</td>\n",
       "      <td>90173.0</td>\n",
       "      <td>0.000315</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10939</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19123840.0</td>\n",
       "      <td>19123840</td>\n",
       "      <td>chr1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>96.0</td>\n",
       "      <td>396.0</td>\n",
       "      <td>0.99759</td>\n",
       "      <td>ENSG00000127481</td>\n",
       "      <td>ENSG00000127481</td>\n",
       "      <td>69.666</td>\n",
       "      <td>rs6696220</td>\n",
       "      <td>86426.0</td>\n",
       "      <td>0.002288</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10940</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19124111.0</td>\n",
       "      <td>19124111</td>\n",
       "      <td>chr1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>102.0</td>\n",
       "      <td>396.0</td>\n",
       "      <td>0.99582</td>\n",
       "      <td>ENSG00000127481</td>\n",
       "      <td>ENSG00000127481</td>\n",
       "      <td>69.666</td>\n",
       "      <td>rs6688770</td>\n",
       "      <td>86155.0</td>\n",
       "      <td>0.000315</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10941</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19131240.0</td>\n",
       "      <td>19131240</td>\n",
       "      <td>chr1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>96.0</td>\n",
       "      <td>396.0</td>\n",
       "      <td>0.99516</td>\n",
       "      <td>ENSG00000127481</td>\n",
       "      <td>ENSG00000127481</td>\n",
       "      <td>69.666</td>\n",
       "      <td>rs67443992</td>\n",
       "      <td>79026.0</td>\n",
       "      <td>0.002341</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10942 rows × 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrA      startA        endA   chrB      startB        endB  \\\n",
       "0      chr18  12780463.0  12780464.0  chr18  12288307.0  12288308.0   \n",
       "1      chr18  12780463.0  12780464.0  chr18  12288307.0  12288308.0   \n",
       "2      chr18  12780463.0  12780464.0  chr18  12288307.0  12288308.0   \n",
       "3      chr18  12780463.0  12780464.0  chr18  12288307.0  12288308.0   \n",
       "4      chr18  12780463.0  12780464.0  chr18  12288307.0  12288308.0   \n",
       "...      ...         ...         ...    ...         ...         ...   \n",
       "10937   chr1  19111810.0    19111810   chr1         NaN         NaN   \n",
       "10938   chr1  19120093.0    19120093   chr1         NaN         NaN   \n",
       "10939   chr1  19123840.0    19123840   chr1         NaN         NaN   \n",
       "10940   chr1  19124111.0    19124111   chr1         NaN         NaN   \n",
       "10941   chr1  19131240.0    19131240   chr1         NaN         NaN   \n",
       "\n",
       "                       name score strand1 strand2  ...     ac     an       r2  \\\n",
       "0      rs9952991_AP005264.1     .       +       +  ...    NaN    NaN      NaN   \n",
       "1      rs9952991_AP005264.1     .       +       +  ...    NaN    NaN      NaN   \n",
       "2      rs9952991_AP005264.1     .       +       +  ...    NaN    NaN      NaN   \n",
       "3      rs9952991_AP005264.1     .       +       +  ...    NaN    NaN      NaN   \n",
       "4      rs9952991_AP005264.1     .       +       +  ...    NaN    NaN      NaN   \n",
       "...                     ...   ...     ...     ...  ...    ...    ...      ...   \n",
       "10937                   NaN   NaN     NaN     NaN  ...    7.0  396.0  0.89079   \n",
       "10938                   NaN   NaN     NaN     NaN  ...  104.0  396.0  0.99999   \n",
       "10939                   NaN   NaN     NaN     NaN  ...   96.0  396.0  0.99759   \n",
       "10940                   NaN   NaN     NaN     NaN  ...  102.0  396.0  0.99582   \n",
       "10941                   NaN   NaN     NaN     NaN  ...   96.0  396.0  0.99516   \n",
       "\n",
       "      molecular_trait_object_id          gene_id  median_tpm         rsid  \\\n",
       "0                           NaN              NaN         NaN          NaN   \n",
       "1                           NaN              NaN         NaN          NaN   \n",
       "2                           NaN              NaN         NaN          NaN   \n",
       "3                           NaN              NaN         NaN          NaN   \n",
       "4                           NaN              NaN         NaN          NaN   \n",
       "...                         ...              ...         ...          ...   \n",
       "10937           ENSG00000162542  ENSG00000162542       2.315  rs142679596   \n",
       "10938           ENSG00000127481  ENSG00000127481      69.666    rs6693149   \n",
       "10939           ENSG00000127481  ENSG00000127481      69.666    rs6696220   \n",
       "10940           ENSG00000127481  ENSG00000127481      69.666    rs6688770   \n",
       "10941           ENSG00000127481  ENSG00000127481      69.666   rs67443992   \n",
       "\n",
       "           dist       fdr is_eqtl_pair  \n",
       "0           NaN       NaN            0  \n",
       "1           NaN       NaN            0  \n",
       "2           NaN       NaN            0  \n",
       "3           NaN       NaN            0  \n",
       "4           NaN       NaN            0  \n",
       "...         ...       ...          ...  \n",
       "10937  688135.0  0.007247            1  \n",
       "10938   90173.0  0.000315            1  \n",
       "10939   86426.0  0.002288            1  \n",
       "10940   86155.0  0.000315            1  \n",
       "10941   79026.0  0.002341            1  \n",
       "\n",
       "[10942 rows x 36 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "7998e54a",
   "metadata": {},
   "outputs": [],
   "source": [
    "master.loc[~master.molecular_trait_id.isna(), 'gid'] = master.loc[~master.molecular_trait_id.isna(), 'molecular_trait_id']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "c28a4f13",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# #### Add info about closests gene\n"
     ]
    }
   ],
   "source": [
    "# add back the original gene start and end\n",
    "master = master.merge(orig_genes_df[['start', 'end', 'gene_id', 'strand']], left_on='gid', right_on='gene_id')\n",
    "\n",
    "# convert the startB/endB position into startB/endB for the TSS\n",
    "# if the gene is + then the startB is uses as the tss otherwise\n",
    "# the endB is used as the tss\n",
    "master.loc[(master.strand == '+'), 'endB'] = master.loc[(master.strand == '+'), 'start']\n",
    "master.loc[(master.strand == '+'), 'startB'] = master.loc[(master.strand == '+'), 'start'] - 1\n",
    "\n",
    "master.loc[(master.strand == '-'), 'endB'] = master.loc[(master.strand == '-'), 'end']\n",
    "master.loc[(master.strand == '-'), 'startB'] = master.loc[(master.strand == '-'), 'end'] - 1\n",
    "\n",
    "# convert the coordinates from floats to ints\n",
    "master.startA = master.startA.astype(int)\n",
    "master.startB = master.startB.astype(int)\n",
    "master.endA = master.endA.astype(int)\n",
    "master.endB = master.endB.astype(int)\n",
    "\n",
    "master.rename(columns={'start': 'gene_start', 'end': 'gene_end', 'strand': 'gene_strand'}, inplace=True)\n",
    "\n",
    "# #### Add info about closests gene\n",
    "print(\"# #### Add info about closests gene\")\n",
    "\n",
    "# check for the closets gene\n",
    "closets_check = [0] * master.shape[0]\n",
    "for i, sr in master.iterrows():\n",
    "\n",
    "    # check closest gene\n",
    "    rs_gene = (sr.sid, sr.gname)\n",
    "    if rs_gene in closest_gene.index:\n",
    "        closets_check[i] = 1\n",
    "\n",
    "master['is_closest_gene'] = closets_check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "05c76aba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# #### Add colocalization data\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/mnt/BioHome/jreyna/tmp//ipykernel_61203/2650951363.py:22: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tmp_coloc.rename(columns={'slope_gwas': 'gwas_slope',\n",
      "/mnt/BioHome/jreyna/tmp//ipykernel_61203/2650951363.py:27: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tmp_coloc['sid'] = tmp_coloc['chr'].str.replace('chr', '') + ':' +  tmp_coloc['pos'].astype(str) # NEW\n"
     ]
    }
   ],
   "source": [
    "# #### Add colocalization data\n",
    "print(\"# #### Add colocalization data\")\n",
    "\n",
    "# add colocalization data for SNP and is_coloc_snp columns\n",
    "tmp_coloc = coloc_sig_full[['chr', 'pos', # NEW\n",
    " 'pp_H0_Coloc_Summary',\n",
    " 'pp_H1_Coloc_Summary',\n",
    " 'pp_H2_Coloc_Summary',\n",
    " 'pp_H3_Coloc_Summary',\n",
    " 'pp_H4_Coloc_Summary',\n",
    " 'rs_id',\n",
    " 'geneName',\n",
    " 'ref',\n",
    " 'alt',\n",
    " 'AC',\n",
    " 'AF',\n",
    " 'AN',\n",
    " 'slope_gwas',\n",
    " 'slope_se_gwas',\n",
    " 'pval_nominal',\n",
    " 'SampleSize']]\n",
    "tmp_coloc.rename(columns={'slope_gwas': 'gwas_slope',\n",
    "                          'slope_se_gwas': 'gwas_slope_se',\n",
    "                          'pval_nominal': 'gwas_pval_nominal',\n",
    "                          'geneName': 'gname'}, inplace=True)\n",
    "\n",
    "tmp_coloc['sid'] = tmp_coloc['chr'].str.replace('chr', '') + ':' +  tmp_coloc['pos'].astype(str) # NEW"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "7c6c2713",
   "metadata": {},
   "outputs": [],
   "source": [
    "master = master.merge(tmp_coloc, left_on=['sid', 'gid'], right_on=['sid', 'gname'], how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "0f5dcb9f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "After left merging master with the colocalization table there are 10942 entries.\n"
     ]
    }
   ],
   "source": [
    "# add column to filter on coloc snp status\n",
    "master['is_coloc_pair'] = (~master['pp_H4_Coloc_Summary'].isna()).astype(int)\n",
    "\n",
    "print('After left merging master with the colocalization table there are {} entries.'.format(master.shape[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "041d2268",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# #### Add loop data\n",
      "There are 75 SNP-Gene loops.\n"
     ]
    }
   ],
   "source": [
    "# #### Add loop data\n",
    "print('# #### Add loop data')\n",
    "\n",
    "# check for the loop gene\n",
    "loop_check = [0] * master.shape[0]\n",
    "for i, sr in master.iterrows():\n",
    "\n",
    "    # check closest gene\n",
    "    rs_gene = (sr.sid, sr.gid)\n",
    "    if rs_gene in fivekb_gloops_uniq:\n",
    "        loop_check[i] = 1       \n",
    "\n",
    "master['has_fithichip_loop'] = loop_check\n",
    "print('There are {} SNP-Gene loops.'.format(sum(loop_check)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "485eab4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# #### Do the final reordering and saving\n",
    "\n",
    "# Include PC HiC data when it's available\n",
    "final_cols = [\n",
    "    'sid', \n",
    "    'rsid',\n",
    "    'gid',\n",
    "    'chrA',\n",
    "    'endA',\n",
    "    'startB',\n",
    "    'endB',\n",
    "    'is_eqtl_pair',\n",
    "    'is_coloc_pair',\n",
    "    'is_closest_gene',\n",
    "    'has_fithichip_loop',\n",
    "    'pvalue', \n",
    "    'beta', \n",
    "    'fdr',\n",
    "    'dist',\n",
    "    'pp_H0_Coloc_Summary',\n",
    "    'pp_H1_Coloc_Summary',\n",
    "    'pp_H2_Coloc_Summary',\n",
    "    'pp_H3_Coloc_Summary',\n",
    "    'pp_H4_Coloc_Summary',\n",
    "    'gene_start',\n",
    "    'gene_end',\n",
    "    'ref_x', \n",
    "    'alt_x',\n",
    "    'AC',\n",
    "    'AF',\n",
    "    'AN',\n",
    "    'gwas_slope',\n",
    "    'gwas_slope_se',\n",
    "    'gwas_pval_nominal',\n",
    "    'SampleSize']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "fb479c04",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'rs_id'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Input \u001b[0;32mIn [26]\u001b[0m, in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m master \u001b[38;5;241m=\u001b[39m master[final_cols]\n\u001b[1;32m      2\u001b[0m master\u001b[38;5;241m.\u001b[39mrename(columns\u001b[38;5;241m=\u001b[39m{\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mchrA\u001b[39m\u001b[38;5;124m'\u001b[39m:\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mchrom\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mendA\u001b[39m\u001b[38;5;124m'\u001b[39m: \u001b[38;5;124m'\u001b[39m\u001b[38;5;124msnp_pos\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mref_x\u001b[39m\u001b[38;5;124m'\u001b[39m: \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mref\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124malt_x\u001b[39m\u001b[38;5;124m'\u001b[39m: \u001b[38;5;124m'\u001b[39m\u001b[38;5;124malt\u001b[39m\u001b[38;5;124m'\u001b[39m,\n\u001b[1;32m      3\u001b[0m                        \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mstartB\u001b[39m\u001b[38;5;124m'\u001b[39m: \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mtss_start\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mendB\u001b[39m\u001b[38;5;124m'\u001b[39m: \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mtss_end\u001b[39m\u001b[38;5;124m'\u001b[39m,\n\u001b[1;32m      4\u001b[0m                        \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mgname\u001b[39m\u001b[38;5;124m'\u001b[39m: \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mgene_name\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mgid\u001b[39m\u001b[38;5;124m'\u001b[39m: \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mgene_id\u001b[39m\u001b[38;5;124m'\u001b[39m}, inplace\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m----> 6\u001b[0m \u001b[43mmaster\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msort_values\u001b[49m\u001b[43m(\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mchrom\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43msnp_pos\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mtss_start\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mrs_id\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43minplace\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m)\u001b[49m\n\u001b[1;32m      7\u001b[0m master\u001b[38;5;241m.\u001b[39msnp_pos \u001b[38;5;241m=\u001b[39m master\u001b[38;5;241m.\u001b[39msnp_pos\u001b[38;5;241m.\u001b[39mastype(\u001b[38;5;28mint\u001b[39m)\n\u001b[1;32m      9\u001b[0m \u001b[38;5;66;03m# write out the master data\u001b[39;00m\n",
      "File \u001b[0;32m~/software/anaconda3/hichip-db/lib/python3.10/site-packages/pandas/util/_decorators.py:311\u001b[0m, in \u001b[0;36mdeprecate_nonkeyword_arguments.<locals>.decorate.<locals>.wrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    305\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(args) \u001b[38;5;241m>\u001b[39m num_allow_args:\n\u001b[1;32m    306\u001b[0m     warnings\u001b[38;5;241m.\u001b[39mwarn(\n\u001b[1;32m    307\u001b[0m         msg\u001b[38;5;241m.\u001b[39mformat(arguments\u001b[38;5;241m=\u001b[39marguments),\n\u001b[1;32m    308\u001b[0m         \u001b[38;5;167;01mFutureWarning\u001b[39;00m,\n\u001b[1;32m    309\u001b[0m         stacklevel\u001b[38;5;241m=\u001b[39mstacklevel,\n\u001b[1;32m    310\u001b[0m     )\n\u001b[0;32m--> 311\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mfunc\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/software/anaconda3/hichip-db/lib/python3.10/site-packages/pandas/core/frame.py:6297\u001b[0m, in \u001b[0;36mDataFrame.sort_values\u001b[0;34m(self, by, axis, ascending, inplace, kind, na_position, ignore_index, key)\u001b[0m\n\u001b[1;32m   6292\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\n\u001b[1;32m   6293\u001b[0m         \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mLength of ascending (\u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mlen\u001b[39m(ascending)\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m) != length of by (\u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mlen\u001b[39m(by)\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m)\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m   6294\u001b[0m     )\n\u001b[1;32m   6295\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(by) \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m1\u001b[39m:\n\u001b[0;32m-> 6297\u001b[0m     keys \u001b[38;5;241m=\u001b[39m [\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_get_label_or_level_values(x, axis\u001b[38;5;241m=\u001b[39maxis) \u001b[38;5;28;01mfor\u001b[39;00m x \u001b[38;5;129;01min\u001b[39;00m by]\n\u001b[1;32m   6299\u001b[0m     \u001b[38;5;66;03m# need to rewrap columns in Series to apply key function\u001b[39;00m\n\u001b[1;32m   6300\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m key \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m   6301\u001b[0m         \u001b[38;5;66;03m# error: List comprehension has incompatible type List[Series];\u001b[39;00m\n\u001b[1;32m   6302\u001b[0m         \u001b[38;5;66;03m# expected List[ndarray]\u001b[39;00m\n",
      "File \u001b[0;32m~/software/anaconda3/hichip-db/lib/python3.10/site-packages/pandas/core/frame.py:6297\u001b[0m, in \u001b[0;36m<listcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m   6292\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\n\u001b[1;32m   6293\u001b[0m         \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mLength of ascending (\u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mlen\u001b[39m(ascending)\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m) != length of by (\u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mlen\u001b[39m(by)\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m)\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m   6294\u001b[0m     )\n\u001b[1;32m   6295\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(by) \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m1\u001b[39m:\n\u001b[0;32m-> 6297\u001b[0m     keys \u001b[38;5;241m=\u001b[39m [\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_get_label_or_level_values\u001b[49m\u001b[43m(\u001b[49m\u001b[43mx\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43maxis\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43maxis\u001b[49m\u001b[43m)\u001b[49m \u001b[38;5;28;01mfor\u001b[39;00m x \u001b[38;5;129;01min\u001b[39;00m by]\n\u001b[1;32m   6299\u001b[0m     \u001b[38;5;66;03m# need to rewrap columns in Series to apply key function\u001b[39;00m\n\u001b[1;32m   6300\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m key \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m   6301\u001b[0m         \u001b[38;5;66;03m# error: List comprehension has incompatible type List[Series];\u001b[39;00m\n\u001b[1;32m   6302\u001b[0m         \u001b[38;5;66;03m# expected List[ndarray]\u001b[39;00m\n",
      "File \u001b[0;32m~/software/anaconda3/hichip-db/lib/python3.10/site-packages/pandas/core/generic.py:1848\u001b[0m, in \u001b[0;36mNDFrame._get_label_or_level_values\u001b[0;34m(self, key, axis)\u001b[0m\n\u001b[1;32m   1846\u001b[0m     values \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39maxes[axis]\u001b[38;5;241m.\u001b[39mget_level_values(key)\u001b[38;5;241m.\u001b[39m_values\n\u001b[1;32m   1847\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m-> 1848\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(key)\n\u001b[1;32m   1850\u001b[0m \u001b[38;5;66;03m# Check for duplicates\u001b[39;00m\n\u001b[1;32m   1851\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m values\u001b[38;5;241m.\u001b[39mndim \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m1\u001b[39m:\n",
      "\u001b[0;31mKeyError\u001b[0m: 'rs_id'"
     ]
    }
   ],
   "source": [
    "master = master[final_cols]\n",
    "master.rename(columns={'chrA':'chrom', 'endA': 'snp_pos', 'ref_x': 'ref', 'alt_x': 'alt',\n",
    "                       'startB': 'tss_start', 'endB': 'tss_end',\n",
    "                       'gname': 'gene_name', 'gid': 'gene_id'}, inplace=True)\n",
    "\n",
    "master.sort_values(['chrom', 'snp_pos', 'tss_start', 'rs_id'], inplace=True)\n",
    "master.snp_pos = master.snp_pos.astype(int)\n",
    "\n",
    "# write out the master data\n",
    "fn = os.path.join(outdir, 'master.tsv')\n",
    "master.to_csv(fn, sep='\\t', header=True, index=False)\n",
    "\n",
    "fn = os.path.join(outdir, 'master.xlsx')\n",
    "excel_master = master.sort_values('rs_id').set_index('rs_id')\n",
    "excel_master.to_excel(fn, na_rep='nan')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b4bf9847",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f81276ef",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9a3c59ed",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bfb6ceb2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9dba9447",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ff95065b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7b289e2b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f7b948bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "#########################################################################################\n",
    "# Make WashU files \n",
    "#########################################################################################\n",
    "\n",
    "def bedpe_to_WashU_longrange(fn, df):\n",
    "    \"\"\"\n",
    "        Convert from a loop bedpe file into WashU longrange, \n",
    "        includes bgzip and tabix of the fn. \n",
    "        \n",
    "        Params\n",
    "        -------\n",
    "        fn: str\n",
    "            path to the longrange output file (without gz)\n",
    "            \n",
    "        df: dataframe\n",
    "            columns 1-6 are as expected and column 7 is the p or q-value. \n",
    "            \n",
    "        Output\n",
    "        ------\n",
    "        gzfn: str\n",
    "            path to the longrange with bgzip compression\n",
    "        tabix_fn: str\n",
    "            path to the index of the longrange file\n",
    "            \n",
    "    \"\"\"\n",
    "\n",
    "    # parsing the data into WashU longrage format\n",
    "    data = []\n",
    "    for sr in df.values.tolist():\n",
    "\n",
    "        # calculate the -log(FDR)\n",
    "        qval = -np.log(sr[6])\n",
    "\n",
    "        # get the first pair data\n",
    "        second_pair_str = '{}:{}-{},{:.5f}'.format(*sr[3:6], qval)\n",
    "        first_row = sr[0:3] + [second_pair_str]\n",
    "\n",
    "        # get the second pair data\n",
    "        first_pair_str = '{}:{}-{},{:.5f}'.format(*sr[0:3], qval)\n",
    "        second_row = sr[3:6] + [first_pair_str]\n",
    "\n",
    "        # add each data row\n",
    "        data.append(first_row)\n",
    "        data.append(second_row)\n",
    "\n",
    "    data = sorted(data, key=lambda x: (x[0], x[1], x[2]))\n",
    "\n",
    "    # writing out the data\n",
    "    with open(fn, 'w') as f:\n",
    "        for line in data:\n",
    "            info = [str(x) for x in line]\n",
    "            info = '\\t'.join(info)\n",
    "            f.write(info + '\\n')\n",
    "            \n",
    "    # run bgzip\n",
    "    cmd = '{} {}'.format(bgzip, fn)\n",
    "    print(cmd)\n",
    "    job = sp.Popen(cmd, stderr=sp.PIPE,stdout=sp.PIPE, shell=True)\n",
    "\n",
    "    out, err = job.communicate()\n",
    "    print('out:', out.decode())\n",
    "    print('err:', err.decode())\n",
    "    \n",
    "    # run tabix\n",
    "    lrange_gzfn = fn + '.gz'\n",
    "    cmd = '{} -f {}'.format(tabix, lrange_gzfn)\n",
    "    print(cmd)\n",
    "    job = sp.Popen(cmd, stderr=sp.PIPE,stdout=sp.PIPE, shell=True)\n",
    "\n",
    "    out, err = job.communicate()\n",
    "    print('out:', out.decode())\n",
    "    print('err:', err.decode())\n",
    "\n",
    "    print('Created the gzfn: {}'.format(fn + '.gz'))\n",
    "    print('Created the tabix: {}'.format(fn + '.gz.tbi'))\n",
    "\n",
    "def bed_WashU_bedgz(fn, df):\n",
    "    \"\"\"\n",
    "        Convert from a bed dataframe into WashU longrange file \n",
    "        includes bgzip and tabix of the fn. \n",
    "        \n",
    "        Params\n",
    "        -------\n",
    "        fn: str\n",
    "            path to the longrange output file (without gz)\n",
    "            \n",
    "        df: dataframe\n",
    "            columns 1-3 are as expected and column 7 is the p or q-value. \n",
    "            \n",
    "        Output\n",
    "        ------\n",
    "        gzfn: str\n",
    "            path to the longrange with bgzip compression\n",
    "        tabix_fn: str\n",
    "            path to the index of the longrange file\n",
    "            \n",
    "    \"\"\"\n",
    "\n",
    "    # parsing the data into WashU longrage format\n",
    "    data = []\n",
    "    for sr in df.values.tolist():\n",
    "        data.append(sr[0:4])\n",
    "    data = sorted(data, key=lambda x: (x[0], x[1], x[2]))\n",
    "\n",
    "    # writing out the data\n",
    "    with open(fn, 'w') as f:\n",
    "        for line in data:\n",
    "            info = [str(x) for x in line]\n",
    "            info = '\\t'.join(info)\n",
    "            f.write(info + '\\n')\n",
    "            \n",
    "    # run bgzip\n",
    "    cmd = '{} {}'.format(bgzip, fn)\n",
    "    print(cmd)\n",
    "    job = sp.Popen(cmd, stderr=sp.PIPE,stdout=sp.PIPE, shell=True)\n",
    "\n",
    "    out, err = job.communicate()\n",
    "    print('out:', out.decode())\n",
    "    print('err:', err.decode())\n",
    "    \n",
    "    # run tabix\n",
    "    gzfn = fn + '.gz'\n",
    "    cmd = '{} -f {}'.format(tabix, gzfn)\n",
    "    print(cmd)\n",
    "    job = sp.Popen(cmd, stderr=sp.PIPE,stdout=sp.PIPE, shell=True)\n",
    "\n",
    "    out, err = job.communicate()\n",
    "    print('out:', out.decode())\n",
    "    print('err:', err.decode())\n",
    "\n",
    "    print('Created the gzfn: {}'.format(fn + '.gz'))\n",
    "    print('Created the tabix: {}'.format(fn + '.gz.tbi'))\n",
    "\n",
    "def bed_to_WashU_refbed(fn, df):\n",
    "    \"\"\" \n",
    "        Convert from a bed dataframe into WashU longrange file \n",
    "        includes bgzip and tabix of the fn. \n",
    "        \n",
    "        Params\n",
    "        -------\n",
    "        fn: str\n",
    "            path to the longrange output file (without gz)\n",
    "            \n",
    "        df: dataframe\n",
    "            columns 1-3 are as expected and column 7 is the p or q-value. \n",
    "            \n",
    "        Output\n",
    "        ------\n",
    "        gzfn: str\n",
    "            path to the longrange with bgzip compression\n",
    "        tabix_fn: str\n",
    "            path to the index of the longrange file\n",
    "            \n",
    "    \"\"\"\n",
    "\n",
    "    # parsing the data into WashU longrage format\n",
    "    data = df.values.tolist()\n",
    "    data = sorted(data, key=lambda x: (x[0], x[1], x[2]))\n",
    "\n",
    "    # writing out the data\n",
    "    with open(fn, 'w') as f:\n",
    "        for line in data:\n",
    "            info = [str(x) for x in line]\n",
    "            info = '\\t'.join(info)\n",
    "            f.write(info + '\\n')\n",
    "    \n",
    "    # run bgzip\n",
    "    cmd = '{} -f {}'.format(bgzip, fn) \n",
    "    print(cmd)\n",
    "    job = sp.Popen(cmd, stderr=sp.PIPE,stdout=sp.PIPE, shell=True)\n",
    "\n",
    "    out, err = job.communicate()\n",
    "    print('out:', out.decode())\n",
    "    print('err:', err.decode())\n",
    "\n",
    "    # run tabix\n",
    "    gzfn = fn + '.gz'\n",
    "    cmd = '{} {}'.format(tabix, gzfn)\n",
    "    print(cmd)\n",
    "    job = sp.Popen(cmd, stderr=sp.PIPE,stdout=sp.PIPE, shell=True)\n",
    "\n",
    "    out, err = job.communicate()\n",
    "    print('out:', out.decode())\n",
    "    print('err:', err.decode())\n",
    "\n",
    "    print('Created the gzfn: {}'.format(fn + '.gz'))\n",
    "    print('Created the tabix: {}'.format(fn + '.gz.tbi'))\n",
    "\n",
    "\n",
    "# make the refbed link for genes (status: running)\n",
    "final_sg_cols = ['chrom', 'gene_start' ,'gene_end', 'gene_name', 'gene_strand']\n",
    "final_sg_genes = master.loc[(master.has_fithichip_loop == 1), final_sg_cols]\n",
    "\n",
    "final_sg_genes.gene_start = final_sg_genes.gene_start.astype(int)\n",
    "final_sg_genes.gene_end = final_sg_genes.gene_end.astype(int)\n",
    "\n",
    "final_sg_genes = final_sg_genes.loc[~final_sg_genes.duplicated()]\n",
    "final_sg_genes['chr'] = final_sg_genes['chrom'] \n",
    "final_sg_genes['transcript_start'] = final_sg_genes['gene_start']\n",
    "final_sg_genes['transcript_stop'] = final_sg_genes['gene_end']\n",
    "final_sg_genes['translation_start'] = final_sg_genes['gene_start']\n",
    "final_sg_genes['translation_stop'] = final_sg_genes['gene_end']\n",
    "final_sg_genes['strand'] = final_sg_genes['gene_strand']\n",
    "final_sg_genes['gene_name'] = final_sg_genes['gene_name']\n",
    "final_sg_genes['transcript_id'] = final_sg_genes['gene_name']\n",
    "final_sg_genes['type'] = 'coding'\n",
    "final_sg_genes['exon_gene_start'] = final_sg_genes['gene_start']\n",
    "final_sg_genes['exon_stops'] = final_sg_genes['gene_end']\n",
    "refcols = ['chr', 'transcript_start', 'transcript_stop', 'translation_start',\n",
    "           'translation_stop', 'strand', 'gene_name', 'transcript_id',\n",
    "           'type', 'exon_gene_start', 'exon_stops']\n",
    "final_sg_genes = final_sg_genes.loc[:, refcols]\n",
    "sg_genes_fn = os.path.join(outdir, 'gs_genes.bed')\n",
    "bed_to_WashU_refbed(sg_genes_fn, final_sg_genes)\n",
    "\n",
    "\n",
    "# make the longrange link for raw fithichip data (status: running)\n",
    "loop_gz = os.path.abspath(loop_fn.replace('.bed', '_WashU.bed.gz'))\n",
    "loop_tbi = os.path.abspath(loop_fn.replace('.bed', '_WashU.bed.gz.tbi'))\n",
    "loop_gz_link = os.path.join(outdir, os.path.basename(loop_gz))\n",
    "loop_tbi_link = os.path.join(outdir, os.path.basename(loop_tbi))\n",
    "if not os.path.exists(loop_gz_link):\n",
    "    os.link(loop_gz, loop_gz_link)\n",
    "    os.link(loop_tbi, loop_tbi_link)\n",
    "\n",
    "\n",
    "# ## All 5kb Washu Files\n",
    "\n",
    "# make the longrange link for snp-gene fivekb pairs (status: running)\n",
    "#fivekb_lrange = fivekb_lrange[fivekb_lrange.rs_id.notna()].reset_index(drop=True)\n",
    "fivekb_lrange = fivekb_genes.copy()\n",
    "\n",
    "# convert full for viz\n",
    "fivekb_lrange = fivekb_lrange[['chrA', 'startA', 'endA', 'chrB', 'startB', 'endB']]\n",
    "fivekb_lrange.iloc[:, 1] -= 1 \n",
    "fivekb_lrange['score'] = 0.01\n",
    "\n",
    "fivekb_snp_gene_pairs_fn = os.path.join(outdir, '5kb.snp_gene_pairs.bed')\n",
    "bedpe_to_WashU_longrange(fivekb_snp_gene_pairs_fn, fivekb_lrange)\n",
    "\n",
    "# plot all snp-gene pairs with a loop\n",
    "fivekb_gloops_lrange = fivekb_gloops.iloc[:, [11,12,13,14,15,16,18]]\n",
    "\n",
    "fivekb_snp_gene_loops_fn = os.path.join(outdir, '5kb.snp_gene_loops.bed')\n",
    "bedpe_to_WashU_longrange(fivekb_snp_gene_loops_fn, fivekb_gloops_lrange)\n",
    "\n",
    "\n",
    "# ## eQTL WashU Files\n",
    "\n",
    "# make the longrange link for snp-gene eQTL pairs (status: running)\n",
    "eqtl_lrange = master.loc[master.is_eqtl_pair == 1]\n",
    "eqtl_lrange = eqtl_lrange[eqtl_lrange.rs_id.notna()].reset_index(drop=True)\n",
    "\n",
    "# convert full for viz\n",
    "eqtl_snp_gene_pairs = eqtl_lrange[['chrom', 'snp_pos', 'snp_pos', 'chrom', 'tss_start', 'tss_end', 'gene_name']]\n",
    "eqtl_snp_gene_pairs.iloc[:, 1] -= 1 \n",
    "eqtl_snp_gene_pairs['score'] = 0.01\n",
    "\n",
    "eqtl_snp_gene_pairs_fn = os.path.join(outdir, 'eqtl.snp_gene_pairs.bed')\n",
    "bedpe_to_WashU_longrange(eqtl_snp_gene_pairs_fn, eqtl_snp_gene_pairs.iloc[:, [0,1,2,3,4,5,7]])\n",
    "\n",
    "# make the bed for egenes only (status: running)\n",
    "eqtl_genes = eqtl_lrange.loc[:, ['chrom', 'gene_start', 'gene_end', 'gene_name']]\n",
    "eqtl_genes_only_fn = os.path.join(outdir, 'eqtl.genes_only.bed')\n",
    "bed_WashU_bedgz(eqtl_genes_only_fn, eqtl_genes)\n",
    "\n",
    "\n",
    "# In[49]:\n",
    "\n",
    "\n",
    "# In[45]:\n",
    "\n",
    "\n",
    "# make the bed for eSNPs only (status: running)\n",
    "eqtl_snps = eqtl_lrange.loc[:, ['chrom', 'snp_pos', 'snp_pos', 'rs_id']]\n",
    "eqtl_snps.iloc[:, 1] -= 1\n",
    "eqtl_snps = pbt.BedTool.from_dataframe(eqtl_snps)\n",
    "eqtl_snps = eqtl_snps.slop(b=500, g=gs_fn).to_dataframe()\n",
    "eqtl_snps_only_fn = os.path.join(outdir, 'eqtl.snps_only.bed')\n",
    "bed_WashU_bedgz(eqtl_snps_only_fn, eqtl_snps)\n",
    "\n",
    "\n",
    "# In[50]:\n",
    "\n",
    "\n",
    "# In[46]:\n",
    "\n",
    "\n",
    "# # make the longrange link for snp-gene eQTL loops (status: didn't add loop coordinates to master.)\n",
    "# eqtl_loops = master.loc[master.is_eqtl_pair == 1]\n",
    "\n",
    "\n",
    "# ## Coloc WashU Files\n",
    "\n",
    "# In[51]:\n",
    "\n",
    "\n",
    "# In[47]:\n",
    "\n",
    "\n",
    "# make the bed for colocalized SNPs (status: running)\n",
    "final_snps = coloc_sig_df.copy()\n",
    "final_snps = final_snps.loc[~final_snps.duplicated()]\n",
    "final_snps = pbt.BedTool.from_dataframe(final_snps)\n",
    "final_snps = final_snps.slop(b=500, g=gs_fn)\n",
    "final_snps = final_snps.to_dataframe()[0:4]\n",
    "\n",
    "\n",
    "# In[52]:\n",
    "\n",
    "\n",
    "# In[48]:\n",
    "\n",
    "\n",
    "coloc_snps_only_fn = os.path.join(outdir, 'coloc.snps_only.bed')\n",
    "bed_WashU_bedgz(coloc_snps_only_fn, final_snps)\n",
    "\n",
    "\n",
    "# In[53]:\n",
    "\n",
    "\n",
    "# In[49]:\n",
    "\n",
    "\n",
    "# make the bed for colocalized genes (status: running)\n",
    "final_coloc_snp_genes_pairs = master[(master.is_coloc_pair == 1)]\n",
    "final_coloc_genes = final_coloc_snp_genes_pairs[['chrom', 'gene_start' ,'gene_end', 'gene_name']]\n",
    "coloc_genes_only_fn = os.path.join(outdir, 'coloc.genes_only.bed')\n",
    "bed_WashU_bedgz(coloc_genes_only_fn, final_coloc_genes)\n",
    "\n",
    "\n",
    "# In[54]:\n",
    "\n",
    "\n",
    "# In[50]:\n",
    "\n",
    "\n",
    "# make the longrange for colocalized snp-gene pairs (status: running)\n",
    "final_coloc_snp_genes_pairs_out = final_coloc_snp_genes_pairs[['chrom', 'snp_pos', 'snp_pos', 'chrom', \n",
    "                                                               'tss_start', 'tss_end', 'gene_name']]\n",
    "final_coloc_snp_genes_pairs_out.iloc[:, 1] -= 1\n",
    "final_coloc_snp_genes_pairs_out.iloc[:, 6] = 0.01\n",
    "\n",
    "coloc_snp_gene_pairs_fn = os.path.join(outdir, 'coloc.snp_gene_pairs.bed')\n",
    "bedpe_to_WashU_longrange(coloc_snp_gene_pairs_fn, final_coloc_snp_genes_pairs_out)\n",
    "\n",
    "\n",
    "# In[55]:\n",
    "\n",
    "\n",
    "# In[51]:\n",
    "\n",
    "\n",
    "# make the longrange for colocalized snp-gene loops (status: running)\n",
    "final_coloc_snp_genes_loops = fivekb_gloops[(fivekb_gloops[10].isin(final_coloc_snp_genes_pairs['rs_id'])) & \n",
    "              (fivekb_gloops[12].isin(final_coloc_snp_genes_pairs['gene_id']))]\n",
    "final_coloc_snp_genes_loops_out = final_coloc_snp_genes_loops.iloc[:, [14,15,16,17,18,19,21]]\n",
    "\n",
    "coloc_snp_gene_loops_fn = os.path.join(outdir, 'coloc.snp_gene_loops.bed')\n",
    "bedpe_to_WashU_longrange(coloc_snp_gene_loops_fn, final_coloc_snp_genes_loops_out)\n",
    "\n",
    "\n",
    "# In[56]:\n",
    "\n",
    "\n",
    "# In[52]:\n",
    "\n",
    "final_coloc_snp_genes_loops_out\n",
    "\n",
    "\n",
    "# In[57]:\n",
    "\n",
    "\n",
    "# In[53]:\n",
    "\n",
    "\n",
    "# make the loop anchors as bed files (status: developing)\n",
    "left = final_coloc_snp_genes_loops[[14,15,16,6]].T.reset_index(drop=True).T\n",
    "left[3] = 'L-' + left[3] \n",
    "right = final_coloc_snp_genes_loops[[17,18,19,6]].T.reset_index(drop=True).T\n",
    "right[3] = 'R-' + right[3] \n",
    "anchors = pd.concat([left, right], ignore_index=True, axis=0)\n",
    "\n",
    "\n",
    "# In[58]:\n",
    "\n",
    "\n",
    "# In[54]:\n",
    "coloc_anchors_fn = os.path.join(outdir, 'coloc.anchors_only.bed')\n",
    "bed_WashU_bedgz(coloc_anchors_fn, anchors)\n",
    "\n",
    "\n",
    "# # make the hub json file\n",
    "\n",
    "#gwas, cline = coloc_fn.split('/')[5:7]\n",
    "\n",
    "print(\"# make the hub json file\")\n",
    "\n",
    "gene_refbed_json = {'type': 'refbed',\n",
    "                    'filename': os.path.basename(sg_genes_fn) + '.gz',\n",
    "                    'name': 'Gencode V19',\n",
    "                    'showOnHubLoad': True\n",
    "                   }\n",
    "\n",
    "orig_loops_json = {'type': 'longrange',\n",
    "                   'filename': os.path.basename(loop_gz_link),\n",
    "                   'name': 'Original Loops',\n",
    "                   'options': {'displayMode': 'arc', 'color':'red'},\n",
    "                   'showOnHubLoad': True\n",
    "                }\n",
    "\n",
    "# fivekb_snp_gene_pairs_json = {'type': 'longrange',\n",
    "#                    'filename': os.path.basename(fivekb_snp_gene_pairs_fn) + '.gz',\n",
    "#                    'name': '5kb SNP-Gene Pairs',\n",
    "#                    'options': {'displayMode': 'arc', 'color':'purple', 'height': 200},\n",
    "#                    'showOnHubLoad': False\n",
    "#                 }\n",
    "\n",
    "# fivekb_snp_gene_loops_json = {'type': 'longrange',\n",
    "#                    'filename': os.path.basename(fivekb_snp_gene_loops_fn) + '.gz',\n",
    "#                    'name': '5kb SNP-Gene Loops',\n",
    "#                    'options': {'displayMode': 'arc', 'color':'red', 'height': 200},\n",
    "#                    'showOnHubLoad': False\n",
    "#                 }\n",
    "\n",
    "\n",
    "eqtl_snp_gene_pairs_json = {'type': 'longrange',\n",
    "                 'filename': os.path.basename(eqtl_snp_gene_pairs_fn) + '.gz',\n",
    "                 'name': 'eQTL SNP-Gene Pairs',\n",
    "                 'options': {'displayMode': 'arc', 'color':'purple', 'height': 200},\n",
    "                 'showOnHubLoad': True\n",
    "                }\n",
    "\n",
    "\n",
    "eqtl_snps_only_json = {'type': 'bed',\n",
    "                 'filename': os.path.basename(eqtl_snps_only_fn) + '.gz',\n",
    "                 'name': 'eQTL SNPs only',\n",
    "                 'options': {'color':'purple'},\n",
    "                 'showOnHubLoad': True\n",
    "\n",
    "                }\n",
    "\n",
    "\n",
    "eqtl_genes_only_json = {'type': 'bed',\n",
    "                 'filename': os.path.basename(eqtl_genes_only_fn) + '.gz',\n",
    "                 'name': 'eQTL genes only',\n",
    "                 'options': {'color':'purple'},\n",
    "                 'showOnHubLoad': True\n",
    "                }\n",
    "\n",
    "\n",
    "coloc_snp_gene_pairs_json = {'type': 'longrange',\n",
    "                 'filename': os.path.basename(coloc_snp_gene_pairs_fn) + '.gz',\n",
    "                 'name': 'coloc SNP-Gene Pairs',\n",
    "                 'options': {'displayMode': 'arc', 'color':'purple', 'height': 200},\n",
    "                 'showOnHubLoad': True\n",
    "                }\n",
    "\n",
    "coloc_snp_gene_loops_json = {'type': 'longrange',\n",
    "                 'filename': os.path.basename(coloc_snp_gene_loops_fn) + '.gz',\n",
    "                 'name': 'coloc SNP-Gene Loops',\n",
    "                 'options': {'displayMode': 'arc', 'color':'red', 'height': 200},\n",
    "                 'showOnHubLoad': True\n",
    "                }\n",
    "\n",
    "coloc_snps_only_json = {'type': 'bed',\n",
    "                 'filename': os.path.basename(coloc_snps_only_fn) + '.gz',\n",
    "                 'name': 'coloc SNPs only',\n",
    "                 'options': {'color':'purple'},\n",
    "                 'showOnHubLoad': False\n",
    "                }\n",
    "\n",
    "\n",
    "coloc_genes_only_json = {'type': 'bed',\n",
    "                 'filename': os.path.basename(coloc_genes_only_fn) + '.gz',\n",
    "                 'name': 'coloc genes only',\n",
    "                 'options': {'color':'purple'},\n",
    "                 'showOnHubLoad': False\n",
    "                }\n",
    "\n",
    "\n",
    "coloc_anchors_only_json = {'type': 'bed',\n",
    "                 'filename': os.path.basename(coloc_anchors_fn) + '.gz',\n",
    "                 'name': 'coloc loop anchors only',\n",
    "                 'options': {'color':'red'},\n",
    "                 'showOnHubLoad': False\n",
    "                }\n",
    "\n",
    "\n",
    "#hub_json = [orig_loops_json, sg_pairs_json, sg_loops_json, sg_snps_json, sg_genes_json]\n",
    "hub_json = [gene_refbed_json, \n",
    "            #fivekb_snp_gene_pairs_json,\n",
    "            #fivekb_snp_gene_loops_json,\n",
    "            eqtl_snps_only_json,\n",
    "            eqtl_genes_only_json,\n",
    "            eqtl_snp_gene_pairs_json,\n",
    "            coloc_snps_only_json,\n",
    "            coloc_genes_only_json,\n",
    "            coloc_snp_gene_pairs_json, \n",
    "            coloc_anchors_only_json,\n",
    "            coloc_snp_gene_loops_json,\n",
    "            orig_loops_json, \n",
    "           ]\n",
    "\n",
    "hub_json_fn = os.path.join(outdir, 'hub.config.json')\n",
    "with open(hub_json_fn, 'w') as f:\n",
    "    f.write(json.dumps(hub_json, indent=4))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4ff60e56",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "296652de",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bf866646",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "HiChIP DB",
   "language": "python",
   "name": "hichip-db"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
