{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "912288c2",
   "metadata": {},
   "source": [
    "# Make the master table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d0ab3f6d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os \n",
    "import sys\n",
    "import pandas as pd\n",
    "import subprocess as sp\n",
    "import glob\n",
    "os.chdir('/mnt/BioHome/jreyna/jreyna/projects/dchallenge/')\n",
    "hic_python = '/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python'\n",
    "\n",
    "# default values for the command line\n",
    "# sys.argv = [0] * 8\n",
    "# sys.argv[1] =  'results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/'\n",
    "# sys.argv[1] += 'BLUEPRINT_eQTL_Monocyte/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed'\n",
    "# sys.argv[2] = 'results/refs/ensembl/gencode.v19.annotation.bed'\n",
    "# sys.argv[3] = 'results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CM/FitHiChIP_L/FitHiChIP.interactions_FitHiC_Q0.01.bed'\n",
    "# sys.argv[4] = 'results/refs/spp/SPP_D-Challenge_networks.xlsx'\n",
    "# sys.argv[5] = 'results/refs/hg19/hg19.chrom.sizes'\n",
    "# sys.argv[6] = 'results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/Monocyte.txt.gz'\n",
    "# sys.argv[7] = 'results/main/loop_analysis/washU/'\n",
    "\n",
    "# # parsing the commandline arguments\n",
    "# coloc_fn = sys.argv[1]\n",
    "# genes_fn = sys.argv[2]\n",
    "# loop_fn = sys.argv[3]\n",
    "# spp_fn = sys.argv[4]\n",
    "# gs_fn = sys.argv[5]\n",
    "# eqtl_fn = sys.argv[6]\n",
    "# outdir = sys.argv[7]\n",
    "\n",
    "# # setting the output file names\n",
    "# os.makedirs(outdir, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "aeca83ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "# %load ./scripts/loop_analysis/Annotation_SNPs_with_Surround_Genes_Loops.sh\n",
    "#!/usr/bin/env python\n",
    "\n",
    "# In[1]:\n",
    "\n",
    "\n",
    "# Commandline Specification\n",
    "# Input\n",
    "# ----------------------------------------------\n",
    "# 1) colocalization file  x\n",
    "# 2) gencode gene annotation file in bed format x\n",
    "# 3) loop data x \n",
    "# 4) spp network data x \n",
    "# 5) reference chrom sizes x \n",
    "# Output\n",
    "# ----------------------------------------------\n",
    "# 1) snp-gene loop summary file\n",
    "# 2) snp-gene pairs longrange file with index\n",
    "# 3) snp-gene loops longrange file with index\n",
    "\n",
    "# default values for the command line\n",
    "#sys.argv = [0] * 7\n",
    "#sys.argv[1] =  'results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/'\n",
    "#sys.argv[1] += 'BLUEPRINT_eQTL_Monocyte/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed'\n",
    "#sys.argv[2] = 'results/refs/ensembl/gencode.v19.annotation.bed'\n",
    "#sys.argv[3] = 'results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CM/FitHiChIP_L/FitHiChIP.interactions_FitHiC_Q0.01.bed'\n",
    "#sys.argv[4] = 'results/refs/spp/SPP_D-Challenge_networks.xlsx'\n",
    "#sys.argv[5] = 'results/refs/hg19/hg19.chrom.sizes'\n",
    "#sys.argv[6] = 'results/main/loop_analysis/washU/'\n",
    "script = \"scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops.py\"\n",
    "script = \"scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py\"\n",
    "gs = \"results/refs/hg19/hg19.chrom.sizes\"\n",
    "gencode = \"results/refs/ensembl/gencode.v19.annotation.bed\"\n",
    "spp = \"results/refs/spp/SPP_D-Challenge_networks.xlsx\"\n",
    "\n",
    "# template file names for colocalization, hichip, eqtl and outdir\n",
    "coloc_tmp = \"results/main/2021_Nikhil_eQTL/Results/Colocalization/{gwas}/{eqtl}_{cline}/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed\"\n",
    "fithichip_tmp = \"results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/{cline}/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed\"\n",
    "eqtl_tmp = \"results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/{eqtl}/{cline}.txt.gz\"\n",
    "outdir_tmp = \"results/main/loop_analysis/Coloc_Approach/{gwas}_test/{dice_cline}/{eqtl_study}/{eqtl_cline}/\"\n",
    "\n",
    "# getting a list of DICE clines with HiChIP data\n",
    "fithichip_clines = glob.glob(\"results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/*\")\n",
    "fithichip_clines = [x.split('/')[-1] for x in fithichip_clines]\n",
    "\n",
    "# eqtl mapper\n",
    "eqtl_mapper = pd.read_table('results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/eqtl.cline.mapper.study.tsv')\n",
    "eqtl_mapper.sort_values(['will_analyze', 'study', 'dice_shortname'], ascending=[False, True, True], inplace=True)\n",
    "eqtl_mapper[['study', 'dice_shortname', 'will_analyze', 'eqtl_file']]\n",
    "eqtl_mapper = eqtl_mapper.loc[eqtl_mapper['will_analyze'] == 'Yes']\n",
    "eqtl_mapper = eqtl_mapper.loc[eqtl_mapper.dice_shortname.isin(fithichip_clines)]\n",
    "eqtl_clines = eqtl_mapper['eqtl_file'].str.split('/')\n",
    "eqtl_clines = [x[-1].split('.')[0] for x in eqtl_clines]\n",
    "eqtl_mapper['eqtl_cline'] = eqtl_clines\n",
    "eqtl_mapper = eqtl_mapper[['study', 'dice_shortname', 'eqtl_cline', 'eqtl_file']]\n",
    "\n",
    "# pchic dataset\n",
    "pchic_fn = 'results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c0859bcd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study</th>\n",
       "      <th>dice_shortname</th>\n",
       "      <th>eqtl_cline</th>\n",
       "      <th>eqtl_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>CM</td>\n",
       "      <td>Monocyte</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>NCM</td>\n",
       "      <td>Monocyte</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>TFH</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>TH1</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>TH17</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            study dice_shortname eqtl_cline  \\\n",
       "0  BLUEPRINT_eQTL             CM   Monocyte   \n",
       "1  BLUEPRINT_eQTL            NCM   Monocyte   \n",
       "2  BLUEPRINT_eQTL            TFH     T-cell   \n",
       "3  BLUEPRINT_eQTL            TH1     T-cell   \n",
       "4  BLUEPRINT_eQTL           TH17     T-cell   \n",
       "\n",
       "                                           eqtl_file  \n",
       "0  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "1  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "2  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "3  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "4  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eqtl_mapper.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "c414df46",
   "metadata": {},
   "outputs": [],
   "source": [
    "# get gwas runs\n",
    "gwas_runs = [os.path.basename(x) for x in glob.glob('results/main/2021_Nikhil_eQTL/Data/T1D_GWAS/*')]\n",
    "gwas_runs = gwas_runs[2:3]\n",
    "\n",
    "# get eqtl runs \n",
    "eqtl_runs = [x for x in glob.glob('results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/*/*')]\n",
    "eqtl_runs = [x for x in eqtl_runs if x != 'README' and 'eQTL' in x]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "2feed471",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study</th>\n",
       "      <th>dice_shortname</th>\n",
       "      <th>eqtl_cline</th>\n",
       "      <th>eqtl_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>CM</td>\n",
       "      <td>Monocyte</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>NCM</td>\n",
       "      <td>Monocyte</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>TFH</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>TH1</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>TH17</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            study dice_shortname eqtl_cline  \\\n",
       "0  BLUEPRINT_eQTL             CM   Monocyte   \n",
       "1  BLUEPRINT_eQTL            NCM   Monocyte   \n",
       "2  BLUEPRINT_eQTL            TFH     T-cell   \n",
       "3  BLUEPRINT_eQTL            TH1     T-cell   \n",
       "4  BLUEPRINT_eQTL           TH17     T-cell   \n",
       "\n",
       "                                           eqtl_file  \n",
       "0  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "1  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "2  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "3  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "4  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eqtl_mapper.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "c3518121",
   "metadata": {},
   "outputs": [],
   "source": [
    "# qscripts_dir = 'results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton/qjobs/'\n",
    "# os.makedirs(qscripts_dir, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "a0249d14",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study</th>\n",
       "      <th>dice_shortname</th>\n",
       "      <th>eqtl_cline</th>\n",
       "      <th>eqtl_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>CM</td>\n",
       "      <td>Monocyte</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>NCM</td>\n",
       "      <td>Monocyte</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>TFH</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>TH1</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>TH17</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>TH2</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>THSTAR</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>TREGMEM</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>BLUEPRINT_eQTL</td>\n",
       "      <td>TREGNAIVE</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>DICE_eQTL</td>\n",
       "      <td>CD4N</td>\n",
       "      <td>CD4_NAIVE</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>DICE_eQTL</td>\n",
       "      <td>CD4N</td>\n",
       "      <td>CD4_N_STIM</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>DICE_eQTL</td>\n",
       "      <td>CD8N</td>\n",
       "      <td>CD8_NAIVE</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>DICE_eQTL</td>\n",
       "      <td>CD8N</td>\n",
       "      <td>CD8_N_STIM</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>DICE_eQTL</td>\n",
       "      <td>CM</td>\n",
       "      <td>CLASSICAL_MONOCYTES</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>DICE_eQTL</td>\n",
       "      <td>NB</td>\n",
       "      <td>B_NAIVE</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>DICE_eQTL</td>\n",
       "      <td>NCM</td>\n",
       "      <td>NONCLASSICAL_MONOCYTES</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>DICE_eQTL</td>\n",
       "      <td>NK</td>\n",
       "      <td>NK_CD16POS</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>DICE_eQTL</td>\n",
       "      <td>TFH</td>\n",
       "      <td>TFH</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>DICE_eQTL</td>\n",
       "      <td>TH1</td>\n",
       "      <td>TH1</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>DICE_eQTL</td>\n",
       "      <td>TH17</td>\n",
       "      <td>TH17</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>DICE_eQTL</td>\n",
       "      <td>TH2</td>\n",
       "      <td>TH2</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>DICE_eQTL</td>\n",
       "      <td>THSTAR</td>\n",
       "      <td>TH1-17</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>DICE_eQTL</td>\n",
       "      <td>TREGMEM</td>\n",
       "      <td>TREG_MEMORY</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>DICE_eQTL</td>\n",
       "      <td>TREGNAIVE</td>\n",
       "      <td>TREG_NAIVE</td>\n",
       "      <td>results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             study dice_shortname              eqtl_cline  \\\n",
       "0   BLUEPRINT_eQTL             CM                Monocyte   \n",
       "1   BLUEPRINT_eQTL            NCM                Monocyte   \n",
       "2   BLUEPRINT_eQTL            TFH                  T-cell   \n",
       "3   BLUEPRINT_eQTL            TH1                  T-cell   \n",
       "4   BLUEPRINT_eQTL           TH17                  T-cell   \n",
       "5   BLUEPRINT_eQTL            TH2                  T-cell   \n",
       "6   BLUEPRINT_eQTL         THSTAR                  T-cell   \n",
       "7   BLUEPRINT_eQTL        TREGMEM                  T-cell   \n",
       "8   BLUEPRINT_eQTL      TREGNAIVE                  T-cell   \n",
       "23       DICE_eQTL           CD4N               CD4_NAIVE   \n",
       "24       DICE_eQTL           CD4N              CD4_N_STIM   \n",
       "25       DICE_eQTL           CD8N               CD8_NAIVE   \n",
       "26       DICE_eQTL           CD8N              CD8_N_STIM   \n",
       "27       DICE_eQTL             CM     CLASSICAL_MONOCYTES   \n",
       "22       DICE_eQTL             NB                 B_NAIVE   \n",
       "29       DICE_eQTL            NCM  NONCLASSICAL_MONOCYTES   \n",
       "28       DICE_eQTL             NK              NK_CD16POS   \n",
       "30       DICE_eQTL            TFH                     TFH   \n",
       "32       DICE_eQTL            TH1                     TH1   \n",
       "33       DICE_eQTL           TH17                    TH17   \n",
       "34       DICE_eQTL            TH2                     TH2   \n",
       "31       DICE_eQTL         THSTAR                  TH1-17   \n",
       "35       DICE_eQTL        TREGMEM             TREG_MEMORY   \n",
       "36       DICE_eQTL      TREGNAIVE              TREG_NAIVE   \n",
       "\n",
       "                                            eqtl_file  \n",
       "0   results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "1   results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "2   results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "3   results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "4   results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "5   results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "6   results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "7   results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "8   results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "23  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "24  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "25  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "26  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "27  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "22  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "29  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "28  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "30  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "32  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "33  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "34  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "31  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "35  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  \n",
       "36  results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_s...  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eqtl_mapper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "ee0e8af1",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/BLUEPRINT_eQTL_Monocyte/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CM/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/Monocyte.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv Monocyte results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/\n",
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/BLUEPRINT_eQTL_Monocyte/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CM/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/Monocyte.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv Monocyte results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/\n",
      "There are 15 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 635 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 84298 significant loops.\n",
      "There are 35 SNP-Gene pairs with a loop.\n",
      "Master is starting with 635 snp-gene pairs.\n",
      "There are 7716 eQTLs.\n",
      "After outer merging with eqtls master has 8336 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 8336 entries.\n",
      "There are 36 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/BLUEPRINT_eQTL/Monocyte/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/BLUEPRINT_eQTL_Monocyte/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/NCM/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/Monocyte.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv Monocyte results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/BLUEPRINT_eQTL_Monocyte/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/NCM/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/Monocyte.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv Monocyte results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/\n",
      "There are 15 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 635 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 103342 significant loops.\n",
      "There are 85 SNP-Gene pairs with a loop.\n",
      "Master is starting with 635 snp-gene pairs.\n",
      "There are 7716 eQTLs.\n",
      "After outer merging with eqtls master has 8336 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 8336 entries.\n",
      "There are 86 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/BLUEPRINT_eQTL/Monocyte/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/BLUEPRINT_eQTL_T-cell/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TFH/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/T-cell.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv T-cell results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/BLUEPRINT_eQTL_T-cell/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TFH/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/T-cell.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv T-cell results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/\n",
      "There are 16 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 613 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 46172 significant loops.\n",
      "There are 28 SNP-Gene pairs with a loop.\n",
      "Master is starting with 613 snp-gene pairs.\n",
      "There are 7211 eQTLs.\n",
      "After outer merging with eqtls master has 7808 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 7808 entries.\n",
      "There are 30 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "sys:1: PerformanceWarning: indexing past lexsort depth may impact performance.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/BLUEPRINT_eQTL/T-cell/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/BLUEPRINT_eQTL_T-cell/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TH1/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/T-cell.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv T-cell results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/BLUEPRINT_eQTL_T-cell/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TH1/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/T-cell.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv T-cell results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/\n",
      "There are 16 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 613 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 63241 significant loops.\n",
      "There are 61 SNP-Gene pairs with a loop.\n",
      "Master is starting with 613 snp-gene pairs.\n",
      "There are 7211 eQTLs.\n",
      "After outer merging with eqtls master has 7808 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 7808 entries.\n",
      "There are 63 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "sys:1: PerformanceWarning: indexing past lexsort depth may impact performance.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/BLUEPRINT_eQTL/T-cell/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/BLUEPRINT_eQTL_T-cell/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TH17/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/T-cell.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv T-cell results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/BLUEPRINT_eQTL_T-cell/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TH17/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/T-cell.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv T-cell results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/\n",
      "There are 16 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 613 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 76270 significant loops.\n",
      "There are 70 SNP-Gene pairs with a loop.\n",
      "Master is starting with 613 snp-gene pairs.\n",
      "There are 7211 eQTLs.\n",
      "After outer merging with eqtls master has 7808 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 7808 entries.\n",
      "There are 71 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "sys:1: PerformanceWarning: indexing past lexsort depth may impact performance.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/BLUEPRINT_eQTL/T-cell/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/BLUEPRINT_eQTL_T-cell/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TH2/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/T-cell.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv T-cell results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/BLUEPRINT_eQTL_T-cell/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TH2/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/T-cell.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv T-cell results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/\n",
      "There are 16 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 613 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 58115 significant loops.\n",
      "There are 37 SNP-Gene pairs with a loop.\n",
      "Master is starting with 613 snp-gene pairs.\n",
      "There are 7211 eQTLs.\n",
      "After outer merging with eqtls master has 7808 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 7808 entries.\n",
      "There are 43 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "sys:1: PerformanceWarning: indexing past lexsort depth may impact performance.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/BLUEPRINT_eQTL/T-cell/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/BLUEPRINT_eQTL_T-cell/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/THSTAR/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/T-cell.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv T-cell results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/BLUEPRINT_eQTL_T-cell/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/THSTAR/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/T-cell.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv T-cell results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/\n",
      "There are 16 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 613 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 50757 significant loops.\n",
      "There are 34 SNP-Gene pairs with a loop.\n",
      "Master is starting with 613 snp-gene pairs.\n",
      "There are 7211 eQTLs.\n",
      "After outer merging with eqtls master has 7808 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 7808 entries.\n",
      "There are 36 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "sys:1: PerformanceWarning: indexing past lexsort depth may impact performance.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/BLUEPRINT_eQTL/T-cell/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/BLUEPRINT_eQTL_T-cell/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TREGMEM/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/T-cell.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv T-cell results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/BLUEPRINT_eQTL_T-cell/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TREGMEM/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/T-cell.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv T-cell results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/\n",
      "There are 16 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 613 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 56567 significant loops.\n",
      "There are 56 SNP-Gene pairs with a loop.\n",
      "Master is starting with 613 snp-gene pairs.\n",
      "There are 7211 eQTLs.\n",
      "After outer merging with eqtls master has 7808 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 7808 entries.\n",
      "There are 57 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "sys:1: PerformanceWarning: indexing past lexsort depth may impact performance.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/BLUEPRINT_eQTL/T-cell/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/BLUEPRINT_eQTL_T-cell/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TREGNAIVE/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/T-cell.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv T-cell results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/BLUEPRINT_eQTL_T-cell/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TREGNAIVE/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/BLUEPRINT_eQTL/T-cell.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv T-cell results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/\n",
      "There are 16 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 613 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 51613 significant loops.\n",
      "There are 55 SNP-Gene pairs with a loop.\n",
      "Master is starting with 613 snp-gene pairs.\n",
      "There are 7211 eQTLs.\n",
      "After outer merging with eqtls master has 7808 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 7808 entries.\n",
      "There are 57 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "sys:1: PerformanceWarning: indexing past lexsort depth may impact performance.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/BLUEPRINT_eQTL/T-cell/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_CD4_NAIVE/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CD4N/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/CD4_NAIVE.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv CD4_NAIVE results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_CD4_NAIVE/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CD4N/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/CD4_NAIVE.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv CD4_NAIVE results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/\n",
      "There are 9 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 393 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 114421 significant loops.\n",
      "There are 8 SNP-Gene pairs with a loop.\n",
      "Master is starting with 393 snp-gene pairs.\n",
      "There are 2504 eQTLs.\n",
      "After outer merging with eqtls master has 2888 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 2888 entries.\n",
      "There are 9 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/lib/python3.6/site-packages/pybedtools/bedtool.py:3685: UserWarning: Default names for filetype bed are:\n",
      "['chrom', 'start', 'end', 'name', 'score', 'strand', 'thickStart', 'thickEnd', 'itemRgb', 'blockCount', 'blockSizes', 'blockStarts']\n",
      "but file has 16 fields; you can supply custom names with the `names` kwarg\n",
      "  \"`names` kwarg\" % (self.file_type, _names, self.field_count())\n",
      "sys:1: PerformanceWarning: indexing past lexsort depth may impact performance.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_NAIVE/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_CD4_N_STIM/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CD4N/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/CD4_N_STIM.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv CD4_N_STIM results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_CD4_N_STIM/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CD4N/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/CD4_N_STIM.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv CD4_N_STIM results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/\n",
      "There are 5 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 247 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 114421 significant loops.\n",
      "There are 25 SNP-Gene pairs with a loop.\n",
      "Master is starting with 247 snp-gene pairs.\n",
      "There are 1883 eQTLs.\n",
      "After outer merging with eqtls master has 2126 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 2126 entries.\n",
      "There are 25 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD4N/DICE_eQTL/CD4_N_STIM/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_CD8_NAIVE/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CD8N/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/CD8_NAIVE.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv CD8_NAIVE results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_CD8_NAIVE/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CD8N/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/CD8_NAIVE.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv CD8_NAIVE results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/\n",
      "There are 8 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 355 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 84599 significant loops.\n",
      "There are 14 SNP-Gene pairs with a loop.\n",
      "Master is starting with 355 snp-gene pairs.\n",
      "There are 2373 eQTLs.\n",
      "After outer merging with eqtls master has 2722 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 2722 entries.\n",
      "There are 14 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/lib/python3.6/site-packages/pybedtools/bedtool.py:3685: UserWarning: Default names for filetype bed are:\n",
      "['chrom', 'start', 'end', 'name', 'score', 'strand', 'thickStart', 'thickEnd', 'itemRgb', 'blockCount', 'blockSizes', 'blockStarts']\n",
      "but file has 16 fields; you can supply custom names with the `names` kwarg\n",
      "  \"`names` kwarg\" % (self.file_type, _names, self.field_count())\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_NAIVE/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_CD8_N_STIM/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CD8N/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/CD8_N_STIM.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv CD8_N_STIM results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_CD8_N_STIM/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CD8N/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/CD8_N_STIM.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv CD8_N_STIM results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/\n",
      "There are 7 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 278 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 84599 significant loops.\n",
      "There are 13 SNP-Gene pairs with a loop.\n",
      "Master is starting with 278 snp-gene pairs.\n",
      "There are 2001 eQTLs.\n",
      "After outer merging with eqtls master has 2273 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 2273 entries.\n",
      "There are 13 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CD8N/DICE_eQTL/CD8_N_STIM/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_CLASSICAL_MONOCYTES/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CM/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/CLASSICAL_MONOCYTES.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv CLASSICAL_MONOCYTES results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_CLASSICAL_MONOCYTES/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/CM/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/CLASSICAL_MONOCYTES.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv CLASSICAL_MONOCYTES results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/\n",
      "There are 7 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 319 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 84298 significant loops.\n",
      "There are 26 SNP-Gene pairs with a loop.\n",
      "Master is starting with 319 snp-gene pairs.\n",
      "There are 2252 eQTLs.\n",
      "After outer merging with eqtls master has 2564 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 2564 entries.\n",
      "There are 27 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/lib/python3.6/site-packages/pybedtools/bedtool.py:3685: UserWarning: Default names for filetype bed are:\n",
      "['chrom', 'start', 'end', 'name', 'score', 'strand', 'thickStart', 'thickEnd', 'itemRgb', 'blockCount', 'blockSizes', 'blockStarts']\n",
      "but file has 16 fields; you can supply custom names with the `names` kwarg\n",
      "  \"`names` kwarg\" % (self.file_type, _names, self.field_count())\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/CM/DICE_eQTL/CLASSICAL_MONOCYTES/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_B_NAIVE/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/NB/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/B_NAIVE.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv B_NAIVE results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_B_NAIVE/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/NB/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/B_NAIVE.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv B_NAIVE results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/\n",
      "There are 11 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 431 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 128288 significant loops.\n",
      "There are 59 SNP-Gene pairs with a loop.\n",
      "Master is starting with 431 snp-gene pairs.\n",
      "There are 1826 eQTLs.\n",
      "After outer merging with eqtls master has 2247 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 2247 entries.\n",
      "There are 60 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/lib/python3.6/site-packages/pybedtools/bedtool.py:3685: UserWarning: Default names for filetype bed are:\n",
      "['chrom', 'start', 'end', 'name', 'score', 'strand', 'thickStart', 'thickEnd', 'itemRgb', 'blockCount', 'blockSizes', 'blockStarts']\n",
      "but file has 16 fields; you can supply custom names with the `names` kwarg\n",
      "  \"`names` kwarg\" % (self.file_type, _names, self.field_count())\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NB/DICE_eQTL/B_NAIVE/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_NONCLASSICAL_MONOCYTES/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/NCM/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/NONCLASSICAL_MONOCYTES.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv NONCLASSICAL_MONOCYTES results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_NONCLASSICAL_MONOCYTES/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/NCM/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/NONCLASSICAL_MONOCYTES.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv NONCLASSICAL_MONOCYTES results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/\n",
      "There are 7 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 272 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 103342 significant loops.\n",
      "There are 36 SNP-Gene pairs with a loop.\n",
      "Master is starting with 272 snp-gene pairs.\n",
      "There are 2104 eQTLs.\n",
      "After outer merging with eqtls master has 2369 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 2369 entries.\n",
      "There are 36 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NCM/DICE_eQTL/NONCLASSICAL_MONOCYTES/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_NK_CD16POS/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/NK/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/NK_CD16POS.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv NK_CD16POS results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_NK_CD16POS/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/NK/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/NK_CD16POS.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv NK_CD16POS results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/\n",
      "There are 9 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 402 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 129890 significant loops.\n",
      "There are 51 SNP-Gene pairs with a loop.\n",
      "Master is starting with 402 snp-gene pairs.\n",
      "There are 1882 eQTLs.\n",
      "After outer merging with eqtls master has 2275 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 2275 entries.\n",
      "There are 51 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/NK/DICE_eQTL/NK_CD16POS/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_TFH/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TFH/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/TFH.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv TFH results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_TFH/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TFH/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/TFH.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv TFH results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/\n",
      "There are 8 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 340 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 46172 significant loops.\n",
      "There are 11 SNP-Gene pairs with a loop.\n",
      "Master is starting with 340 snp-gene pairs.\n",
      "There are 2053 eQTLs.\n",
      "After outer merging with eqtls master has 2385 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 2385 entries.\n",
      "There are 12 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TFH/DICE_eQTL/TFH/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_TH1/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TH1/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/TH1.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv TH1 results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_TH1/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TH1/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/TH1.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv TH1 results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/\n",
      "There are 9 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 388 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 63241 significant loops.\n",
      "There are 20 SNP-Gene pairs with a loop.\n",
      "Master is starting with 388 snp-gene pairs.\n",
      "There are 1902 eQTLs.\n",
      "After outer merging with eqtls master has 2281 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 2281 entries.\n",
      "There are 20 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH1/DICE_eQTL/TH1/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_TH17/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TH17/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/TH17.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv TH17 results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_TH17/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TH17/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/TH17.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv TH17 results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/\n",
      "There are 7 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 281 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 76270 significant loops.\n",
      "There are 33 SNP-Gene pairs with a loop.\n",
      "Master is starting with 281 snp-gene pairs.\n",
      "There are 2520 eQTLs.\n",
      "After outer merging with eqtls master has 2794 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 2794 entries.\n",
      "There are 34 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH17/DICE_eQTL/TH17/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_TH2/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TH2/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/TH2.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv TH2 results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_TH2/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TH2/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/TH2.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv TH2 results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/\n",
      "There are 7 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 291 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 58115 significant loops.\n",
      "There are 10 SNP-Gene pairs with a loop.\n",
      "Master is starting with 291 snp-gene pairs.\n",
      "There are 2160 eQTLs.\n",
      "After outer merging with eqtls master has 2444 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 2444 entries.\n",
      "There are 10 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TH2/DICE_eQTL/TH2/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_TH1-17/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/THSTAR/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/TH1-17.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv TH1-17 results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_TH1-17/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/THSTAR/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/TH1-17.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv TH1-17 results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/\n",
      "There are 5 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 204 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 50757 significant loops.\n",
      "There are 20 SNP-Gene pairs with a loop.\n",
      "Master is starting with 204 snp-gene pairs.\n",
      "There are 2187 eQTLs.\n",
      "After outer merging with eqtls master has 2386 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 2386 entries.\n",
      "There are 20 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/THSTAR/DICE_eQTL/TH1-17/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_TREG_MEMORY/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TREGMEM/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/TREG_MEMORY.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv TREG_MEMORY results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_TREG_MEMORY/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TREGMEM/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/TREG_MEMORY.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv TREG_MEMORY results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/\n",
      "There are 10 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 407 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 56567 significant loops.\n",
      "There are 31 SNP-Gene pairs with a loop.\n",
      "Master is starting with 407 snp-gene pairs.\n",
      "There are 1971 eQTLs.\n",
      "After outer merging with eqtls master has 2368 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 2368 entries.\n",
      "There are 31 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGMEM/DICE_eQTL/TREG_MEMORY/\n",
      "\n",
      "/mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_TREG_NAIVE/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TREGNAIVE/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/TREG_NAIVE.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv TREG_NAIVE results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now running: /mnt/BioHome/jreyna/software/anaconda3/envs/hic_tls/bin/python scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py results/main/2021_Nikhil_eQTL/Results/Colocalization/T1D_34012112_Gaulton/DICE_eQTL_TREG_NAIVE/FINAL_Summary_Coloc_Gene_SNP_Pairs.bed results/refs/ensembl/gencode.v19.annotation.bed results/main/2021_Nikhil_eQTL/Data/FitHiChIP_Loops/TREGNAIVE/FitHiChIP_S/FitHiChIP.interactions_FitHiC_Q0.01.bed results/refs/spp/SPP_D-Challenge_networks.xlsx results/refs/hg19/hg19.chrom.sizes results/main/2021_Nikhil_eQTL/Data/eqtl_sqtl_summ_stats/DICE_eQTL/TREG_NAIVE.txt.gz results/main/pc_hic/data/PCHiC_peak_matrix_cutoff5.tsv TREG_NAIVE results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/\n",
      "There are 6 colocalized SNP-gene pairs\n",
      "There are 57774 genes in this GTF-derived file.\n",
      "There are 321 colocalized snp-gene pairs within +/- 5kb.\n",
      "FitHiChIP found 51613 significant loops.\n",
      "There are 23 SNP-Gene pairs with a loop.\n",
      "Master is starting with 321 snp-gene pairs.\n",
      "There are 2105 eQTLs.\n",
      "After outer merging with eqtls master has 2421 snp-gene pairs.\n",
      "After left merging master with the colocalization table there are 2421 entries.\n",
      "There are 23 SNP-Gene loops.\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/gs_genes.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/gs_genes.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/gs_genes.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/gs_genes.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/5kb.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/5kb.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/5kb.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/5kb.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/5kb.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/5kb.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/5kb.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/5kb.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/eqtl.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/eqtl.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/eqtl.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/eqtl.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/eqtl.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/eqtl.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/eqtl.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/eqtl.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/eqtl.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/eqtl.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/eqtl.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/eqtl.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.snps_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.snps_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.snps_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.snps_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.genes_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.genes_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.genes_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.genes_only.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.snp_gene_pairs.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.snp_gene_pairs.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.snp_gene_pairs.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.snp_gene_pairs.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.snp_gene_loops.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.snp_gene_loops.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.snp_gene_loops.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.snp_gene_loops.bed.gz.tbi\n",
      "/mnt/BioApps/tabix/tabix-0.2.6/bgzip results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.anchors_only.bed\n",
      "out: \n",
      "err: \n",
      "/mnt/BioApps/tabix/tabix-0.2.6/tabix -f results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.anchors_only.bed.gz\n",
      "out: \n",
      "err: \n",
      "Created the gzfn: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.anchors_only.bed.gz\n",
      "Created the tabix: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/coloc.anchors_only.bed.gz.tbi\n",
      "# make the hub json file\n",
      "\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1596: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "sys:1: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "scripts/loop_analysis/Annotate_Coloc_SNPs_with_Nearby_Genes_Loops_V2.py:935: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  eqtl_snp_gene_pairs['score'] = 0.01\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(ilocs[0], value)\n",
      "/mnt/BioHome/jreyna/.local/lib/python3.6/site-packages/pandas/core/indexing.py:1763: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  isetter(loc, value)\n",
      "\n",
      "Results can now be found at: results/main/loop_analysis/Coloc_Approach/T1D_34012112_Gaulton_test/TREGNAIVE/DICE_eQTL/TREG_NAIVE/\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for gwas in gwas_runs:\n",
    "    for i, sr in eqtl_mapper.iterrows():\n",
    "        \n",
    "        coloc = coloc_tmp.format(gwas=gwas, eqtl=sr.study, cline=sr.eqtl_cline)\n",
    "        fithichip = fithichip_tmp.format(cline=sr.dice_shortname)\n",
    "        outdir = outdir_tmp.format(gwas=gwas, \n",
    "                                   dice_cline=sr.dice_shortname,\n",
    "                                   eqtl_cline=sr.eqtl_cline,\n",
    "                                   eqtl_study=sr.study)\n",
    "        \n",
    "        \n",
    "        # Changing the celltype naming for BLUEPRINT monocytes in order to include\n",
    "        # Chicago scores from PC HiC\n",
    "        if sr.eqtl_cline == 'Monocyte':\n",
    "            celltype = 'CLASSICAL_MONOCYTES'\n",
    "        else:\n",
    "            celltype = sr.eqtl_cline\n",
    "            \n",
    "        cmd = \"{} {} {} {} {} {} {} {} {} {} {}\".format(hic_python, script, coloc, gencode, \n",
    "                                                  fithichip, spp, gs, sr.eqtl_file,\n",
    "                                                  pchic_fn, sr.eqtl_cline, outdir)  \n",
    "        \n",
    "        print(cmd)\n",
    "        \n",
    "        if not os.path.exists(outdir):\n",
    "            \n",
    "            #             script_fn = '{}_{}_{}_{}.qscript'.format(gwas, sr.dice_shortname, sr.eqtl_cline, sr.study)\n",
    "            #             script_fn = os.path.join(qscripts_dir, script_fn)\n",
    "            #             with open(script_fn, 'w') as fw:\n",
    "            #                 fw.write(cmd)\n",
    "                \n",
    "            job = sp.Popen(cmd, shell=True, stderr=sp.PIPE, stdout=sp.PIPE)\n",
    "            out, err = job.communicate()\n",
    "            print('Now running: {}'.format(cmd))\n",
    "            print(out.decode())\n",
    "            print(err.decode())\n",
    "            print('Results can now be found at: {}'.format(outdir))\n",
    "        else:\n",
    "            msg = 'Previously ran {}'.format(outdir)\n",
    "            print(msg)\n",
    "            \n",
    "        print()\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e058e778",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6d830632",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a49f6e5c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8f73d949",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5111c31a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c02b68c0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "368d4e80",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "56fcf006",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3cd89a5c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "496e52d2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "223dd4ba",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fd2b9274",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f2ac27f6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "HiC",
   "language": "python",
   "name": "hic_tls"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
