{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "7c53ff12",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import subprocess\n",
    "import pandas as pd \n",
    "os.chdir('/mnt/BioHome/jreyna/jreyna/projects/dchallenge/')\n",
    "outdir = 'results/refs/ensembl/'\n",
    "os.makedirs(outdir, exist_ok=True)\n",
    "\n",
    "fn = os.path.join(outdir, 'Homo_sapiens.GRCh37.87.gtf.gz')\n",
    "if not os.path.exists(fn):\n",
    "    url = 'http://ftp.ensembl.org/pub/grch37/release-100/gtf/homo_sapiens/Homo_sapiens.GRCh37.87.gtf.gz'\n",
    "    cmd = 'wget -O {} {}'.format(fn, url)\n",
    "    output = subprocess.check_output(cmd, shell=True)\n",
    "\n",
    "gtf = pd.read_table(fn, header=None, skiprows=5)\n",
    "\n",
    "# extract the main columns\n",
    "parsed = gtf.iloc[:, [0,3,4,8,6,2]]\n",
    "\n",
    "# get the gene id and gene name\n",
    "parsed['gene_id'] = parsed[8].str.extract('gene_id \"(ENSG[0-9]+)\"')\n",
    "parsed['gene_name'] = parsed[8].str.extract('gene_name \"([A-Za-z0-9]+)\"')\n",
    "\n",
    "# final data cleanup\n",
    "parsed = parsed.iloc[:, [0,1,2,4,5,6,7]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "241f7d96",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save the final output\n",
    "bed = os.path.join(outdir, 'Homo_sapiens.GRCh37.87.bed')\n",
    "parsed.to_csv(bed, sep='\\t', header=None, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "20174e40",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>6</th>\n",
       "      <th>2</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2181196</th>\n",
       "      <td>17</td>\n",
       "      <td>44107282</td>\n",
       "      <td>44302733</td>\n",
       "      <td>-</td>\n",
       "      <td>gene</td>\n",
       "      <td>ENSG00000120071</td>\n",
       "      <td>KANSL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2181197</th>\n",
       "      <td>17</td>\n",
       "      <td>44107282</td>\n",
       "      <td>44123487</td>\n",
       "      <td>-</td>\n",
       "      <td>transcript</td>\n",
       "      <td>ENSG00000120071</td>\n",
       "      <td>KANSL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2181198</th>\n",
       "      <td>17</td>\n",
       "      <td>44117068</td>\n",
       "      <td>44123487</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>ENSG00000120071</td>\n",
       "      <td>KANSL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2181199</th>\n",
       "      <td>17</td>\n",
       "      <td>44116393</td>\n",
       "      <td>44116581</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>ENSG00000120071</td>\n",
       "      <td>KANSL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2181200</th>\n",
       "      <td>17</td>\n",
       "      <td>44115904</td>\n",
       "      <td>44116052</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>ENSG00000120071</td>\n",
       "      <td>KANSL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2181463</th>\n",
       "      <td>17</td>\n",
       "      <td>44249483</td>\n",
       "      <td>44249509</td>\n",
       "      <td>-</td>\n",
       "      <td>CDS</td>\n",
       "      <td>ENSG00000120071</td>\n",
       "      <td>KANSL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2181464</th>\n",
       "      <td>17</td>\n",
       "      <td>44249507</td>\n",
       "      <td>44249509</td>\n",
       "      <td>-</td>\n",
       "      <td>start_codon</td>\n",
       "      <td>ENSG00000120071</td>\n",
       "      <td>KANSL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2181465</th>\n",
       "      <td>17</td>\n",
       "      <td>44301927</td>\n",
       "      <td>44302080</td>\n",
       "      <td>-</td>\n",
       "      <td>five_prime_utr</td>\n",
       "      <td>ENSG00000120071</td>\n",
       "      <td>KANSL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2181466</th>\n",
       "      <td>17</td>\n",
       "      <td>44301037</td>\n",
       "      <td>44301170</td>\n",
       "      <td>-</td>\n",
       "      <td>five_prime_utr</td>\n",
       "      <td>ENSG00000120071</td>\n",
       "      <td>KANSL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2181467</th>\n",
       "      <td>17</td>\n",
       "      <td>44249510</td>\n",
       "      <td>44249598</td>\n",
       "      <td>-</td>\n",
       "      <td>five_prime_utr</td>\n",
       "      <td>ENSG00000120071</td>\n",
       "      <td>KANSL1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>272 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          0         3         4  6               2          gene_id gene_name\n",
       "2181196  17  44107282  44302733  -            gene  ENSG00000120071    KANSL1\n",
       "2181197  17  44107282  44123487  -      transcript  ENSG00000120071    KANSL1\n",
       "2181198  17  44117068  44123487  -            exon  ENSG00000120071    KANSL1\n",
       "2181199  17  44116393  44116581  -            exon  ENSG00000120071    KANSL1\n",
       "2181200  17  44115904  44116052  -            exon  ENSG00000120071    KANSL1\n",
       "...      ..       ...       ... ..             ...              ...       ...\n",
       "2181463  17  44249483  44249509  -             CDS  ENSG00000120071    KANSL1\n",
       "2181464  17  44249507  44249509  -     start_codon  ENSG00000120071    KANSL1\n",
       "2181465  17  44301927  44302080  -  five_prime_utr  ENSG00000120071    KANSL1\n",
       "2181466  17  44301037  44301170  -  five_prime_utr  ENSG00000120071    KANSL1\n",
       "2181467  17  44249510  44249598  -  five_prime_utr  ENSG00000120071    KANSL1\n",
       "\n",
       "[272 rows x 7 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "parsed[parsed.gene_name == 'KANSL1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "a2265799",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>ensembl_havana</td>\n",
       "      <td>gene</td>\n",
       "      <td>11869</td>\n",
       "      <td>14412</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972\"; gene_version \"4\"; g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>havana</td>\n",
       "      <td>transcript</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972\"; gene_version \"4\"; t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>havana</td>\n",
       "      <td>exon</td>\n",
       "      <td>11869</td>\n",
       "      <td>12227</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972\"; gene_version \"4\"; t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>havana</td>\n",
       "      <td>exon</td>\n",
       "      <td>12613</td>\n",
       "      <td>12721</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972\"; gene_version \"4\"; t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>havana</td>\n",
       "      <td>exon</td>\n",
       "      <td>13221</td>\n",
       "      <td>14409</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972\"; gene_version \"4\"; t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2613760</th>\n",
       "      <td>GL000231.1</td>\n",
       "      <td>ensembl</td>\n",
       "      <td>transcript</td>\n",
       "      <td>21878</td>\n",
       "      <td>21950</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000265557\"; gene_version \"1\"; t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2613761</th>\n",
       "      <td>GL000231.1</td>\n",
       "      <td>ensembl</td>\n",
       "      <td>exon</td>\n",
       "      <td>21878</td>\n",
       "      <td>21950</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000265557\"; gene_version \"1\"; t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2613762</th>\n",
       "      <td>GL000229.1</td>\n",
       "      <td>ensembl</td>\n",
       "      <td>gene</td>\n",
       "      <td>17285</td>\n",
       "      <td>17399</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000238643\"; gene_version \"1\"; g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2613763</th>\n",
       "      <td>GL000229.1</td>\n",
       "      <td>ensembl</td>\n",
       "      <td>transcript</td>\n",
       "      <td>17285</td>\n",
       "      <td>17399</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000238643\"; gene_version \"1\"; t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2613764</th>\n",
       "      <td>GL000229.1</td>\n",
       "      <td>ensembl</td>\n",
       "      <td>exon</td>\n",
       "      <td>17285</td>\n",
       "      <td>17399</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000238643\"; gene_version \"1\"; t...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2613765 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  0               1           2      3      4  5  6  7  \\\n",
       "0                 1  ensembl_havana        gene  11869  14412  .  +  .   \n",
       "1                 1          havana  transcript  11869  14409  .  +  .   \n",
       "2                 1          havana        exon  11869  12227  .  +  .   \n",
       "3                 1          havana        exon  12613  12721  .  +  .   \n",
       "4                 1          havana        exon  13221  14409  .  +  .   \n",
       "...             ...             ...         ...    ...    ... .. .. ..   \n",
       "2613760  GL000231.1         ensembl  transcript  21878  21950  .  -  .   \n",
       "2613761  GL000231.1         ensembl        exon  21878  21950  .  -  .   \n",
       "2613762  GL000229.1         ensembl        gene  17285  17399  .  +  .   \n",
       "2613763  GL000229.1         ensembl  transcript  17285  17399  .  +  .   \n",
       "2613764  GL000229.1         ensembl        exon  17285  17399  .  +  .   \n",
       "\n",
       "                                                         8  \n",
       "0        gene_id \"ENSG00000223972\"; gene_version \"4\"; g...  \n",
       "1        gene_id \"ENSG00000223972\"; gene_version \"4\"; t...  \n",
       "2        gene_id \"ENSG00000223972\"; gene_version \"4\"; t...  \n",
       "3        gene_id \"ENSG00000223972\"; gene_version \"4\"; t...  \n",
       "4        gene_id \"ENSG00000223972\"; gene_version \"4\"; t...  \n",
       "...                                                    ...  \n",
       "2613760  gene_id \"ENSG00000265557\"; gene_version \"1\"; t...  \n",
       "2613761  gene_id \"ENSG00000265557\"; gene_version \"1\"; t...  \n",
       "2613762  gene_id \"ENSG00000238643\"; gene_version \"1\"; g...  \n",
       "2613763  gene_id \"ENSG00000238643\"; gene_version \"1\"; t...  \n",
       "2613764  gene_id \"ENSG00000238643\"; gene_version \"1\"; t...  \n",
       "\n",
       "[2613765 rows x 9 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtf[gtf[8].str.]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "183adc27",
   "metadata": {},
   "outputs": [],
   "source": [
    "t = gtf[8]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "29a622d5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [0, 1, 2, 3, 4, 5, 6, 7, 8]\n",
       "Index: []"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtf.loc[t.str.match('ENSG00000236935')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "48686f03",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [0, 1, 2, 3, 4, 5, 6, 7, 8]\n",
       "Index: []"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtf.loc[t.str.match('AP003774.1')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "23fcb31f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b16a2e21",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1597a246",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cdfb7c1f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c6dd5b6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "HiC",
   "language": "python",
   "name": "hic_tls"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
